// @ingglish/ipa/dist/index.js
// src/index.ts
import { registerFormat } from "@ingglish/phonemes";

// src/to-ipa.ts
import { findOnsetStart, getStress, isVowel, stripStress } from "@ingglish/phonemes";

// src/ipa-maps.ts
var IPA_VOWEL_MAP = {
  // Monophthongs
  AA: "\u0251",
  // father, hot, bother
  AE: "\xE6",
  // cat, bat, had
  AH: "\u028C",
  // but, cup, son (stressed)
  AO: "\u0254",
  // thought, caught, law
  // Diphthongs
  AW: "a\u028A",
  // cow, how, out
  AY: "a\u026A",
  // my, eye, time
  EH: "\u025B",
  // bed, red, said
  ER: "\u025D",
  // bird, her, nurse
  EY: "e\u026A",
  // say, day, make
  IH: "\u026A",
  // bit, sit, gym
  IY: "i",
  // bee, see, machine
  OW: "o\u028A",
  // go, show, coat
  OY: "\u0254\u026A",
  // boy, toy, coin
  UH: "\u028A",
  // book, put, could
  UW: "u"
  // too, blue, food
};
var IPA_CONSONANT_MAP = {
  // Stops (plosives)
  B: "b",
  // Affricates
  CH: "t\u0283",
  // chat, batch
  D: "d",
  // Fricatives
  DH: "\xF0",
  // the, this (voiced dental)
  F: "f",
  G: "\u0261",
  // Note: IPA uses ɡ (U+0261), not g
  HH: "h",
  JH: "d\u0292",
  // just, edge
  K: "k",
  // Liquids
  L: "l",
  // Nasals
  M: "m",
  N: "n",
  NG: "\u014B",
  // sing, thing
  P: "p",
  R: "\u0279",
  // alveolar approximant
  S: "s",
  SH: "\u0283",
  // ship
  T: "t",
  TH: "\u03B8",
  // think (voiceless dental)
  V: "v",
  // Glides (semivowels)
  W: "w",
  Y: "j",
  Z: "z",
  ZH: "\u0292"
  // measure, beige
};
var ARPABET_TO_IPA_MAP = {
  ...IPA_VOWEL_MAP,
  ...IPA_CONSONANT_MAP
};
var IPA_VARIANT_MAP = {
  a: "AE",
  // plain /a/ — maps to "a" (cat) for recognizable foreign word output
  e: "EH",
  // plain /e/ — mid front vowel, like "bed"
  \u0259: "AH0",
  // schwa (unstressed) — forward map uses ʌ→AH for the stressed variant
  \u025A: "ER",
  // r-colored schwa variant — forward map uses ɝ→ER
  g: "G",
  // ASCII g — forward map uses ɡ (U+0261)
  \u026B: "L",
  // dark l
  o: "OW",
  // some IPA uses plain o for goat vowel
  r: "R"
  // common variant — forward map uses ɹ (alveolar approximant)
  // Note: IPA /y/ is the close front rounded vowel (French "tu", German "über").
  // It's handled in IPA_APPROXIMATION_MAP as y→UW. The consonant /j/ (palatal
  // approximant) is already mapped via the forward map reversal (j→Y).
};
var IPA_APPROXIMATION_MAP = {
  // --- Open vowels ---
  \u00E4: "AA",
  // /ä/ open central — IPA diacritic variant ≈ "father"
  // --- Open vowel diphthongs (Finnish, etc.) ---
  // English uses /aɪ/ and /aʊ/ (with plain 'a'), but Finnish/other languages
  // use /ɑi/ and /ɑu/ (with open back 'ɑ'). Treat as diphthongs, not two vowels.
  \u00E6i: "AY",
  // /æi/ — Finnish "päivä" ≈ "my" diphthong (uses near-open front æ)
  \u0250: "AH",
  // /ɐ/ near-open central — Portuguese unstressed "a" ≈ "but"
  \u0251i: "AY",
  // /ɑi/ — Finnish "taivas" ≈ "my" diphthong
  \u0251u: "AW",
  // /ɑu/ — Finnish "sauna" ≈ "cow" diphthong
  \u0251\u028A: "AW",
  // /ɑʊ/ — Chinese 好 /xɑʊ/ ≈ "cow" diphthong (uses IPA ʊ not plain u)
  // Nasal vowels (ɑ̃, ɛ̃, ɔ̃, etc.) are handled in from-ipa.ts by
  // converting vowel+combining-tilde to vowel+"n" before map lookup.
  \u0252: "AO",
  // /ɒ/ open back rounded — British "lot" ≈ "thought"
  // --- Implosives and other stops ---
  \u0253: "B",
  // /ɓ/ voiced bilabial implosive ≈ B
  \u00E7: "SH",
  // /ç/ voiceless palatal fricative — "ich" (German) ≈ "sh"
  // --- Alveolo-palatal (Mandarin, Japanese, Polish) ---
  \u0255: "SH",
  // /ɕ/ voiceless alveolo-palatal fricative — Mandarin "xi" ≈ "sh"
  d\u0291: "JH",
  // /dʑ/ voiced alveolo-palatal affricate — Japanese "ji" ≈ "j"
  // --- Retroflex (Hindi, Mandarin) ---
  \u0256: "D",
  // /ɖ/ voiced retroflex stop — Hindi ≈ D
  \u0257: "D",
  // /ɗ/ voiced alveolar implosive ≈ D
  // --- Diphthong sequences (non-English vowel pairs) ---
  // These use standard IPA vowels that don't appear in English diphthongs
  // (English uses ɪ/ʊ as second element, not i/u/o).
  ei: "EY",
  // /ei/ — Finnish "ei", various ≈ "say" diphthong
  \u0258: "AH0",
  // /ɘ/ close-mid central ≈ schwa
  \u025C: "ER",
  // /ɜ/ open-mid central — non-rhotic "bird" ≈ "er"
  \u025E: "ER",
  // /ɞ/ open-mid central rounded ≈ "er"
  \u0264: "AH",
  // /ɤ/ close-mid back unrounded — Korean "ㅓ" ≈ "but"
  \u0262: "G",
  // /ɢ/ voiced uvular stop ≈ G
  \u0263: "G",
  // /ɣ/ voiced velar fricative — Spanish "lago" ≈ G
  // --- Pharyngeal (Arabic) ---
  \u0127: "HH",
  // /ħ/ voiceless pharyngeal fricative — Arabic "ha" ≈ H
  \u0266: "HH",
  // /ɦ/ voiced glottal fricative — Korean 합 ≈ H
  // --- Central/back vowels not in English ---
  \u0268: "IH",
  // /ɨ/ close central — Russian "ы" ≈ "bit"
  \u026C: "L",
  // /ɬ/ voiceless lateral fricative — Welsh "ll" ≈ L
  \u026D: "L",
  // /ɭ/ retroflex lateral ≈ L
  // --- Laterals ---
  \u028E: "L Y",
  // /ʎ/ palatal lateral — Italian "figlio", Spanish "ll" ≈ LY
  \u0271: "M",
  // /ɱ/ labiodental nasal ≈ M
  \u0274: "N",
  // /ɴ/ uvular nasal — Japanese moraic ん ≈ "n" (not "ng")
  // --- Nasals ---
  \u0272: "N Y",
  // /ɲ/ palatal nasal — Spanish "ñ", Italian "gn" ≈ NY
  // tɕ and dʑ are handled as two-char sequences below
  \u0273: "N",
  // /ɳ/ retroflex nasal ≈ N
  \u00F8: "UH",
  // /ø/ close-mid front rounded — "peu" (French), "schön" (German) ≈ "u"
  \u0153: "AH1",
  // /œ/ open-mid front rounded — "peur" (French) ≈ "uh"
  \u0153y: "OY",
  // /œy/ — Dutch "huis" diphthong ≈ "boy"
  oi: "OY",
  // /oi/ — Finnish "koira", Portuguese "coisa" ≈ "boy" diphthong
  ou: "OW",
  // /ou/ — Finnish "koulu" ≈ "go" diphthong
  o\u026F: "OW",
  // /oɯ/ — Japanese long /oː/ (dict convention: 東京 /toɯkjoɯ/) ≈ "oh"
  \u0278: "F",
  // /ɸ/ voiceless bilabial fricative — Japanese "fu" ≈ F
  q: "K",
  // /q/ voiceless uvular stop — Arabic "Quran" ≈ K
  \u0280: "R",
  // /ʀ/ uvular trill — some German dialects ≈ English R
  // --- Taps, trills, and approximants (Spanish, Italian, Mandarin, etc.) ---
  \u027B: "R",
  // /ɻ/ voiced retroflex approximant — Mandarin "er" (二) ≈ R
  \u027D: "D",
  // /ɽ/ retroflex flap — Hindi ≈ D
  \u027E: "R",
  // /ɾ/ alveolar tap — Spanish "pero" ≈ R (also like "butter" flap)
  // --- Uvular consonants (French, German, Arabic) ---
  \u0281: "R",
  // /ʁ/ voiced uvular fricative — French/German R ≈ English R
  \u0282: "SH",
  // /ʂ/ voiceless retroflex fricative — Mandarin "shi" ≈ "sh"
  // --- Affricates (two-char sequences matched by the converter) ---
  t\u0255: "CH",
  // /tɕ/ voiceless alveolo-palatal affricate — Mandarin "ji", Korean "ㅈ" ≈ "ch"
  \u0288: "T",
  // /ʈ/ voiceless retroflex stop — Hindi ≈ T
  \u0288\u0282: "CH",
  // /ʈʂ/ voiceless retroflex affricate — Mandarin "zhi" ≈ "ch"
  // --- Labial-velar ---
  \u0265: "W",
  // /ɥ/ labial-palatal approximant — French "lui" ≈ W
  \u026F: "UH",
  // /ɯ/ close back unrounded — Japanese "u", Turkish "ı" ≈ "book" (shorter than "oo")
  \u0270: "W",
  // /ɰ/ voiced velar approximant — Japanese 川 /kaɰᵝa/ ≈ W
  \u028B: "V",
  // /ʋ/ labiodental approximant — Hindi, Dutch ≈ V
  // --- Velar/palatal fricatives (German, Mandarin, etc.) ---
  x: "HH",
  // /x/ voiceless velar fricative — default H (override to K for German)
  // --- Front rounded vowels (French, German, Turkish, etc.) ---
  y: "UW",
  // /y/ close front rounded — "tu" (French) ≈ "too"
  \u028F: "UH",
  // /ʏ/ near-close front rounded — "Glück" (German) ≈ "book"
  \u0290: "ZH",
  // /ʐ/ voiced retroflex fricative — Mandarin "ri" ≈ "zh"
  \u0291: "ZH",
  // /ʑ/ voiced alveolo-palatal fricative ≈ "zh"
  // --- Glottal ---
  \u0294: "",
  // /ʔ/ glottal stop — often silent in approximation
  // Note: length mark ː and combining diacritics are stripped in from-ipa.ts
  // before lookup. Nasal vowels (◌̃) are converted to vowel+n there.
  // --- Dental fricatives already in English ---
  // θ and ð are in the main map
  \u0295: "AH",
  // /ʕ/ voiced pharyngeal fricative — Arabic "ain" ≈ "uh" (rough)
  // --- Bilabial fricatives (Spanish) ---
  \u03B2: "V",
  // /β/ voiced bilabial fricative — Spanish "b" between vowels ≈ V
  \u03C7: "HH"
  // /χ/ voiceless uvular fricative — default H (override to K for German)
};
var IPA_TO_ARPABET_MAP = {
  ...IPA_APPROXIMATION_MAP,
  ...Object.fromEntries(Object.entries(ARPABET_TO_IPA_MAP).map(([arpabet, ipa]) => [ipa, arpabet])),
  ...IPA_VARIANT_MAP
};
var IPA_LANGUAGE_OVERRIDES = {
  // German: /x/ and /χ/ are the "ach-Laut" — English speakers say "bahk" for Bach
  de: { x: "K", \u03C7: "K" },
  // Dutch: /ɣ/ is a breathy G (closer to H than hard G) — "goed" ≈ "hood" not "good"
  nl: { \u0263: "HH" }
};

// src/to-ipa.ts
var WORD_JOINER = "\u2060";
var STRESS_MARKERS = {
  0: "",
  1: WORD_JOINER + "\u02C8" + WORD_JOINER,
  2: WORD_JOINER + "\u02CC" + WORD_JOINER
};
function arpabetPhonemeToIPA(phoneme) {
  const base = stripStress(phoneme);
  const stress = getStress(phoneme);
  const ipa = ARPABET_TO_IPA_MAP[base];
  if (ipa === void 0) {
    return phoneme.toLowerCase();
  }
  if (base === "AH" && stress === 0) {
    return "\u0259";
  }
  if (stress === 1 || stress === 2) {
    return STRESS_MARKERS[stress] + ipa;
  }
  return ipa;
}
function arpabetToIPA(arpabet) {
  const ipaSegments = [];
  const stressPositions = [];
  for (let i = 0; i < arpabet.length; i++) {
    const symbol = arpabet[i];
    const base = stripStress(symbol);
    const stress = getStress(symbol);
    const ipa = ARPABET_TO_IPA_MAP[base];
    if (ipa === void 0) {
      ipaSegments.push(symbol.toLowerCase());
      continue;
    }
    if (base === "AH" && stress === 0) {
      ipaSegments.push("\u0259");
      continue;
    }
    if (stress === 1 || stress === 2) {
      const marker = stress === 1 ? WORD_JOINER + "\u02C8" + WORD_JOINER : WORD_JOINER + "\u02CC" + WORD_JOINER;
      let onsetIndex = ipaSegments.length;
      if (i > 0) {
        let j = i - 1;
        const consonants = [];
        while (j >= 0 && !isVowel(arpabet[j])) {
          consonants.push(stripStress(arpabet[j]));
          j--;
        }
        consonants.reverse();
        if (consonants.length > 0) {
          const onsetStartInCluster = findOnsetStart(consonants);
          onsetIndex = j + 1 + onsetStartInCluster;
        }
      }
      stressPositions.push({ index: onsetIndex, marker });
    }
    ipaSegments.push(ipa);
  }
  const sortedStress = stressPositions.toSorted((a, b) => a.index - b.index);
  const result = [];
  let stressIdx = 0;
  for (let i = 0; i <= ipaSegments.length; i++) {
    while (stressIdx < sortedStress.length && sortedStress[stressIdx].index === i) {
      result.push(sortedStress[stressIdx].marker);
      stressIdx++;
    }
    if (i < ipaSegments.length) {
      result.push(ipaSegments[i]);
    }
  }
  return `/${result.join("")}/`;
}
function arpabetToIPARaw(arpabet) {
  const full = arpabetToIPA(arpabet);
  return full.slice(1, -1);
}

// src/foreign.ts
import {
  applyCasePattern,
  detectCasePattern,
  normalizeApostrophes,
  stripDiacritics
} from "@ingglish/normalize";
import {
  arpabetToFormat,
  arpabetToIngglish,
  getFormatPreservesCase,
  getStress as getStress2,
  isVowel as isVowel2
} from "@ingglish/phonemes";

// src/from-ipa.ts
import { stripStress as stripStress2 } from "@ingglish/phonemes";
var ARPABET_VOWELS = new Set(Object.keys(IPA_VOWEL_MAP));
var NASAL_VOWEL_RE = /([aeiouɑɛɔəɐɒæøœʌɝɚɘɜɞɤʏʊɪɨɯy])\u0303/g;
var MODIFIER_RE = /[\u02B0\u02D0\u02D1\u02E5-\u02E9\u0303\u1D5D]/g;
var COMBINING_NON_CEDILLA_RE = /(?!\u0327)\p{Mn}/gu;
var mergedMapCache = /* @__PURE__ */ new Map();
function ipaToArpabet(ipa, overrides) {
  const normalized = ipa.normalize("NFD");
  const denasalized = normalized.replaceAll(NASAL_VOWEL_RE, "$1n");
  const stripped = denasalized.replaceAll(MODIFIER_RE, "");
  const stripped2 = stripped.replaceAll(COMBINING_NON_CEDILLA_RE, "");
  const clean = stripped2.normalize("NFC");
  let map = IPA_TO_ARPABET_MAP;
  if (overrides) {
    let cached = mergedMapCache.get(overrides);
    if (!cached) {
      cached = { ...IPA_TO_ARPABET_MAP, ...overrides };
      mergedMapCache.set(overrides, cached);
    }
    map = cached;
  }
  const result = [];
  let pendingStress = null;
  let i = 0;
  const push = (phoneme) => {
    const base = stripStress2(phoneme);
    if (ARPABET_VOWELS.has(base) && pendingStress !== null) {
      result.push(base + String(pendingStress));
      pendingStress = null;
    } else {
      result.push(phoneme);
    }
  };
  while (i < clean.length) {
    const ch = clean[i];
    if (ch === "\u02C8") {
      pendingStress = 1;
      i++;
      continue;
    }
    if (ch === "\u02CC") {
      pendingStress = 2;
      i++;
      continue;
    }
    if (i + 1 < clean.length) {
      const twoChar = clean.slice(i, i + 2);
      const twoCharArpabet = map[twoChar];
      if (twoCharArpabet !== void 0) {
        if (twoCharArpabet.includes(" ")) {
          for (const p of twoCharArpabet.split(" ")) {
            push(p);
          }
        } else {
          push(twoCharArpabet);
        }
        i += 2;
        continue;
      }
    }
    const oneCharArpabet = map[ch];
    if (oneCharArpabet !== void 0) {
      if (oneCharArpabet.includes(" ")) {
        for (const p of oneCharArpabet.split(" ")) {
          push(p);
        }
      } else {
        push(oneCharArpabet);
      }
    }
    i++;
  }
  const deduped = [];
  for (const phoneme of result) {
    if (phoneme !== deduped.at(-1)) {
      deduped.push(phoneme);
    }
  }
  return deduped;
}
function ipaToArpabetClean(ipa) {
  const arpabet = ipaToArpabet(ipa).map((p) => stripStress2(p));
  return arpabet.length > 0 ? arpabet : null;
}

// src/g2p.ts
var G2P_CONVERTERS = {
  eo: esperantoG2P,
  fi: finnishG2P,
  ma: malayG2P,
  sw: swahiliG2P
};
var IPA_VOWELS = new Set("aeiou\u0251\xE6\xF8y\u025B\u0254\u0259");
function addFirstSyllableStress(ipa) {
  return ipa ? "\u02C8" + ipa : ipa;
}
function addPenultimateStress(ipa) {
  const vowelPositions = [];
  for (const [i, ch] of Array.from(ipa).entries()) {
    if (IPA_VOWELS.has(ch)) {
      vowelPositions.push(i);
    }
  }
  if (vowelPositions.length <= 1) {
    return ipa ? "\u02C8" + ipa : ipa;
  }
  const stressPos = vowelPositions.at(-2);
  let onset = stressPos;
  while (onset > 0 && !IPA_VOWELS.has(ipa[onset - 1]) && ipa[onset - 1] !== "\u02D0") {
    onset--;
  }
  return ipa.slice(0, onset) + "\u02C8" + ipa.slice(onset);
}
function applyRules(word, rules) {
  const normalized = word.normalize("NFC");
  let result = "";
  let i = 0;
  while (i < normalized.length) {
    let matched = false;
    for (const [grapheme, phoneme] of rules) {
      if (normalized.startsWith(grapheme, i)) {
        result += phoneme;
        i += grapheme.length;
        matched = true;
        break;
      }
    }
    if (!matched) {
      i++;
    }
  }
  return result;
}
var FINNISH_RULES = [
  // Digraphs (must precede single-char rules)
  ["nk", "\u014Bk"],
  ["ng", "\u014B\u02D0"],
  // Long vowels (double letters → vowel + length mark)
  ["aa", "\u0251\u02D0"],
  ["ee", "e\u02D0"],
  ["ii", "i\u02D0"],
  ["oo", "o\u02D0"],
  ["uu", "u\u02D0"],
  ["yy", "y\u02D0"],
  ["\xE4\xE4", "\xE6\u02D0"],
  ["\xF6\xF6", "\xF8\u02D0"],
  // Geminate consonants
  ["pp", "p\u02D0"],
  ["tt", "t\u02D0"],
  ["kk", "k\u02D0"],
  ["mm", "m\u02D0"],
  ["nn", "n\u02D0"],
  ["ll", "l\u02D0"],
  ["rr", "r\u02D0"],
  ["ss", "s\u02D0"],
  // Single vowels
  ["a", "\u0251"],
  ["e", "e"],
  ["i", "i"],
  ["o", "o"],
  ["u", "u"],
  ["y", "y"],
  ["\xE4", "\xE6"],
  ["\xF6", "\xF8"],
  // Single consonants
  ["b", "b"],
  ["d", "d"],
  ["f", "f"],
  ["g", "\u0261"],
  ["h", "h"],
  ["j", "j"],
  ["k", "k"],
  ["l", "l"],
  ["m", "m"],
  ["n", "n"],
  ["p", "p"],
  ["r", "r"],
  ["s", "s"],
  ["t", "t"],
  ["v", "\u028B"],
  ["w", "\u028B"],
  ["z", "ts"]
];
function finnishG2P(word) {
  return addFirstSyllableStress(applyRules(word, FINNISH_RULES));
}
var ESPERANTO_RULES = [
  // Special Esperanto characters (must precede base letters)
  ["\u0109", "t\u0283"],
  ["\u011D", "d\u0292"],
  ["\u0125", "x"],
  ["\u0135", "\u0292"],
  ["\u015D", "\u0283"],
  ["\u016D", "w"],
  ["c", "ts"],
  // Vowels
  ["a", "a"],
  ["e", "e"],
  ["i", "i"],
  ["o", "o"],
  ["u", "u"],
  // Consonants
  ["b", "b"],
  ["d", "d"],
  ["f", "f"],
  ["g", "\u0261"],
  ["h", "h"],
  ["j", "j"],
  ["k", "k"],
  ["l", "l"],
  ["m", "m"],
  ["n", "n"],
  ["p", "p"],
  ["r", "r"],
  ["s", "s"],
  ["t", "t"],
  ["v", "v"],
  ["z", "z"]
];
function esperantoG2P(word) {
  return addPenultimateStress(applyRules(word, ESPERANTO_RULES));
}
var SWAHILI_RULES = [
  // Trigraph (must precede digraph 'ng')
  ["ng'", "\u014B"],
  // Digraphs
  ["ch", "t\u0283"],
  ["dh", "\xF0"],
  ["gh", "\u0263"],
  ["ng", "\u014B\u0261"],
  ["nj", "nd\u0292"],
  ["ny", "\u0272"],
  ["sh", "\u0283"],
  ["th", "\u03B8"],
  // Vowels
  ["a", "a"],
  ["e", "\u025B"],
  ["i", "i"],
  ["o", "\u0254"],
  ["u", "u"],
  // Consonants
  ["b", "b"],
  ["d", "d"],
  ["f", "f"],
  ["g", "\u0261"],
  ["h", "h"],
  ["j", "d\u0292"],
  ["k", "k"],
  ["l", "l"],
  ["m", "m"],
  ["n", "n"],
  ["p", "p"],
  ["r", "\u027E"],
  ["s", "s"],
  ["t", "t"],
  ["v", "v"],
  ["w", "w"],
  ["y", "j"],
  ["z", "z"]
];
function swahiliG2P(word) {
  return addPenultimateStress(applyRules(word, SWAHILI_RULES));
}
var MALAY_RULES = [
  // Digraphs
  ["gh", "\u0263"],
  ["kh", "x"],
  ["ng", "\u014B"],
  ["ny", "\u0272"],
  ["sy", "\u0283"],
  // Vowels
  ["a", "a"],
  ["e", "\u0259"],
  ["i", "i"],
  ["o", "o"],
  ["u", "u"],
  // Consonants
  ["b", "b"],
  ["c", "t\u0283"],
  ["d", "d"],
  ["f", "f"],
  ["g", "\u0261"],
  ["h", "h"],
  ["j", "d\u0292"],
  ["k", "k"],
  ["l", "l"],
  ["m", "m"],
  ["n", "n"],
  ["p", "p"],
  ["r", "\u027E"],
  ["s", "s"],
  ["t", "t"],
  ["v", "v"],
  ["w", "w"],
  ["y", "j"],
  ["z", "z"]
];
function addMalayStress(ipa) {
  const vowelPositions = [];
  for (const [i, ch] of Array.from(ipa).entries()) {
    if (IPA_VOWELS.has(ch)) {
      vowelPositions.push(i);
    }
  }
  if (vowelPositions.length <= 1) {
    return ipa ? "\u02C8" + ipa : ipa;
  }
  const penultPos = vowelPositions.at(-2);
  const stressPos = ipa[penultPos] === "\u0259" ? vowelPositions.at(-1) : penultPos;
  let onset = stressPos;
  while (onset > 0 && !IPA_VOWELS.has(ipa[onset - 1]) && ipa[onset - 1] !== "\u02D0") {
    onset--;
  }
  return ipa.slice(0, onset) + "\u02C8" + ipa.slice(onset);
}
function malayG2P(word) {
  return addMalayStress(applyRules(word, MALAY_RULES));
}

// src/lemmatizers.ts
var LEMMATIZERS = {
  eo: lemmatizeEo,
  fa: lemmatizeFa,
  fi: lemmatizeFi,
  ma: lemmatizeMa,
  nb: lemmatizeNb,
  ro: lemmatizeRo,
  sv: lemmatizeSv,
  sw: lemmatizeSw
};
function tryLookup(dict, ...candidates) {
  for (const c of candidates) {
    if (c && dict[c]) {
      return dict[c];
    }
  }
  return void 0;
}
var SV_SUFFIXES = [
  // 4+ char suffixes
  ["erna", ["", "e"]],
  ["orna", ["", "a"]],
  ["ande", ["", "a"]],
  ["ende", ["", "a"]],
  ["aste", [""]],
  // 3 char
  ["ade", ["", "a"]],
  ["igt", ["ig"]],
  // 2 char
  ["en", [""]],
  ["et", ["", "e"]],
  ["an", ["", "a"]],
  ["ar", [""]],
  ["er", ["", "a"]],
  ["de", ["", "a"]],
  ["te", ["", "a"]],
  // 1 char
  ["a", [""]],
  ["t", ["", "a"]],
  ["s", [""]],
  ["r", ["", "a"]],
  ["n", [""]]
];
function lemmatizeSv(dict, word) {
  for (const [suffix, replacements] of SV_SUFFIXES) {
    if (word.length > suffix.length && word.endsWith(suffix)) {
      const stem = word.slice(0, -suffix.length);
      const candidates = replacements.map((r) => stem + r);
      const ipa = tryLookup(dict, ...candidates);
      if (ipa) {
        return ipa;
      }
    }
  }
  if (word.endsWith("s") && word.length > 2) {
    const inner = word.slice(0, -1);
    if (dict[inner]) {
      return dict[inner];
    }
    return lemmatizeSv(dict, inner);
  }
  return void 0;
}
var RO_SUFFIXES = [
  // 4+ char
  ["ului", [""]],
  ["ilor", [""]],
  ["e\u0219te", ["", "i"]],
  // 3 char
  ["ele", ["", "\u0103"]],
  ["uri", [""]],
  ["eau", ["", "i", "ea"]],
  // 2 char
  ["ul", [""]],
  ["ii", ["", "ie", "iu"]],
  ["ea", ["", "e"]],
  ["\u021Bi", ["t"]],
  ["c\u0103", ["c", "ca"]],
  // 1 char
  ["a", ["", "\u0103"]],
  ["e", ["", "\u0103"]],
  ["i", ["", "e"]]
];
var RO_PREFIX_RESTORE = [
  ["n", "\xEEn"],
  ["l", "\xEEl"],
  ["m", "m\u0103"]
];
function lemmatizeRo(dict, word) {
  for (const [suffix, replacements] of RO_SUFFIXES) {
    if (word.length > suffix.length && word.endsWith(suffix)) {
      const stem = word.slice(0, -suffix.length);
      const candidates = replacements.map((r) => stem + r);
      const ipa = tryLookup(dict, ...candidates);
      if (ipa) {
        return ipa;
      }
    }
  }
  for (const [prefix, restored] of RO_PREFIX_RESTORE) {
    if (word.startsWith(prefix)) {
      const remainder = restored + word.slice(prefix.length);
      if (dict[remainder]) {
        return dict[remainder];
      }
    }
  }
  const withI = "\xEE" + word;
  if (dict[withI]) {
    return dict[withI];
  }
  return void 0;
}
var SW_VERB_PREFIXES = [
  // 5+ char
  "hatuku",
  "hawaku",
  "haiku",
  "hatua",
  "hatui",
  // 4 char
  "wali",
  "tuli",
  "nili",
  "aali",
  "wame",
  "tume",
  "nime",
  "ame",
  "wana",
  "tuna",
  "nina",
  "ana",
  "wata",
  "tuta",
  "nita",
  "ata",
  "yame",
  "yata",
  "yana",
  "yali",
  "kime",
  "kita",
  "kina",
  "kili",
  "lime",
  "lita",
  "lina",
  "lili",
  "vime",
  "vita",
  "vina",
  "vili",
  "zime",
  "zita",
  "zina",
  "zili",
  "haku",
  "hatu",
  "hani",
  "hawa",
  // 3 char
  "ali",
  "uli",
  "ume",
  "una",
  "uta",
  "tua",
  "tui",
  "wai",
  "wal",
  "iku",
  "ina",
  "hue",
  "huj",
  "hui",
  "yat",
  "yam",
  "yan",
  "kum",
  "kui",
  "kua",
  // 2 char
  "wa",
  "tu",
  "ni",
  "li",
  "ki",
  "vi",
  "zi",
  "ya",
  "ku",
  "hu"
];
var SW_DERIV_SUFFIXES = [
  ["ika", ["a"]],
  ["isha", ["a"]],
  ["ana", ["a"]],
  ["wa", ["a"]],
  ["ia", ["a"]],
  ["ika", ["a", "ea"]]
];
function lemmatizeEo(dict, word) {
  let w = word;
  if (w.endsWith("n") && w.length > 2) {
    const stripped = w.slice(0, -1);
    if (dict[stripped]) {
      return dict[stripped];
    }
    w = stripped;
  }
  if (w.endsWith("j") && w.length > 2) {
    const stripped = w.slice(0, -1);
    if (dict[stripped]) {
      return dict[stripped];
    }
    w = stripped;
  }
  if (dict[w]) {
    return dict[w];
  }
  for (const ending of ["as", "is", "os", "us"]) {
    if (w.endsWith(ending) && w.length > ending.length + 1) {
      const stem = w.slice(0, -ending.length);
      const ipa = tryLookup(dict, stem + "i", stem + "o", stem);
      if (ipa) {
        return ipa;
      }
    }
  }
  if (w.endsWith("u") && w.length > 2) {
    const stem = w.slice(0, -1);
    const ipa = tryLookup(dict, stem + "i", stem + "o", stem);
    if (ipa) {
      return ipa;
    }
  }
  for (const suffix of [
    "anta",
    "inta",
    "onta",
    "ata",
    "ita",
    "ota",
    "ante",
    "inte",
    "onte"
  ]) {
    if (w.endsWith(suffix) && w.length > suffix.length + 1) {
      const stem = w.slice(0, -suffix.length);
      const ipa = tryLookup(dict, stem + "i", stem + "o", stem);
      if (ipa) {
        return ipa;
      }
    }
  }
  if (w.endsWith("e") && w.length > 2) {
    const stem = w.slice(0, -1);
    const ipa = tryLookup(dict, stem + "o", stem + "a", stem + "i", stem);
    if (ipa) {
      return ipa;
    }
  }
  for (const [suffix, replacements] of [
    ["isto", ["o", "i", ""]],
    ["ejo", ["o", "i", ""]],
    ["ilo", ["o", "i", ""]],
    ["eco", ["o", "a", ""]],
    ["ado", ["o", "i", ""]],
    ["igo", ["o", "a", ""]],
    ["i\u011Do", ["o", "a", ""]]
  ]) {
    if (w.endsWith(suffix) && w.length > suffix.length + 1) {
      const stem = w.slice(0, -suffix.length);
      const ipa = tryLookup(dict, ...replacements.map((r) => stem + r));
      if (ipa) {
        return ipa;
      }
    }
  }
  for (const prefix of ["mal", "ek", "re", "ne", "sen"]) {
    if (w.startsWith(prefix) && w.length > prefix.length + 1) {
      const remainder = w.slice(prefix.length);
      if (dict[remainder]) {
        return dict[remainder];
      }
      const inner = lemmatizeEo(dict, remainder);
      if (inner) {
        return inner;
      }
    }
  }
  return void 0;
}
function lemmatizeSw(dict, word) {
  for (const prefix of SW_VERB_PREFIXES) {
    if (word.length > prefix.length + 1 && word.startsWith(prefix)) {
      const remainder = word.slice(prefix.length);
      if (dict[remainder]) {
        return dict[remainder];
      }
      const kuForm = "ku" + remainder;
      if (dict[kuForm]) {
        return dict[kuForm];
      }
      for (const [suffix, replacements] of SW_DERIV_SUFFIXES) {
        if (remainder.length > suffix.length && remainder.endsWith(suffix)) {
          const stem = remainder.slice(0, -suffix.length);
          for (const r of replacements) {
            const candidate = stem + r;
            if (dict[candidate]) {
              return dict[candidate];
            }
            if (dict["ku" + candidate]) {
              return dict["ku" + candidate];
            }
          }
        }
      }
    }
  }
  for (const [suffix, replacements] of SW_DERIV_SUFFIXES) {
    if (word.length > suffix.length && word.endsWith(suffix)) {
      const stem = word.slice(0, -suffix.length);
      for (const r of replacements) {
        const candidate = stem + r;
        if (dict[candidate]) {
          return dict[candidate];
        }
      }
    }
  }
  return void 0;
}
var FI_SUFFIXES = [
  // Possessive + case combinations
  ["ssani", ["", "nen"]],
  ["ss\xE4ni", ["", "nen"]],
  ["llani", ["", "nen"]],
  ["ll\xE4ni", ["", "nen"]],
  ["stani", ["", "nen"]],
  ["st\xE4ni", ["", "nen"]],
  ["ssaan", ["", "nen"]],
  ["ss\xE4\xE4n", ["", "nen"]],
  // Plural case endings (4+ chars)
  ["issa", ["", "a"]],
  ["iss\xE4", ["", "\xE4"]],
  ["illa", ["", "a"]],
  ["ill\xE4", ["", "\xE4"]],
  ["ista", ["", "a"]],
  ["ist\xE4", ["", "\xE4"]],
  ["ihin", ["", "i"]],
  ["ojen", ["o"]],
  ["ujen", ["u"]],
  ["yjen", ["y"]],
  ["iden", ["i"]],
  ["jen", [""]],
  // Inessive -ssa/-ssä
  ["ssa", ["", "s"]],
  ["ss\xE4", ["", "s"]],
  // Elative -sta/-stä
  ["sta", ["", "s"]],
  ["st\xE4", ["", "s"]],
  // Adessive -lla/-llä
  ["lla", ["", "a"]],
  ["ll\xE4", ["", "\xE4"]],
  // Ablative -lta/-ltä
  ["lta", ["", "a"]],
  ["lt\xE4", ["", "\xE4"]],
  // Allative -lle
  ["lle", ["", "i"]],
  // Essive -na/-nä
  ["na", ["", "nen"]],
  ["n\xE4", ["", "nen"]],
  // Translative -ksi
  ["ksi", ["", "si"]],
  // Possessive -ni, -si, -nsa/-nsä, -mme, -nne
  ["nsa", [""]],
  ["ns\xE4", [""]],
  ["mme", [""]],
  ["nne", [""]],
  ["ni", ["", "n"]],
  ["si", ["", "s"]],
  // Partitive -a/-ä, -ta/-tä, -tta/-ttä
  ["tta", [""]],
  ["tt\xE4", [""]],
  ["ta", ["", "nen"]],
  ["t\xE4", ["", "nen"]],
  // Genitive -n, plural -t
  ["en", ["", "i"]],
  ["ot", ["o"]],
  ["ut", ["u"]],
  ["yt", ["y"]],
  ["\xE4t", ["\xE4"]],
  ["at", ["a"]],
  ["et", ["e", "i"]],
  // Verb past -i
  ["oi", ["o", "oa"]],
  ["ui", ["u", "ua"]],
  // General fallbacks
  ["a", [""]],
  ["\xE4", [""]],
  ["n", [""]],
  ["t", [""]]
];
var FI_VERB_SUFFIXES = [
  // Archaic Kalevala -(tt)elevi/-(tt)avi patterns
  ["ttelevi", ["tella", "della"]],
  ["televi", ["tella", "della"]],
  ["ttavi", ["ttaa", "t\xE4\xE4"]],
  ["ttevi", ["tt\xE4\xE4", "ttaa"]],
  ["elevi", ["ella", "ell\xE4"]],
  ["alevi", ["alla", "all\xE4"]],
  ["evi", ["", "a", "\xE4"]],
  ["avi", ["", "a", "aa"]],
  ["ovi", ["", "o", "oa"]],
  ["uvi", ["", "u", "ua"]],
  // Past participle -nut/-nyt, -neet
  ["neet", ["", "a", "\xE4"]],
  ["nut", ["", "a", "da"]],
  ["nyt", ["", "\xE4", "d\xE4"]],
  // Present participle -va/-vä
  ["va", ["", "a"]],
  ["v\xE4", ["", "\xE4"]],
  // Past tense 3rd person
  ["tui", ["tua", "ty\xE4"]],
  ["lui", ["la", "l\xE4"]],
  // Conditional
  ["isi", ["", "a", "\xE4"]],
  // Agent noun -ja/-jä
  ["ja", ["", "a"]],
  ["j\xE4", ["", "\xE4"]]
];
function applyFiGradation(stem) {
  if (stem.endsWith("nt")) {
    return stem.slice(0, -2) + "nn";
  }
  if (stem.endsWith("lt")) {
    return stem.slice(0, -2) + "ll";
  }
  if (stem.endsWith("rt")) {
    return stem.slice(0, -2) + "rr";
  }
  if (stem.endsWith("nk")) {
    return stem.slice(0, -2) + "ng";
  }
  if (stem.endsWith("mp")) {
    return stem.slice(0, -2) + "mm";
  }
  if (stem.endsWith("lk")) {
    return stem.slice(0, -2) + "l";
  }
  if (stem.endsWith("rk")) {
    return stem.slice(0, -2) + "r";
  }
  if (stem.endsWith("hk")) {
    return stem.slice(0, -2) + "h";
  }
  return stem;
}
function applyFiStrengthening(stem) {
  if (stem.endsWith("nn")) {
    return stem.slice(0, -2) + "nt";
  }
  if (stem.endsWith("ll")) {
    return stem.slice(0, -2) + "lt";
  }
  if (stem.endsWith("rr")) {
    return stem.slice(0, -2) + "rt";
  }
  if (stem.endsWith("ng")) {
    return stem.slice(0, -2) + "nk";
  }
  if (stem.endsWith("mm")) {
    return stem.slice(0, -2) + "mp";
  }
  return stem;
}
function lemmatizeFi(dict, word) {
  for (const [suffix, replacements] of FI_VERB_SUFFIXES) {
    if (word.length > suffix.length + 1 && word.endsWith(suffix)) {
      const stem = word.slice(0, -suffix.length);
      const candidates = replacements.map((r) => stem + r);
      const ipa = tryLookup(dict, ...candidates);
      if (ipa) {
        return ipa;
      }
    }
  }
  for (const [suffix, replacements] of FI_SUFFIXES) {
    if (word.length > suffix.length + 1 && word.endsWith(suffix)) {
      const stem = word.slice(0, -suffix.length);
      const candidates = replacements.map((r) => stem + r);
      const ipa = tryLookup(dict, ...candidates);
      if (ipa) {
        return ipa;
      }
      if (stem.length >= 2) {
        const gradated = applyFiGradation(stem);
        if (gradated !== stem) {
          const gradIpa = tryLookup(dict, ...replacements.map((r) => gradated + r));
          if (gradIpa) {
            return gradIpa;
          }
        }
        const strengthened = applyFiStrengthening(stem);
        if (strengthened !== stem) {
          const strIpa = tryLookup(dict, ...replacements.map((r) => strengthened + r));
          if (strIpa) {
            return strIpa;
          }
        }
      }
    }
  }
  for (const poss of ["ni", "si", "nsa", "ns\xE4", "mme", "nne"]) {
    if (word.endsWith(poss) && word.length > poss.length + 2) {
      const inner = word.slice(0, -poss.length);
      if (dict[inner]) {
        return dict[inner];
      }
      for (const [suffix, replacements] of FI_SUFFIXES) {
        if (inner.length > suffix.length + 1 && inner.endsWith(suffix)) {
          const stem = inner.slice(0, -suffix.length);
          const ipa = tryLookup(dict, ...replacements.map((r) => stem + r));
          if (ipa) {
            return ipa;
          }
        }
      }
    }
  }
  return void 0;
}
function modernizeNb(word) {
  const variants = [];
  if (word.includes("aa")) {
    variants.push(word.replaceAll("aa", "\xE5"));
  }
  if (word.includes("Aa")) {
    variants.push(word.replaceAll("Aa", "\xC5"));
  }
  if (word === "af") {
    variants.push("av");
  }
  if (word === "efter") {
    variants.push("etter");
  }
  if (word === "imod") {
    variants.push("imot");
  }
  return variants;
}
var NB_SUFFIXES = [
  // 4+ char
  ["erne", ["", "e"]],
  ["enes", ["", "e"]],
  ["ande", ["", "e"]],
  ["ende", ["", "e"]],
  ["else", ["", "e"]],
  // Definite plural
  ["ene", ["", "e"]],
  ["ane", ["", "e"]],
  // 2 char definite singular
  ["en", ["", "e"]],
  ["et", ["", "e"]],
  // Past tense / participle
  ["te", ["", "e"]],
  ["de", ["", "e"]],
  ["dde", [""]],
  // Indefinite plural
  ["er", ["", "e"]],
  // Comparative / superlative
  ["ere", [""]],
  ["est", [""]],
  ["este", [""]],
  // Present tense
  ["ar", ["", "e"]],
  // General
  ["t", ["", "e"]],
  ["a", ["", "e"]],
  ["s", [""]],
  ["e", [""]],
  ["n", ["", "e"]],
  ["r", ["", "e"]]
];
function lemmatizeNb(dict, word) {
  const modern = modernizeNb(word);
  for (const m of modern) {
    if (dict[m]) {
      return dict[m];
    }
    const lower = m.toLowerCase();
    if (dict[lower]) {
      return dict[lower];
    }
  }
  for (const [suffix, replacements] of NB_SUFFIXES) {
    if (word.length > suffix.length + 1 && word.endsWith(suffix)) {
      const stem = word.slice(0, -suffix.length);
      const ipa = tryLookup(dict, ...replacements.map((r) => stem + r));
      if (ipa) {
        return ipa;
      }
    }
  }
  for (const [suffix, replacements] of NB_SUFFIXES) {
    if (word.length > suffix.length + 1 && word.endsWith(suffix)) {
      const stem = word.slice(0, -suffix.length);
      for (const r of replacements) {
        const candidate = stem + r;
        const modernized = modernizeNb(candidate);
        for (const m of modernized) {
          if (dict[m]) {
            return dict[m];
          }
        }
      }
    }
  }
  return void 0;
}
var MA_SUFFIXES = ["nya", "mu", "ku", "kan", "an", "lah", "kah", "i"];
var MA_PREFIXES = [
  ["memper", [""]],
  ["member", [""]],
  ["menge", [""]],
  ["meny", ["s", "c"]],
  ["meng", ["k", "g", "h", ""]],
  ["mem", ["p", "b", "f", ""]],
  ["men", ["t", "d", "c", "j", ""]],
  ["me", [""]],
  ["diper", [""]],
  ["ber", [""]],
  ["per", [""]],
  ["ter", [""]],
  ["di", [""]],
  ["ke", [""]],
  ["se", [""]],
  ["ku", [""]]
];
function lemmatizeMa(dict, word) {
  for (const suffix of MA_SUFFIXES) {
    if (word.length > suffix.length + 2 && word.endsWith(suffix)) {
      const stem = word.slice(0, -suffix.length);
      if (dict[stem]) {
        return dict[stem];
      }
      const fromPrefix = tryMaPrefixStrip(dict, stem);
      if (fromPrefix) {
        return fromPrefix;
      }
    }
  }
  return tryMaPrefixStrip(dict, word);
}
function tryMaPrefixStrip(dict, word) {
  for (const [prefix, restorations] of MA_PREFIXES) {
    if (word.length > prefix.length + 1 && word.startsWith(prefix)) {
      const remainder = word.slice(prefix.length);
      if (dict[remainder]) {
        return dict[remainder];
      }
      for (const consonant of restorations) {
        if (consonant) {
          const restored = consonant + remainder;
          if (dict[restored]) {
            return dict[restored];
          }
        }
      }
    }
  }
  return void 0;
}
var ZWNJ = "\u200C";
function lemmatizeFa(dict, word) {
  if (word.includes(ZWNJ)) {
    const parts = word.split(ZWNJ);
    for (const part of parts) {
      if (dict[part]) {
        return dict[part];
      }
    }
    if (parts.length === 2 && (parts[0] === "\u0645\u06CC" || parts[0] === "\u0646\u0645\u06CC")) {
      const verb = parts[1];
      if (dict[verb]) {
        return dict[verb];
      }
      for (const ending of ["\u0646\u062F", "\u0645", "\u06CC", "\u062F", "\u06CC\u0645", "\u06CC\u062F"]) {
        if (verb.endsWith(ending) && verb.length > ending.length) {
          const stem = verb.slice(0, -ending.length);
          if (dict[stem]) {
            return dict[stem];
          }
        }
      }
    }
    const joined = parts.join("");
    if (dict[joined]) {
      return dict[joined];
    }
  }
  for (const suffix of [
    "\u0647\u0627\u06CC\u06CC",
    "\u0647\u0627\u06CC",
    "\u0647\u0627",
    "\u0627\u06CC",
    "\u0627\u0646",
    "\u0627\u062A",
    "\u06CC\u0646",
    "\u062A\u0631",
    "\u062A\u0631\u06CC\u0646",
    "\u0634",
    "\u0645",
    "\u062A"
  ]) {
    if (word.endsWith(suffix) && word.length > suffix.length + 1) {
      const stem = word.slice(0, -suffix.length);
      if (dict[stem]) {
        return dict[stem];
      }
    }
  }
  return void 0;
}

// src/overrides/ar.ts
var ar = {
  \u0623\u062D\u062F\u0627: "/\u0294a\u02C8\u0127adan/",
  // anyone
  \u0623\u062D\u062F\u0647\u0645\u0627: "/\u0294a\u0127adu\u02C8huma\u02D0/",
  // one of them two
  \u0623\u062D\u0631\u0627\u0631\u0627: "/\u0294a\u0127\u02C8ra\u02D0ran/",
  \u0623\u0633\u062E\u064A\u0644\u064A\u0648\u0633: "/\u0294asxi\u02D0\u02C8li\u02D0ju\u02D0s/",
  // Aeschylus
  \u0623\u0645\u0647\u0627\u062A: "/um\u02D0a\u02C8ha\u02D0t/",
  \u0623\u0646\u0633\u064A\u062A: "/\u0294un\u02C8si\u02D0t/",
  \u0623\u0646\u0639\u0645\u062A: "/\u0294an\u02C8\u0295amta/",
  \u0623\u0648\u0631\u0641\u0644\u064A\u0633: "/\u0294ur\u02C8fali\u02D0s/",
  // Orphalese (Gibran)
  \u0623\u064A\u062F\u064A\u0647\u0645: "/\u0294aj\u02C8di\u02D0him/",
  \u0627\u0633\u062A\u064A\u0642\u0638\u062A: "/istaj\u02C8qa\xF0\u02E4tu/",
  \u0627\u0633\u0645\u0647: "/\u02C8ismuhu/",
  // his name
  \u0627\u0639\u062A\u0632\u0627\u0644\u0647: "/i\u0295ti\u02C8za\u02D0lihi/",
  // his seclusion
  \u0627\u0644\u0623\u0642\u0627\u0648\u0627\u0644: "/al\u0294aqa\u02D0\u02C8wi\u02D0l/",
  // the sayings/talk
  \u0627\u0644\u0623\u0644\u064A\u0641\u0629: "/al\u0294a\u02C8li\u02D0fah/",
  \u0627\u0644\u0623\u0648\u0644\u0649: "/al\u02C8\u0294u\u02D0la\u02D0/",
  // the first (fem.)
  \u0627\u0644\u062D\u064A\u0627\u0629: "/al\u0127a\u02C8ja\u02D0h/",
  \u0627\u0644\u062E\u0644\u064A\u0641\u0629: "/alxa\u02C8li\u02D0fah/",
  \u0627\u0644\u062E\u0644\u064A\u0642\u0629: "/alxa\u02C8li\u02D0qah/",
  // creation/nature
  \u0627\u0644\u0630\u0643\u0631\u064A\u0627\u062A: "/a\xF0\u02D0ikra\u02C8ja\u02D0t/",
  \u0627\u0644\u0633\u0645\u0627\u0648\u0627\u062A: "/as\u02D0ama\u02D0\u02C8wa\u02D0t/",
  \u0627\u0644\u0636\u0627\u0644\u064A\u0646: "/ad\u02E4\u02D0a\u02D0\u02C8li\u02D0n/",
  \u0627\u0644\u0639\u0627\u0644\u0645\u064A\u0646: "/al\u0295a\u02D0la\u02C8mi\u02D0n/",
  \u0627\u0644\u0639\u0634\u0631\u0629: "/al\u02C8\u0295a\u0283ara/",
  // the ten
  \u0627\u0644\u063A\u0631\u0628\u0629: "/al\u02C8\u0263urbah/",
  \u0627\u0644\u063A\u0632\u0627\u0629: "/al\u0263u\u02C8za\u02D0h/",
  \u0627\u0644\u0643\u0631\u0627\u0645\u0629: "/alka\u02C8ra\u02D0mah/",
  \u0627\u0644\u0644\u0648\u0649: "/al\u02C8liwa\u02D0/",
  \u0627\u0644\u0645\u0635\u0637\u0641\u0649: "/almus\u02C8t\u02E4afa\u02D0/",
  \u0627\u0644\u0646\u062F\u0648\u0627\u062A: "/annada\u02C8wa\u02D0t/",
  // the seminars/forums
  \u0627\u0645\u0631\u0623\u0629: "/im\u02C8ra\u0294ah/",
  \u0627\u0647\u062F\u0646\u0627: "/ih\u02C8dina\u02D0/",
  \u0628\u0623\u0648\u0642\u0627\u0641\u0647: "/bi\u0294aw\u02C8qa\u02D0fihi/",
  // with his endowments
  \u0628\u0625\u0630\u0646\u0647: "/bi\u0294i\xF0\u02C8nihi/",
  \u0628\u0628\u0631\u0642\u0629: "/bi\u02C8barqah/",
  // in Barqa (place)
  \u0628\u062C\u0632\u0627\u0626\u0631: "/bid\u0292a\u02C8za\u02D0\u0294ir/",
  // with islands
  \u0628\u062F\u0627\u064A\u0629: "/bi\u02C8da\u02D0jah/",
  \u0628\u0630\u064A: "/bi\u02C8\xF0i\u02D0/",
  // of importance
  \u0628\u0634\u064A\u0621: "/bi\u02C8\u0283aj\u0294/",
  \u0628\u0639\u0636\u0627: "/\u02C8ba\u0295d\u02E4an/",
  \u0628\u0639\u0636\u0647\u0645: "/ba\u0295\u02C8d\u02E4uhum/",
  \u0628\u0639\u064A\u0646\u0647: "/bi\u02C8\u0295ajnihi/",
  \u0628\u0639\u064A\u0646\u064A\u0647\u0627: "/bi\u0295aj\u02C8najha\u02D0/",
  // Egyptian National Anthem — بلادي بلادي
  \u0628\u0644\u0627\u062F\u064A: "/bi\u02C8la\u02D0di\u02D0/",
  // my country
  \u0628\u064A\u062F\u0628\u0627: "/bajda\u02C8ba\u02D0/",
  // Bidpai
  \u062A\u0623\u062E\u0630\u0647: "/ta\u0294\u02C8xu\xF0uhu/",
  \u062A\u0633\u0642\u0646\u064A: "/tas\u02C8qini\u02D0/",
  \u062A\u0639\u0644\u0645\u062A: "/ta\u0295al\u02D0\u02C8amtu/",
  \u062A\u0642\u0641\u0646: "/taqif\u02C8na/",
  \u062A\u0642\u0644\u0628\u062A: "/taqal\u02D0a\u02C8bat/",
  // it turned/fluctuated
  \u062A\u0643\u0644\u0645\u064A: "/takal\u02D0a\u02C8mi\u02D0/",
  // speak! (fem.)
  \u062B\u0647\u0645\u062F: "/\u02C8\u03B8ahmad/",
  // Thahmad (place name)
  \u062C\u0644\u0633\u062A: "/d\u0292a\u02C8lasat/",
  \u062D\u0627\u0631\u062A\u0646\u0627: "/\u0127a\u02D0\u02C8ratna\u02D0/",
  \u062D\u0641\u0638\u0647\u0645\u0627: "/\u0127if\u02C8\xF0\u02E4ahuma\u02D0/",
  \u062D\u0643\u0627\u064A\u0629: "/\u0127i\u02C8ka\u02D0jah/",
  \u062D\u064A\u0627\u062A\u0647: "/\u0127a\u02C8ja\u02D0tahu/",
  \u062E\u0641\u064A\u0641\u0627: "/xa\u02C8fi\u02D0fan/",
  \u062E\u0644\u0627\u0644\u0647\u0627: "/xi\u02C8la\u02D0laha\u02D0/",
  \u062E\u0644\u0641\u0647\u0645: "/xal\u02C8fahum/",
  \u062E\u0645\u0631\u0627: "/\u02C8xamran/",
  \u062F\u0628\u0634\u0644\u064A\u0645: "/dab\u0283a\u02C8li\u02D0m/",
  // King Dabshelim
  \u0631\u0627\u0626\u062D\u0629: "/\u02C8ra\u02D0\u0294i\u0127ah/",
  \u0631\u0627\u0641\u0639\u0627: "/\u02C8ra\u02D0fi\u0295an/",
  \u0631\u0633\u0645\u0647\u0627: "/ras\u02C8muha\u02D0/",
  \u0631\u064A\u0627\u062D\u0643: "/ri\u02C8ja\u02D0\u0127aka/",
  \u0632\u0639\u0645\u0648\u0627: "/za\u02C8\u0295amu\u02D0/",
  \u0632\u0645\u0627\u0646\u0647: "/za\u02C8ma\u02D0nihi/",
  \u0633\u0627\u062F\u062A\u064A: "/sa\u02D0\u02C8dati\u02D0/",
  \u0633\u0627\u0633\u0627\u0646: "/sa\u02D0\u02C8sa\u02D0n/",
  // Sasanian dynasty
  \u0633\u0628\u0639\u0629: "/\u02C8sab\u0295ah/",
  \u0633\u062A\u062D\u0645\u0644\u0647: "/sata\u0127\u02C8miluhu/",
  \u0633\u0641\u064A\u0646\u062A\u0647: "/sa\u02C8fi\u02D0natuhu/",
  \u0633\u0643\u0631\u0629: "/\u02C8sakrah/",
  \u0633\u0646\u0629: "/\u02C8sanah/",
  \u0633\u0646\u064A\u0646: "/si\u02C8ni\u02D0n/",
  \u0634\u0647\u0631\u064A\u0627\u0631: "/\u0283ahri\u02C8ja\u02D0r/",
  // King Shahryar (1001 Nights)
  \u0634\u064A\u0626\u0627: "/\u02C8\u0283aj\u0294an/",
  \u0635\u0628\u0627\u062D\u0627: "/s\u02E4a\u02C8ba\u02D0\u0127an/",
  // morning
  \u0635\u062D\u0628\u064A: "/\u02C8s\u02E4a\u0127bi\u02D0/",
  // my companions
  \u0635\u063A\u0627\u0631\u0647\u0627: "/s\u02E4i\u02C8\u0263a\u02D0ruha\u02D0/",
  \u0637\u0627\u0644\u0639\u0627: "/\u02C8t\u02E4a\u02D0li\u0295an/",
  \u0637\u0648\u064A\u0644\u0629: "/t\u02E4a\u02C8wi\u02D0lah/",
  // Ibn Khaldun — المقدمة
  \u0638\u0627\u0647\u0631\u0647: "/\xF0\u02E4a\u02D0\u02C8hirihi/",
  // in its outward appearance
  \u0639\u0627\u0645\u0627: "/\u02C8\u0295a\u02D0man/",
  \u0639\u0628\u0644\u0629: "/\u02C8\u0295ablah/",
  // Abla (name)
  \u0639\u062C\u064A\u0628\u0629: "/\u0295a\u02C8d\u0292i\u02D0bah/",
  \u0639\u062F\u062A: "/\u02C8\u0295udtu/",
  \u0639\u0631\u0641\u062A: "/\u0295a\u02C8rafta/",
  // you knew
  \u0639\u0635\u0631\u0647: "/\u02C8\u0295as\u02E4rihi/",
  \u0639\u0644\u0645\u0647: "/\u02C8\u0295ilmihi/",
  \u0639\u0644\u064A\u0643: "/\u0295a\u02C8lajka/",
  \u0639\u0644\u064A\u0647\u0645: "/\u0295a\u02C8lajhim/",
  \u0639\u0646\u062F\u0647: "/\u02C8\u0295indahu/",
  \u0639\u0646\u0647\u0627: "/\u0295an\u02C8ha\u02D0/",
  // about it
  \u063A\u0627\u064A\u062A\u064A: "/\u0263a\u02D0\u02C8jati\u02D0/",
  // my goal
  \u063A\u064A\u0628\u0629: "/\u02C8\u0263ajbah/",
  \u0641\u0625\u0646: "/fa\u02C8\u0294in/",
  \u0641\u0627\u0633\u0642\u0646\u064A: "/fas\u02C8qini\u02D0/",
  \u0641\u0627\u063A\u062A\u0646\u0645\u0647\u0627: "/fa\u0263\u02C8tanimha\u02D0/",
  \u0641\u0627\u0644\u0645\u0642\u0631\u0627\u0629: "/fal\u02C8miqra\u02D0h/",
  \u0641\u062C\u0631\u0627: "/\u02C8fad\u0292ran/",
  \u0641\u062D\u0648\u0645\u0644: "/fa\u02C8\u0127awmal/",
  \u0641\u0645\u0627: "/fa\u02C8ma\u02D0/",
  \u0641\u064A\u0647: "/fi\u02D0hi/",
  \u0641\u064A\u0647\u0627: "/\u02C8fi\u02D0ha\u02D0/",
  \u0642\u0627\u0644\u062A: "/\u02C8qa\u02D0lat/",
  \u0642\u0635\u0629: "/\u02C8qis\u02E4\u02D0ah/",
  \u0642\u0644\u062A: "/\u02C8qultu/",
  // I said
  \u0643\u062A\u0627\u0628\u0627\u062A: "/kita\u02D0\u02C8ba\u02D0t/",
  \u0643\u0631\u0633\u064A\u0647: "/kur\u02C8sij\u02D0uhu/",
  \u0644\u062E\u0648\u0644\u0629: "/lixaw\u02C8latah/",
  // for Khawla (name)
  // Tarafa ibn al-Abd — المعلقة
  \u0644\u0643\u0650: "/\u02C8laki/",
  // to you (fem.)
  \u0644\u0646\u064A\u0644\u0643: "/li\u02C8najlika/",
  // for your Nile
  \u0644\u0648\u0647\u0644\u0629: "/li\u02C8wahlah/",
  \u0645\u0628\u0646\u064A\u0629: "/mab\u02C8nij\u02D0ah/",
  // Antara ibn Shaddad — المعلقة
  \u0645\u062A\u0631\u062F\u0645: "/muta\u02C8rad\u02D0im/",
  // worn out/in ruins
  \u0645\u062A\u0642\u0637\u0639\u0627: "/mutaqat\u02E4\u02C8t\u02E4i\u0295an/",
  \u0645\u062F\u064A\u0646\u0629: "/ma\u02C8di\u02D0nah/",
  \u0645\u0637\u064A\u0647\u0645: "/mat\u02E4aj\u02D0a\u02C8hum/",
  // their riding beasts
  // Abu al-Ala al-Ma'arri — غير مجد في ملتي
  \u0645\u0644\u062A\u064A: "/mil\u02D0a\u02C8ti\u02D0/",
  // my religion/creed
  \u0645\u0647\u064A\u0627\u0631: "/mah\u02C8ja\u02D0r/",
  // Mahyar (Adonis)
  \u0645\u064A\u0644\u0627\u062F\u0647: "/mi\u02D0\u02C8la\u02D0dihi/",
  \u0646\u0633\u062C\u062A\u0647\u0627: "/nasa\u02C8d\u0292atha\u02D0/",
  \u0646\u0648\u0645\u0627: "/\u02C8nawman/",
  \u0647\u0627\u062F\u0645\u0627: "/\u02C8ha\u02D0diman/",
  \u0647\u0648\u0630\u0627: "/ha\u02D0\u02C8\xF0a\u02D0/",
  \u0648\u0623\u0639\u0648\u0627\u0646: "/wa\u0294a\u0295\u02C8wa\u02D0n/",
  // and followers
  \u0648\u0625\u0646\u0645\u0627: "/wa\u0294in\u02D0a\u02C8ma\u02D0/",
  \u0648\u0625\u064A\u0627\u0643: "/wa\u0294ij\u02C8ja\u02D0ka/",
  \u0648\u0627\u0633\u0644\u0645\u064A: "/wasla\u02C8mi\u02D0/",
  // and farewell/be safe (fem.)
  \u0648\u0627\u0639\u062A\u0642\u0627\u062F\u064A: "/wa\u0294i\u0295tiqa\u02D0\u02C8di\u02D0/",
  // and my belief
  \u0648\u0627\u0644\u0633\u0646\u0629: "/was\u02D0\u02C8anah/",
  \u0648\u0627\u0644\u0642\u0627\u0644: "/wal\u02C8qa\u02D0l/",
  // and the talk
  \u0648\u0627\u0644\u0645\u062D\u0628\u0629: "/walma\u02C8\u0127ab\u02D0ah/",
  \u0648\u0628\u0634\u0631\u0648\u0637\u0647: "/wabi\u0283u\u02C8ru\u02D0tihi/",
  // and his conditions
  \u0648\u062D\u0643\u0627\u064A\u0627\u062A: "/wa\u0127ika\u02D0\u02C8ja\u02D0t/",
  \u0648\u0634\u0628\u064A\u0647: "/wa\u0283a\u02C8bi\u02D0h/",
  // and similar to
  \u0648\u0634\u0645\u0623\u0644: "/wa\u0283a\u02C8ma\u0294l/",
  \u0648\u0636\u0639\u0647: "/\u02C8wad\u02E4\u0295ihi/",
  \u0648\u0639\u0644\u0649: "/wa\u02C8\u0295ala\u02D0/",
  // and upon
  \u0648\u0639\u0644\u064A\u0647\u0645: "/wa\u0295a\u02C8lajhim/",
  \u0648\u063A\u0627\u0628: "/wa\u02C8\u0263a\u02D0b/",
  \u0648\u0641\u0624\u0627\u062F\u064A: "/wafu\u02C8\u0294a\u02D0di\u02D0/",
  // and my heart
  \u0648\u0641\u0637\u0646\u0629: "/wa\u02C8fit\u02E4nah/",
  \u0648\u0642\u0648\u0641\u0627: "/wu\u02C8qu\u02D0fan/",
  // standing
  \u0648\u0645\u0627: "/wa\u02C8ma\u02D0/",
  \u0648\u0645\u062A\u0633\u0627\u0648\u064A\u0646: "/wamutasa\u02D0\u02C8wi\u02D0n/",
  \u0648\u0645\u0646\u0632\u0644: "/wa\u02C8manzil/",
  \u0648\u0647\u0628\u0648\u0627: "/wu\u02C8hibu\u02D0/",
  \u064A\u0624\u0648\u062F\u0647: "/ja\u0294u\u02D0\u02C8duhu/",
  \u064A\u062A\u062E\u0637\u0649: "/jata\u02C8xat\u02E4\u02D0a\u02D0/",
  \u064A\u062D\u064A\u0637\u0648\u0646: "/ju\u0127i\u02D0\u02C8t\u02E4u\u02D0n/",
  \u064A\u0636\u0639\u0647: "/jad\u02E4a\u02C8\u0295uhu/",
  \u064A\u0639\u0627\u0645\u0644\u0648\u0627: "/ju\u02C8\u0295a\u02D0milu\u02D0/",
  \u064A\u0642\u0648\u0644\u0648\u0646: "/jaqu\u02D0\u02C8lu\u02D0n/",
  // they say
  \u064A\u0647\u062A\u0645\u0648\u0627: "/jahtam\u02C8mu\u02D0/"
  // they cared
};

// src/overrides/de.ts
var de = {
  abendsonnenschein: "/\u02C8a\u02D0b\u0259nt\u02CCz\u0254n\u0259n\u0283a\u026A\u032Fn/",
  // Foreign city names in German texts
  Aires: "/\u02C8a\u026A\u032F\u0281\u025Bs/",
  andrer: "/\u02C8and\u0281\u0250/",
  balde: "/\u02C8bald\u0259/",
  bewu\u00DFtsein: "/b\u0259\u02C8v\u028Ast\u02CCza\u026A\u032Fn/",
  bl\u00FCtenschimmer: "/\u02C8bly\u02D0t\u0259n\u02CC\u0283\u026Am\u0250/",
  brahmanensohn: "/\u02C8b\u0281a\u02D0ma\u02D0n\u0259n\u02CCzo\u02D0n/",
  Buenos: "/\u02C8bue\u02D0n\u0254s/",
  "c\u2019est": "/s\u025B/",
  // French in Mann's Buddenbrooks
  ch\u00E8re: "/\u0283\u025B\u02D0\u0281/",
  // French in Mann's Buddenbrooks
  da\u00DF: "/das/",
  "davos-platz": "/\u02C8da\u02D0v\u0254s\u02CCplats/",
  // Swiss resort area
  demoiselle: "/d\u0259mwa\u02C8z\u025Bl/",
  // French in Mann's Buddenbrooks
  d\u00FCwel: "/\u02C8dy\u02D0v\u0259l/",
  // Low German: devil
  fa\u00DFt: "/fast/",
  feuertrunken: "/\u02C8f\u0254\u026A\u032F\u0250\u02CCt\u0281\u028A\u014Bk\u0259n/",
  flu\u00DFufers: "/\u02C8fl\u028As\u02CC\u0294u\u02D0f\u0250s/",
  fr\u00FChlingsnachmittag: "/\u02C8f\u0281y\u02D0l\u026A\u014Bs\u02CCna\u02D0xm\u026Ata\u02D0k/",
  gefahrdrohende: "/\u0261\u0259\u02C8fa\u02D0\u0250\u032F\u02CCd\u0281o\u02D0\u0259nd\u0259/",
  gek\u00FC\u00DFt: "/\u0261\u0259\u02C8k\u028Fst/",
  g\u00F6tterfunken: "/\u02C8\u0261\u0153t\u0250\u02CCf\u028A\u014Bk\u0259n/",
  govinda: "/\u0261o\u02C8v\u026Anda/",
  // Sanskrit name
  graub\u00FCndischen: "/\u0261\u0281a\u028A\u032F\u02C8b\u028Fnd\u026A\u0283\u0259n/",
  guizot: "/\u0261i\u02C8zo\u02D0/",
  // French name
  h\u00E4tt: "/h\u025Bt/",
  // contraction of hätte
  "ich\u2019s": "/\u026A\xE7s/",
  k\u00FC\u00DFnacht: "/\u02C8k\u028Fsnaxt/",
  macheath: "/m\u0259\u02C8ki\u02D0\u03B8/",
  // English name
  metternich: "/\u02C8m\u025Bt\u0250n\u026A\xE7/",
  m\u00FC\u00DFt: "/m\u028Fst/",
  mu\u00DFte: "/\u02C8m\u028Ast\u0259/",
  New: "/nju\u02D0/",
  ook: "/o\u02D0k/",
  // Low German: also
  "prinz-regentenstra\xDFe": "/\u02C8p\u0281\u026Ants\u0281e\u02CC\u0261\u025Bnt\u0259n\u02CC\u0283t\u0281a\u02D0s\u0259/",
  question: "/k\u025Bs\u02C8tj\u0254\u0303/",
  // French in Mann's Buddenbrooks
  salwaldes: "/\u02C8zal\u02CCvald\u0259s/",
  // Sal forest
  samsa: "/\u02C8zamza/",
  // Kafka character
  schriee: "/\u02C8\u0283\u0281i\u02D0\u0259/",
  siddhartha: "/z\u026A\u02C8da\u02D0\u0281ta/",
  sternklar: "/\u02C8\u0283t\u025B\u0281nkla\u02D0\u0250\u032F/",
  tr\u00E8s: "/t\u0281\u025B/",
  // French in Mann's Buddenbrooks
  verliess: "/f\u025B\u0250\u032F\u02C8li\u02D0s/",
  York: "/j\u0254\u0281k/",
  zarathustra: "/tsa\u0281a\u02C8t\u028Ast\u0281a/"
};

// src/overrides/eo.ts
var eo = {
  // G2P handles all Esperanto words with correct penultimate stress.
  // Overrides here are only needed for words where the IPA dictionary
  // has an incorrect entry that needs correction.
};

// src/overrides/es.ts
var es = {
  arts: "/a\u0281/",
  // French "Arts" in Cortázar (Pont des Arts)
  aureliano: "/aw\u027Ee\u02C8ljano/",
  beatriz: "/bea\u02C8t\u027Eis/",
  buend\u00EDa: "/bwen\u02C8dia/",
  ca\u00F1abrava: "/ka\u0272a\u02C8\u03B2\u027Ea\u03B2a/",
  conti: "/\u02C8konti/",
  // Italian name in Cortázar
  fierro: "/\u02C8fje\u027Eo/",
  macondo: "/ma\u02C8kondo/",
  pont: "/p\u0254\u0303/",
  // French in Cortázar
  porfirio: "/po\u027E\u02C8fi\u027Ejo/",
  quai: "/ke/",
  // French in Cortázar
  qued\u00F3se: "/ke\u02C8\xF0ose/",
  rue: "/\u0281y/",
  // French in Cortázar
  seine: "/s\u025Bn/",
  // French river in Cortázar
  s\u00E9pase: "/\u02C8sepase/",
  urbino: "/u\u027E\u02C8\u03B2ino/",
  vacilante: "/basi\u02C8lante/",
  viterbo: "/bi\u02C8te\u027E\u03B2o/"
};

// src/overrides/fa.ts
var fa = {
  \u0622\u0628\u0650: "/\u0252\u02D0be/",
  \u0622\u0631\u0627\u06CC: "/\u0252\u02D0\u027E\u0252\u02D0j/",
  \u0622\u0641\u0631\u06CC\u0646: "/\u0252\u02D0f\xE6\u027Ei\u02D0n/",
  // creator
  \u0622\u0645\u0648\u06CC: "/\u0252\u02D0mu\u02D0j/",
  \u0622\u0648\u0627\u0631\u0647: "/\u0252\u02D0v\u0252\u02D0\u027Ee/",
  // wandering, exiled
  \u0622\u0648\u0627\u0632\u0647: "/\u0252\u02D0v\u0252\u02D0ze/",
  // fame, renown
  \u0622\u06CC\u064E\u062F: "/\u0252\u02D0j\xE6d/",
  // --- Forough Farrokhzad, Another Birth ---
  \u0622\u06CC\u0647: "/\u0252\u02D0je/",
  // verse (of scripture)
  \u0627\u0628\u062F\u06CC: "/\xE6b\xE6di\u02D0/",
  // eternal
  \u0627\u0628\u0648\u0627\u0644\u0641\u0636\u0644: "/\xE6bolf\xE6zl/",
  \u0627\u062C\u062A\u0645\u0627\u0639\u06CC: "/ed\u0361\u0292tem\u0252\u02D0\u0294i\u02D0/",
  \u0627\u062F\u0631: "/\xE6de\u027E/",
  // pour! (Arabic imperative)
  \u0627\u0633\u0627\u0633: "/\xE6s\u0252\u02D0s/",
  // --- Iranian Constitution ---
  \u0627\u0633\u0627\u0633\u06CC: "/\xE6s\u0252\u02D0si\u02D0/",
  \u0627\u0635\u0648\u0644: "/osu\u02D0l/",
  \u0627\u0641\u062A\u0627\u062F: "/oft\u0252\u02D0d/",
  // fell, befell
  // --- UDHR Article 1 ---
  \u0627\u0641\u0631\u0627\u062F: "/\xE6f\u027E\u0252\u02D0d/",
  // individuals
  \u0627\u0642\u062A\u0635\u0627\u062F\u06CC: "/e\u0262tes\u0252\u02D0di\u02D0/",
  // --- Hafez, Divan (Ghazal 1) ---
  \u0627\u0644\u0627: "/\xE6l\u0252\u02D0/",
  // O! behold
  \u0627\u0644\u0633\u0627\u0642\u06CC: "/\xE6ss\u0252\u02D0\u0262i\u02D0/",
  // the cupbearer (Arabic definite)
  \u0627\u0644\u0644\u0651\u0647: "/\xE6ll\u0252\u02D0h/",
  \u0627\u0646\u062F\u0631\u0634: "/\xE6nd\xE6\u027E\xE6\u0283/",
  // within it
  \u0627\u0646\u0632\u0648\u0627: "/enzev\u0252\u02D0/",
  // seclusion
  \u0627\u0646\u0639\u06A9\u0627\u0633: "/en\u0294ek\u0252\u02D0s/",
  // --- Sepehri ---
  \u0627\u0647\u0644: "/\xE6hl/",
  \u0627\u06CC: "/ej/",
  // O! (vocative)
  \u0627\u06CC\u0647\u0627: "/\xE6jjoh\u0252\u02D0/",
  // O! (Arabic vocative)
  \u0628\u0627\u062F\u0647\u0627\u06CC: "/b\u0252\u02D0dh\u0252\u02D0je/",
  // winds (with ezafeh)
  "\u0628\u0628\u0631\u06CC\u062F\u0647\u200C\u0627\u0646\u062F": "/bob\u027Ei\u02D0de\xE6nd/",
  // they have cut
  // --- Baba Taher ---
  \u0628\u0628\u0646\u062F\u0645: "/beb\xE6nd\xE6m/",
  \u0628\u062A\u0627\u0628\u062F: "/bet\u0252\u02D0b\xE6d/",
  // will shine
  \u0628\u062C\u0627\u0646: "/bed\u0361\u0292\u0252\u02D0n/",
  // --- Attar, Conference of the Birds ---
  \u0628\u062E\u0634\u06CC\u062F: "/b\xE6x\u0283i\u02D0d/",
  \u0628\u062F\u0627\u0631: "/bed\u0252\u02D0\u027E/",
  \u0628\u062F\u0627\u0634\u062A: "/bed\u0252\u02D0\u0283t/",
  \u0628\u0631\u062A\u0631: "/b\xE6\u027Et\xE6\u027E/",
  // higher, superior
  \u0628\u0631\u062F: "/bo\u027Ed/",
  // will carry (subjunctive)
  \u0628\u0633\u062A\u0646\u062F: "/b\xE6st\xE6nd/",
  \u0628\u0634\u0627\u06CC\u062F: "/be\u0283\u0252\u02D0j\xE6d/",
  \u0628\u0634\u0631: "/b\xE6\u0283\xE6\u027E/",
  // humankind
  // --- Rumi, Masnavi (Song of the Reed) ---
  \u0628\u0634\u0646\u0648: "/be\u0283no/",
  // listen!
  \u0628\u0634\u0648\u06CC\u0645: "/be\u0283u\u02D0j\xE6m/",
  \u0628\u06AF\u0631\u062F\u0645: "/be\u0261\xE6\u027Ed\xE6m/",
  \u0628\u06AF\u0634\u0627\u06CC: "/be\u0261o\u0283\u0252\u02D0j/",
  \u0628\u06AF\u0648\u06CC\u0645: "/be\u0261u\u02D0j\xE6m/",
  // let me say
  \u0628\u0646\u0627\u0632\u0645: "/ben\u0252\u02D0z\xE6m/",
  // --- Saadi, Bustan ---
  \u0628\u0646\u0627\u0645: "/ben\u0252\u02D0m/",
  // in the name of
  \u0628\u0646\u0634\u0633\u062A\u0647: "/bene\u0283\xE6ste/",
  // has sat
  \u0628\u0646\u0645\u0627\u06CC: "/ben\xE6m\u0252\u02D0j/",
  // --- Rudaki ---
  \u0628\u0648\u06CC\u0650: "/bu\u02D0je/",
  \u0628\u06CC\u0647\u0642\u06CC: "/bejh\xE6\u0262i\u02D0/",
  \u067E\u064E\u0631\u0646\u06CC\u0627\u0646: "/p\xE6\u027Eni\u02D0\u0252\u02D0n/",
  // --- Sanai ---
  \u067E\u0631\u0648\u0631: "/p\xE6\u027Ev\xE6\u027E/",
  \u067E\u0633\u062A\u06CC: "/p\xE6sti\u02D0/",
  "\u067E\u0648\u0632\u0634\u200C\u067E\u0630\u06CC\u0631": "/pu\u02D0ze\u0283p\xE6zi\u02D0\u027E/",
  // excuse-accepting
  \u062A\u0627\u0631\u06CC\u06A9\u06CC\u0633\u062A: "/t\u0252\u02D0\u027Ei\u02D0ki\u02D0st/",
  // is a darkness (contraction)
  \u062A\u062D\u0642\u06CC\u0642: "/t\xE6h\u0262i\u02D0\u0262/",
  \u062A\u0648\u0641\u06CC\u0642: "/to\u028Afi\u02D0\u0262/",
  "\u062C\u062F\u0627\u06CC\u06CC\u200C\u0647\u0627": "/d\u0361\u0292od\u0252\u02D0ji\u02D0h\u0252\u02D0/",
  // separations
  \u062C\u0644: "/d\u0361\u0292\xE6ll/",
  // majesty
  \u062C\u0648\u0627\u0628\u0650: "/d\u0361\u0292\xE6v\u0252\u02D0be/",
  \u062C\u0648\u06CC\u0650: "/d\u0361\u0292u\u02D0je/",
  \u062D\u0627\u0641\u0638: "/h\u0252\u02D0fez/",
  \u062D\u0627\u0644: "/h\u0252\u02D0l/",
  // state, condition
  \u062D\u0627\u0644\u06CC: "/h\u0252\u02D0li\u02D0/",
  // pleasant (archaic: now)
  \u062D\u0642\u0648\u0642: "/ho\u0262u\u02D0\u0262/",
  // rights
  \u062D\u06A9\u06CC\u0645: "/h\xE6ki\u02D0m/",
  // wise
  \u062D\u06CC\u0627\u062A: "/h\xE6j\u0252\u02D0t/",
  // life
  \u062D\u06CC\u062B\u06CC\u062A: "/hejsi\u02D0j\xE6t/",
  // dignity
  // --- Nezami, Khosrow o Shirin ---
  \u062E\u062F\u0627\u0648\u0646\u062F\u0627: "/xod\u0252\u02D0v\xE6nd\u0252\u02D0/",
  // --- Saadi, Golestan ---
  \u062E\u062F\u0627\u06CC: "/xod\u0252\u02D0j/",
  // God (with ezafeh)
  \u062E\u0631\u062F\u0628\u062E\u0634: "/xe\u027E\xE6db\xE6x\u0283/",
  \u062E\u0637\u0627\u0628\u062E\u0634: "/x\xE6t\u0252\u02D0b\xE6x\u0283/",
  // sin-forgiving
  \u062E\u0650\u0646\u06AF\u0650: "/xen\u0261e/",
  \u062E\u0648\u0634\u0627: "/xo\u0283\u0252\u02D0/",
  \u062E\u0648\u0634\u062E\u0648\u0627\u0646: "/xo\u0283x\u0252\u02D0n/",
  // sweet-singing
  // --- Behbahani ---
  \u062E\u0648\u06CC\u0634: "/xi\u02D0\u0283/",
  \u062E\u06CC\u0632\u0631\u0627\u0646: "/xi\u02D0z\u027E\u0252\u02D0n/",
  // bamboo
  \u062F\u0627\u0631\u0627\u06CC: "/d\u0252\u02D0\u027E\u0252\u02D0je/",
  // possessing
  \u062F\u0627\u0646\u0646\u062F: "/d\u0252\u02D0n\xE6nd/",
  // they know
  \u062F\u0631\u0650: "/d\xE6\u027Ee/",
  \u062F\u064F\u0631\u0634\u062A\u06CC\u200C\u0647\u0627\u06CC: "/do\u027Eo\u0283ti\u02D0h\u0252\u02D0je/",
  \u062F\u0631\u06CC\u0686\u0647\u200C\u06CC: "/d\xE6\u027Ei\u02D0t\u0361\u0283eje/",
  \u062F\u0633\u062A\u06AF\u06CC\u0631: "/d\xE6st\u0261i\u02D0\u027E/",
  // helper
  \u062F\u0644\u06CC: "/deli/",
  \u062F\u0648\u0633\u062A\u0627\u0646\u06CC: "/du\u02D0st\u0252\u02D0ni\u02D0/",
  \u062F\u06CC\u062F\u0627\u0631: "/di\u02D0d\u0252\u02D0\u027E/",
  \u0630\u0648\u0642\u06CC: "/zo\u028A\u0262i\u02D0/",
  \u0631\u0627: "/\u027E\u0252\u02D0/",
  // object marker
  \u0631\u0627\u0632\u0642: "/\u027E\u0252\u02D0ze\u0262/",
  "\u0631\u0633\u062A\u0646\u200C\u0647\u0627\u06CC": "/\u027E\xE6st\xE6nh\u0252\u02D0je/",
  // growings (with ezafeh)
  \u0631\u0636\u06CC: "/\u027E\xE6zi\u02D0/",
  \u0631\u0641\u062A\u0627\u0631: "/\u027E\xE6ft\u0252\u02D0\u027E/",
  // behavior
  \u0631\u0647: "/\u027E\xE6h/",
  \u0631\u0647\u0650: "/\u027E\xE6he/",
  \u0631\u0647\u0646\u0645\u0627\u06CC: "/\u027E\xE6hn\xE6m\u0252\u02D0j/",
  // guide
  \u0631\u0648\u062D\u06CC\u0647: "/\u027Eu\u02D0hi\u02D0je/",
  // spirit, morale
  \u0631\u0648\u0632\u06AF\u0627\u0631\u0645: "/\u027Eu\u02D0ze\u0261\u0252\u02D0\u027E\xE6m/",
  "\u0631\u0648\u0632\u06CC\u200C\u062F\u0647": "/\u027Eu\u02D0zi\u02D0deh/",
  // sustenance-giver
  \u0631\u06CC\u06AF\u0650: "/\u027Ei\u02D0\u0261e/",
  \u0632\u0627\u06CC\u06CC\u062F\u0647: "/z\u0252\u02D0ji\u02D0de/",
  // born
  \u0632\u0628\u0627\u0646\u06CC: "/z\xE6b\u0252\u02D0ni\u02D0/",
  // --- Hedayat, The Blind Owl ---
  "\u0632\u062E\u0645\u200C\u0647\u0627\u06CC\u06CC": "/z\xE6xmh\u0252\u02D0ji\u02D0/",
  // wounds (indefinite)
  \u0632\u06CC: "/zi\u02D0/",
  \u0632\u06CC\u0652: "/zi\u02D0/",
  "\u0633\u0627\u062D\u0644\u200C\u0647\u0627": "/s\u0252\u02D0helh\u0252\u02D0/",
  // shores
  \u0633\u0628\u06A9\u0628\u0627\u0631\u0627\u0646: "/s\xE6bokb\u0252\u02D0\u027E\u0252\u02D0n/",
  // light-laden ones
  \u0633\u0628\u06A9\u062A\u06AF\u06CC\u0646: "/sobokte\u0261i\u02D0n/",
  \u0633\u062D\u0631\u06AF\u0627\u0647: "/s\xE6h\xE6\u027E\u0261\u0252\u02D0h/",
  // dawn
  // --- Nezami, Layli o Majnun ---
  \u0633\u0631\u0622\u063A\u0627\u0632: "/s\xE6r\u0252\u02D0\u0261\u0252\u02D0z/",
  \u0633\u0631\u0627\u06CC\u062F: "/so\u027E\u0252\u02D0j\xE6d/",
  \u0633\u0644\u0627\u0645\u06CC: "/s\xE6l\u0252\u02D0mi\u02D0/",
  // --- Khayyam, Rubaiyat ---
  \u0633\u0648\u062F\u0627: "/so\u028Ad\u0252\u02D0/",
  // passion, melancholy
  \u0633\u06CC\u0627\u0633\u06CC: "/sij\u0252\u02D0si\u02D0/",
  \u0634\u0627\u062F\u0645\u0627\u0646: "/\u0283\u0252\u02D0dm\u0252\u02D0n/",
  \u0634\u0631\u062D: "/\u0283\xE6\u027Eh/",
  // explanation
  \u0634\u0631\u062D\u0647: "/\u0283\xE6\u027Ehe/",
  // explanation (with ezafeh)
  \u0634\u06A9\u0627\u06CC\u062A: "/\u0283ek\u0252\u02D0j\xE6t/",
  // complaint
  "\u0634\u06A9\u0641\u062A\u0646\u200C\u0647\u0627": "/\u0283ekoft\xE6nh\u0252\u02D0/",
  // blossomings
  \u0635\u0644\u062A\u0650: "/sele\u02D0te/",
  \u0635\u0646\u0639: "/son\u0294/",
  \u0636\u0648\u0627\u0628\u0637: "/z\xE6v\u0252\u02D0bet/",
  \u0637\u0627\u0639\u062A\u0634: "/t\u0252\u02D0\u0294\xE6t\xE6\u0283/",
  // his obedience
  \u0639\u0632: "/\u0294\xE6zz/",
  // glory
  \u0639\u0645\u0631: "/om\u027E/",
  \u0639\u0646\u0647: "/\xE6nho/",
  \u063A\u0627\u0632\u06CC: "/\u0263\u0252\u02D0zi\u02D0/",
  \u063A\u0627\u06CC\u062A: "/\u0263\u0252\u02D0j\xE6t/",
  \u063A\u0632\u0644: "/\u0263\xE6z\xE6l/",
  \u063A\u0632\u0646\u0648\u06CC: "/\u0263\xE6zn\xE6vi\u02D0/",
  \u063A\u0632\u0646\u06CC: "/\u0263\xE6zni\u02D0/",
  \u0641\u0631\u0627\u0642: "/fe\u027E\u0252\u02D0\u0262/",
  // separation
  \u0641\u0631\u062F: "/f\xE6\u027Ed/",
  // alone, single
  \u0641\u0631\u0647\u0646\u06AF\u06CC: "/f\xE6rh\xE6n\u0261i\u02D0/",
  \u0641\u0631\u0648: "/fo\u027Eu\u02D0/",
  // down
  \u0641\u0644\u06A9: "/f\xE6l\xE6k/",
  \u0642\u062F\u06A9: "/\u0262\xE6d\xE6k/",
  \u0642\u0631\u0628\u062A: "/\u0262o\u027Eb\xE6t/",
  // nearness (to God)
  \u0642\u0635\u06CC\u062F\u0647\u200C\u0627\u06CC: "/\u0262\xE6si\u02D0de\xE6j/",
  // --- Nima Yushij, The Phoenix ---
  \u0642\u0642\u0646\u0648\u0633: "/\u0262o\u0262nu\u02D0s/",
  // phoenix
  \u06A9\u0622\u0641\u0631\u06CC\u0646\u062A: "/k\u0252\u02D0f\xE6\u027Ei\u02D0n\xE6t/",
  \u06A9\u0627\u0631\u06AF\u0634\u0627\u06CC: "/k\u0252\u02D0\u027E\u0261o\u0283\u0252\u02D0j/",
  \u06A9\u0627\u0633\u0627: "/k\xE6\u0294s\u0252\u02D0/",
  // a cup (Arabic accusative, Persianized)
  \u06A9\u0627\u0634\u0627\u0646\u0645: "/k\u0252\u02D0\u0283\u0252\u02D0n\xE6m/",
  \u06A9\u0627\u0648: "/k\u0252\u02D0w/",
  \u06A9\u0631\u06CC\u0645: "/k\xE6\u027Ei\u02D0m/",
  // generous
  // --- Ferdowsi, Shahnameh ---
  \u06A9\u0632: "/k\xE6z/",
  // contraction of که از (that from)
  \u06A9\u0634\u06CC\u062F\u0645: "/ke\u0283i\u02D0d\xE6m/",
  // I drew (a sigh)
  \u06A9\u0644\u0627\u0645: "/k\xE6l\u0252\u02D0m/",
  \u06A9\u0646\u0627\u0646: "/kon\u0252\u02D0n/",
  // doing (present participle suffix)
  \u06A9\u0646\u0645: "/kon\xE6m/",
  \u06A9\u0646\u0646\u062F: "/kon\xE6nd/",
  // they do (subjunctive)
  \u06AF\u0631\u062F\u0627\u0628\u06CC: "/\u0261e\u027Ed\u0252\u02D0bi\u02D0/",
  // a whirlpool (indefinite)
  // --- Akhavan-Sales ---
  \u06AF\u0641\u062A: "/\u0261oft/",
  \u06AF\u064F\u0644: "/\u0261ol/",
  \u06AF\u0648\u06CC\u062F: "/\u0261u\u02D0j\xE6d/",
  // --- Beyhaqi ---
  \u06AF\u0648\u06CC\u0646\u062F\u0647\u0654: "/\u0261u\u02D0j\xE6ndeje/",
  \u0644\u063A\u0632\u0627\u0646: "/l\xE6\u0263z\u0252\u02D0n/",
  \u0645\u0627\u062F\u0631\u06CC: "/m\u0252\u02D0d\xE6\u027Ei\u02D0/",
  \u0645\u0627\u0646\u062F\u0647: "/m\u0252\u02D0nde/",
  // remaining
  \u0645\u0627\u0647\u062A\u0627\u0628: "/m\u0252\u02D0ht\u0252\u02D0b/",
  // moonlight
  \u0645\u0628\u06CC\u0646: "/mob\xE6jjen/",
  \u0645\u062D\u0645\u0648\u062F: "/m\xE6hmu\u02D0d/",
  \u0645\u062F\u0647: "/m\xE6deh/",
  \u0645\u0632\u06CC\u062F: "/m\xE6zi\u02D0d/",
  // increase
  \u0645\u0641\u0631\u062D: "/mof\xE6\u027E\u027Eeh/",
  // gladdening
  \u0645\u0645\u062F: "/momedd/",
  // sustainer
  \u0645\u0648\u062C\u0628: "/mo\u028Ad\u0361\u0292eb/",
  // cause of
  \u0645\u0648\u0644\u06CC\u0627\u0646: "/mu\u02D0li\u02D0\u0252\u02D0n/",
  \u0645\u0648\u0646\u0633: "/mu\u02D0nes/",
  \u0645\u0648\u0647\u0628\u062A: "/mo\u028Aheb\xE6t/",
  // gift, endowment
  "\u0645\u06CC\u200C\u0622\u06CC\u062F": "/mi\u02D0\u0252\u02D0j\xE6d/",
  // comes (up)
  "\u0645\u06CC\u200C\u062A\u0631\u0627\u0634\u062F": "/mi\u02D0t\xE6\u027E\u0252\u02D0\u0283\xE6d/",
  // scrapes, carves
  "\u0645\u06CC\u200C\u062E\u0648\u0631\u062F": "/mi\u02D0xo\u027E\xE6d/",
  // eats, erodes
  "\u0645\u06CC\u200C\u0631\u0648\u062F": "/mi\u02D0\u027E\xE6v\xE6d/",
  // goes (down)
  "\u0645\u06CC\u200C\u0634\u0648\u0646\u062F": "/mi\u02D0\u0283\xE6v\xE6nd/",
  // they become (passive)
  "\u0645\u06CC\u200C\u06A9\u0646\u062F": "/mi\u02D0kon\xE6d/",
  // does, is doing
  \u0646\u0627\u067E\u0633\u0646\u062F\u200C\u0645: "/n\u0252\u02D0p\xE6s\xE6nd\xE6m/",
  \u0646\u0627\u062E\u0648\u0628: "/n\u0252\u02D0xu\u02D0b/",
  \u0646\u0627\u0635\u0631: "/n\u0252\u02D0se\u027E/",
  "\u0646\u0627\u0644\u06CC\u062F\u0647\u200C\u0627\u0646\u062F": "/n\u0252\u02D0li\u02D0de\xE6nd/",
  // they have lamented
  \u0646\u0627\u0646\u06CC: "/n\u0252\u02D0ni\u02D0/",
  \u0646\u0627\u0648\u0644\u0647\u0627: "/n\u0252\u02D0velh\u0252\u02D0/",
  // pass it around
  \u0646\u062A\u0648\u0627\u0646\u062F: "/n\xE6t\xE6v\u0252\u02D0n\xE6d/",
  \u0646\u0634\u0627\u0637\u0650: "/ne\u0283\u0252\u02D0te/",
  \u0646\u0638\u0627\u0645\u06CC: "/nez\u0252\u02D0mi\u02D0/",
  \u0646\u0638\u0631\u0628\u0627\u0632\u06CC\u0627: "/n\xE6z\xE6\u027Eb\u0252\u02D0zi\u02D0\u0252\u02D0/",
  \u0646\u0639\u0645\u062A: "/ne\u0294m\xE6t/",
  // blessing
  \u0646\u0641\u0633\u06CC: "/n\xE6f\xE6si\u02D0/",
  // a breath (indefinite)
  \u0646\u0641\u06CC\u0631\u0645: "/n\xE6fi\u02D0\u027E\xE6m/",
  // my wailing
  \u0646\u06AF\u0627\u0647\u0650: "/ne\u0261\u0252\u02D0he/",
  \u0646\u06AF\u0630\u0631\u062F: "/n\xE6\u0261oz\xE6\u027E\xE6d/",
  // does not pass
  \u0646\u06AF\u0647: "/ne\u0261\xE6h/",
  \u0646\u0645\u0648\u062F: "/nemu\u02D0d/",
  // appeared, showed
  "\u0646\u0645\u06CC\u200C\u0634\u0648\u062F": "/n\xE6mi\u02D0\u0283\xE6v\xE6d/",
  // is not possible
  // --- Shared classical forms ---
  \u0646\u0647\u0627\u062F: "/neh\u0252\u02D0d/",
  \u0646\u0647\u0627\u062F\u0647\u0627\u06CC: "/n\xE6h\u0252\u02D0dh\u0252\u02D0je/",
  \u0646\u0648\u0634: "/nu\u02D0\u0283/",
  // drink!
  \u0646\u06CC\u0627\u0628\u062F: "/n\xE6j\u0252\u02D0b\xE6d/",
  // will not find
  \u0646\u06CC\u0627\u0631\u062F: "/n\xE6j\u0252\u02D0\u027E\xE6d/",
  \u0646\u06CC\u0633\u062A\u0627\u0646: "/nejest\u0252\u02D0n/",
  // reed bed
  // --- Shamlou ---
  \u0646\u06CC\u0633\u062A\u06CC: "/ni\u02D0sti\u02D0/",
  \u0647\u0627\u06CC\u0644: "/h\u0252\u02D0jel/",
  // terrifying
  \u0647\u0633\u062A\u0646\u062F: "/h\xE6st\xE6nd/",
  // they are
  \u0647\u064E\u0645\u06CC: "/h\xE6mi\u02D0/",
  \u0647\u0648\u0634\u06CC: "/hu\u02D0\u0283i\u02D0/",
  \u0648\u062C\u062F\u0627\u0646: "/ved\u0361\u0292d\u0252\u02D0n/",
  // conscience
  \u0648\u062F\u06CC\u0639\u062A: "/v\xE6di\u02D0\xE6t/",
  \u0648\u0632\u0634: "/v\xE6ze\u0283/",
  // blowing
  \u0648\u06CC: "/v\xE6j/",
  \u06CC\u0627\u062F\u0650: "/j\u0252\u02D0de/",
  \u06CC\u0627\u0631\u0650: "/j\u0252\u02D0\u027Ee/",
  \u06CC\u0627\u0641\u062A: "/j\u0252\u02D0ft/",
  \u06CC\u06A9\u062F\u06CC\u06AF\u0631: "/jekdi\u02D0\u0261\xE6\u027E/"
  // one another
};

// src/overrides/fi.ts
var fi = {
  // G2P handles all Finnish words correctly. Overrides here are only
  // needed for words where G2P produces an incorrect entry.
};

// src/overrides/fr.ts
var fr = {
  conflans: "/k\u0254\u0303fl\u0251\u0303/",
  // place name
  est: "/\u025B/",
  // verb "is" — st is silent (dict has /ɛst/)
  jolies: "/\u0292\u0254li/",
  luit: "/l\u0265i/",
  marchiennes: "/ma\u0281\u0283j\u025Bn/",
  // place name (Zola)
  "m\xE9taphysico-th\xE9ologo-cosmolonigologie": "/metafizikoteol\u0254\u0261\u0254k\u0254sm\u0254l\u0254ni\u0261\u0254l\u0254\u0292i/",
  // Voltaire
  montsou: "/m\u0254\u0303su/",
  // fictional town (Zola)
  morgion: "/m\u0254\u0281\u0292j\u0254\u0303/",
  // place name (Dumas)
  myriel: "/mi\u0281j\u025Bl/",
  // character name (Hugo)
  "neuve-sainte-genevi\xE8ve": "/n\u0153vs\u025B\u0303t\u0292\u0259nvj\u025Bv/",
  // Paris street (Balzac)
  nicole: "/nik\u0254l/",
  pangloss: "/p\u0251\u0303\u0261l\u0254s/",
  // Voltaire character
  rainur\u00E9e: "/\u0281\u025Bny\u0281e/",
  rion: "/\u0281j\u0254\u0303/",
  // place name (Dumas)
  "saint-marcel": "/s\u025B\u0303ma\u0281s\u025Bl/",
  // Paris quarter (Dumas)
  "thunder-ten-tronckh": "/t\u0254n\u025B\u0281t\u025Bnt\u0281\u0254nk/",
  // Voltaire
  trieste: "/t\u0281ij\u025Bst/",
  // city name (Dumas)
  vauquer: "/voke/",
  // character name (Balzac)
  y: "/i/"
  // pronoun "there" — dict has letter name /igʁɛk/
};

// src/overrides/is.ts
var is = {
  \u00E1sir: "/\u02C8au\u02D0s\u026Ar/",
  \u00E1synjur: "/\u02C8au\u02D0s\u026Anj\u028Fr/",
  ballir: "/\u02C8patl\u026Ar/",
  beytils: "/\u02C8pei\u02D0t\u02B0\u026Als/",
  Bj\u00E1lfa: "/\u02C8pjaulva/",
  Bjarnar: "/\u02C8pjartnar/",
  bl\u00F3t: "/plou\u02D0t\u02B0/",
  // sacrifice/ritual
  b\u00F3klaus: "/\u02C8pouk\u02B0l\u0153ys/",
  br\u00E9fberi: "/\u02C8prj\u025B\u02D0vp\u025Br\u026A/",
  // letter carrier
  burluf\u00F3ts: "/\u02C8p\u028Frtl\u028Ffouts/",
  efla\u00F0i: "/\u02C8\u025Bpla\xF0\u026A/",
  // past tense: strengthened/performed
  ek: "/\u025Bk\u02B0/",
  fannhv\u00EDtir: "/\u02C8fan\u02D0kvit\u02B0\u026Ar/",
  farandi: "/\u02C8farant\u026A/",
  fars\u00E6lda: "/\u02C8farsailta/",
  fira: "/\u02C8f\u026Ara/",
  flatnefur: "/\u02C8flatn\u025Bv\u028Fr/",
  fl\u00FD\u00F0u: "/\u02C8fli\u02D0\xF0\u028F/",
  fornaldar: "/\u02C8f\u0254rtnaltar/",
  fornrit: "/\u02C8f\u0254rtnr\u026At\u02B0/",
  galdursmenn: "/\u02C8kalt\u028Frsm\u025Bn\u02D0/",
  gautr: "/\u02C8k\u0153y\u02D0tr/",
  // New saga/literary overrides
  gengr: "/k\u025B\u014Bkr/",
  go\u00F0or\u00F0sma\u00F0ur: "/\u02C8k\u0254\u02D0\xF0\u0254r\xF0sma\u02D0\xF0\u028Fr/",
  h\u00E6ngs: "/\u02C8hai\u014Bs/",
  hags\u00E6lda: "/\u02C8haksailta/",
  H\u00E1lfdanarsonar: "/\u02C8haulf\u02CCtanar\u02CCs\u0254nar/",
  h\u00E1lftr\u00F6lls: "/\u02C8haulftr\u0153tls/",
  Hallbjarnar: "/\u02C8hatlpjartnar/",
  Hallfre\u00F0s: "/\u02C8hatlfr\u025B\xF0s/",
  // genitive of Hallfreður
  Hallfre\u00F0ur: "/\u02C8hatlfr\u025B\xF0\u028Fr/",
  // proper name
  h\u00E1rfagra: "/\u02C8haur\u02CCfa\u0263ra/",
  // fair-haired (epithet)
  h\u00E1rfagri: "/\u02C8haur\u02CCfa\u0263r\u026A/",
  hersir: "/\u02C8h\u025Brs\u026Ar/",
  hersis: "/\u02C8h\u025Brs\u026As/",
  herskarar: "/\u02C8h\u025Brsk\u02B0arar/",
  Hrafnkelsdal: "/\u02C8r\u0325apnk\u02B0\u025Blstalr/",
  // place name
  Hreggvi\u00F0sson: "/\u02C8r\u0325\u025Bk\u02D0v\u026A\xF0s\u02D0\u0254n/",
  hr\u00EDmhv\u00EDta: "/\u02C8r\u0325imkvit\u02B0a/",
  Ing\u00F3lfr: "/\u02C8\u026A\u014Bkoulfr/",
  ins: "/\u026Ans/",
  \u00CDvarssonar: "/\u02C8ivars\u02D0\u0254nar/",
  j\u00F6klanna: "/\u02C8j\u0153k\u02B0lan\u02D0a/",
  J\u00F3rsalalands: "/\u02C8jourtsalalants/",
  Ketils: "/\u02C8k\u02B0\u025Bt\u02B0\u026Als/",
  kindir: "/\u02C8k\u02B0\u026Ant\u026Ar/",
  K\u00F3lumkilli: "/\u02C8k\u02B0ou\u02D0l\u028Fmk\u02B0\u026Atl\u026A/",
  konungd\u00E6mi: "/\u02C8k\u02B0\u0254\u02D0n\u028F\u014B\u02CCtai\u02D0m\u026A/",
  konungd\u00F3mur: "/\u02C8k\u02B0\u0254\u02D0n\u028F\u014B\u02CCdou\u02D0m\u028Fr/",
  konungr: "/\u02C8k\u02B0\u0254\u02D0n\u028F\u014Bkr/",
  konungseigninni: "/\u02C8k\u02B0\u0254\u02D0n\u028F\u014Bkseikn\u026An\u02D0\u026A/",
  kotb\u00F3ndi: "/\u02C8k\u02B0\u0254t\u02B0pount\u026A/",
  kunnigt: "/\u02C8k\u02B0\u028Fn\u02D0\u026Ak\u02B0t\u02B0/",
  kvonga\u00F0ur: "/\u02C8k\u02B0v\u0254\u014Bka\xF0\u028Fr/",
  kynst\u00F3r: "/\u02C8k\u02B0\u026Anstou\u02D0r/",
  ma\u00F0r: "/ma\u02D0\xF0r/",
  magra: "/\u02C8ma\u0263ra/",
  mannd\u00E1\u00F0in: "/\u02C8man\u02D0tau\u02D0\xF0\u026An/",
  mannf\u00F3lkit: "/\u02C8man\u02D0fouk\u02B0\u026At\u02B0/",
  mj\u00F6k: "/mj\u0153k\u02B0/",
  m\u00F6gu: "/\u02C8m\u0153\u02D0k\u028F/",
  n\u00F3rr\u00E6nn: "/\u02C8nou\u02D0rrai\u02D0n\u02D0/",
  N\u00F6rvasundum: "/\u02C8n\u0153rvas\u028Fnt\u028Fm/",
  Norvegi: "/\u02C8n\u0254rv\u025Bj\u026A/",
  n\u00FDu: "/\u02C8ni\u02D0\u028F/",
  \u00F3arga: "/\u02C8ou\u02D0arka/",
  \u00D6nundur: "/\u02C8\u0153\u02D0n\u028Fnt\u028Fr/",
  or: "/\u0254r/",
  \u00F3r: "/ou\u02D0r/",
  \u00F6xn: "/\u02C8\u0153ksn/",
  pl\u00F3gsland: "/\u02C8p\u02B0lou\u02D0kslant/",
  r\u00E1\u00F0u: "/\u02C8rau\u02D0\xF0\u028F/",
  r\u00E6kir: "/\u02C8rai\u02D0k\u02B0\u026Ar/",
  Raumar\u00EDki: "/\u02C8r\u0153y\u02D0mar\u026A\u02D0k\u02B0\u026A/",
  Raumsd\u00E6lafylki: "/\u02C8r\u0153ymstailaf\u026Alk\u02B0\u026A/",
  Raumsdal: "/\u02C8r\u0153ymstal/",
  raumur: "/\u02C8r\u0153y\u02D0m\u028Fr/",
  Rein: "/rei\u02D0n/",
  sannliga: "/\u02C8san\u02D0l\u026A\u0263a/",
  Sk\u00ED\u00F0ason: "/\u02C8ski\u02D0\xF0as\u0254n/",
  s\u00F6\u00F0ul: "/\u02C8s\u0153\u02D0\xF0\u028Fl/",
  s\u00F3lkerfum: "/\u02C8soulk\u02B0\u025Brv\u028Fm/",
  tindar: "/\u02C8t\u02B0\u026Antar/",
  \u00FAtsj\u00E1num: "/\u02C8u\u02D0t\u02B0sjau\u02D0n\u028Fm/",
  v\u00E1gskorin: "/\u02C8vau\u0263sk\u0254r\u026An/",
  Valf\u00F6\u00F0r: "/\u02C8valv\u0153\xF0r/",
  v\u00EDkverskur: "/\u02C8vi\u02D0kv\u025Brsk\u028Fr/",
  Yngveldi: "/\u02C8\u026A\u014Bkv\u025Blt\u026A/",
  \u00FEat: "/\u03B8at\u02B0/"
};

// src/overrides/ja.ts
var ja = {
  \u3042\u3051\u307C\u306E: "/akebono/",
  \u3042\u3063\u305F: "/at\u02D0a/",
  \u3042\u3068: "/ato/",
  \u3042\u307E\u305F: "/amata/",
  \u3042\u3089\u306C: "/a\u027Ean\u026F/",
  \u3042\u308A: "/a\u027Ei/",
  \u3042\u308B: "/a\u027E\u026F/",
  \u3042\u308B\u304F: "/a\u027E\u026Fk\u026F/",
  \u3044\u3046: "/i\u026F/",
  \u30A4\u30AE\u30EA\u30B9: "/i\u0261i\u027Eis\u026F/",
  // England
  \u3044\u305F: "/ita/",
  \u3044\u305F\u3046: "/ita\u026F/",
  // archaic: greatly
  // New sample overrides
  \u3044\u3064\u3082: "/its\u026Fmo/",
  \u3044\u3065\u308C: "/id\u026F\u027Ee/",
  // archaic: which
  \u3044\u308B: "/i\u027E\u026F/",
  \u3046\u307F: "/\u026Fmi/",
  \u3046\u3089: "/\u026F\u027Ea/",
  \u304A\u308B: "/o\u027E\u026F/",
  \u304B\u305F: "/kata/",
  \u304B\u305F\u3061: "/katat\u0255i/",
  \u304B\u3064\u3050: "/kats\u026F\u0261\u026F/",
  \u304B\u306E: "/kano/",
  \u304F: "/k\u026F/",
  // archaic verb stem
  \u3054: "/\u0261o/",
  \u3053\u3053: "/koko/",
  \u3054\u3056\u3044\u307E\u3059: "/\u0261ozaimas\u026F/",
  \u3053\u306E: "/kono/",
  \u3055\u3056\u308C: "/saza\u027Ee/",
  \u3055\u3073\u3057\u3044: "/sabi\u0255i\u02D0/",
  \u3055\u3089: "/sa\u027Ea/",
  // archaic: furthermore
  \u3057\u3088\u3046: "/\u0255ijo\u02D0/",
  \u3059\u3050\u308C\u308B: "/s\u026F\u0261\u026F\u027Ee\u027E\u026F/",
  \u3059\u3053\u3057: "/s\u026Fko\u0255i/",
  \u3059\u308B: "/s\u026F\u027E\u026F/",
  \u305D\u3053: "/soko/",
  \u305D\u306E: "/sono/",
  \u3060\u3044\u3076: "/daib\u026F/",
  \u3060\u3051: "/dake/",
  \u305F\u3060: "/tada/",
  \u3060\u3064: "/dats\u026F/",
  // archaic suffix: -ish
  \u305F\u306A\u3073\u304F: "/tanabik\u026F/",
  \u3064\u304F: "/ts\u026Fk\u026F/",
  \u3064\u308C\u308B: "/ts\u026F\u027Ee\u027E\u026F/",
  \u3069\u3053: "/doko/",
  \u3068\u3053\u308D: "/toko\u027Eo/",
  \u3068\u3089\u3048\u308B: "/to\u027Eae\u027E\u026F/",
  \u306A\u304A: "/nao/",
  \u306A\u304B: "/naka/",
  \u306A\u304F: "/nak\u026F/",
  \u306A\u3063\u305F: "/nat\u02D0a/",
  \u306A\u3073\u304F: "/nabik\u026F/",
  \u306E\u3051: "/noke/",
  \u3070\u304B\u308A: "/baka\u027Ei/",
  \u306F\u305F: "/hata/",
  \u30CF\u30F3\u30D6\u30EB\u30AF: "/hamb\u026F\u027E\u026Fk\u026F/",
  \u3075\u3046: "/\u0278\u026F\u02D0/",
  \u3075\u3061: "/\u0278\u026Ft\u0255i/",
  \u307B\u304B: "/hoka/",
  \u307B\u3069: "/hodo/",
  \u307B\u3093\u3068\u3046: "/honto\u02D0/",
  \u307E\u305F: "/mata/",
  \u307E\u3060: "/mada/",
  \u307E\u3067: "/made/",
  \u307F\u306A\u3055\u3093: "/minasa\u0274/",
  \u307F\u3093\u306A: "/min\u02D0a/",
  \u3080\u3059: "/m\u026Fs\u026F/",
  \u30E1\u30ED\u30B9: "/me\u027Eos\u026F/",
  \u3084\u3046\u3084\u3046: "/ja\u026Fja\u026F/",
  // archaic: gradually
  \u3084\u307F: "/jami/",
  \u3084\u3080: "/jam\u026F/",
  \u3084\u3093\u3054\u3068\u306A\u3057: "/ja\u014B\u0261otona\u0255i/",
  // archaic: noble
  \u3088\u304F: "/jok\u026F/",
  \u3088\u307B\u3069: "/johodo/",
  \u308B: "/\u027E\u026F/",
  // classical auxiliary
  \u308F\u304B\u308B: "/waka\u027E\u026F/",
  \u4E00\u8336: "/it\u02D0\u0255a/",
  // Issa (poet)
  \u4E09\u5341\u4E03: "/sa\u0274d\u0291\u026F\u02D0\u0255it\u0255i/",
  \u5019\u3046: "/so\u02D0\u027Eo\u02D0/",
  // archaic polite auxiliary
  \u547C\u3093\u3067: "/jo\u0274de/",
  \u6B62\u307E\u3063\u305F: "/tomat\u02D0a/",
  \u75E9: "/jase/",
  // thin
  \u767D\u304F: "/\u0255i\u027Eok\u026F/",
  \u77E5\u3063\u3066: "/\u0255it\u02D0e/",
  \u7A4D\u307F: "/ts\u026Fmi/",
  \u7F85\u751F\u9580: "/\u027Ea\u0255o\u02D0mo\u0274/",
  // Rashomon
  \u821E\u9DB4: "/maiz\u026F\u027E\u026F/",
  // Maizuru (place)
  \u91D1\u95A3: "/ki\u0274kak\u026F/",
  // Golden Pavilion
  \u9759\u304B\u3055: "/\u0255iz\u026Fkasa/"
};

// src/overrides/km.ts
var km = {
  // --- UDHR Article 1 ---
  \u1780\u17C6\u178E\u17BE\u178F: "/k\u0251mna\u0259t/",
  // birth
  // --- Reamker (Ramayana) ---
  \u1780\u17D2\u178A\u17B8: "/kd\u0259y/",
  // matter, affair
  \u1780\u17D2\u179A\u17C1\u179C: "/kreew/",
  // furious
  \u1780\u17D2\u179A\u17C1\u179C\u1780\u17D2\u179A\u17C4\u1792: "/kreew krout/",
  // furiously angry
  \u1780\u17D2\u179A\u17C4\u1792: "/krout/",
  // anger
  \u1780\u17D2\u179F: "/ks/",
  // consonant cluster (browser splits ក្សត្រី)
  \u1780\u17D2\u179F\u178F\u17D2\u179A: "/ksaat/",
  // king
  // --- Nokor Reach (National Anthem) ---
  \u1780\u17D2\u179F\u178F\u17D2\u179A\u17B6: "/ksaatraa/",
  // king (literary form)
  \u1780\u17D2\u179F\u178F\u17D2\u179A\u17B8: "/ksaatr\u0259y/",
  // queen
  \u1781\u17D2\u1798\u17B8: "/km\u0259y/",
  // ogre (literary)
  \u1781\u17D2\u179C\u17BE\u1780: "/kwa\u0259k/",
  // to stir, disturb
  \u1782\u17B6\u1794\u17CB: "/koap/",
  // beloved
  // --- Tum Teav ---
  \u1782\u17BC: "/kuu/",
  // partner
  \u1782\u17BC\u1782\u17B6\u1794\u17CB: "/kuu koap/",
  // partner, beloved
  \u1783\u17D2\u179B\u17B6\u178F: "/kliet/",
  // separated
  \u1785\u179A: "/c\u0251\u0251/",
  // to walk, go
  \u1785\u17D2\u1793\u17C1\u17C7: "/cneh/",
  // this (literary)
  \u1787\u17D0\u1799: "/cey/",
  // victory
  \u1787\u17D0\u1799\u1798\u1784\u17D2\u1782\u179B: "/cey m\u0254\u014Bk\u0254l/",
  // victory, auspicious
  // --- Proverbs ---
  \u1787\u17B6\u1780\u17CB: "/ceak/",
  // certain, sure
  \u1787\u17B6\u179B\u17C6\u178A\u17B6\u1794\u17CB: "/cie l\u0254m\u0257aap/",
  // successively
  // --- Preah Chinawong ---
  \u1787\u17B7\u1793: "/c\u0268n/",
  // Jin (proper name element)
  \u1787\u17B7\u1793\u179C\u1784\u17D2\u0E2A: "/c\u0268n w\u0254\u014B/",
  // Chinawong (proper name)
  \u1787\u17BC: "/cuu/",
  // proper (literary)
  \u1789\u17B6\u1780\u17CB: "/\u0272eak/",
  // to startle, twitch
  \u1789\u17B6\u1780\u17CB\u1785\u17B7\u1789\u17D2\u1785\u17BE\u1798: "/\u0272eak c\u0259\u0272ca\u0259m/",
  // to raise eyebrows
  \u178A\u17C2\u1793: "/daen/",
  // territory
  \u178E\u17B6\u1799: "/naay/",
  // to yearn
  \u178E\u17B6\u1799\u1785\u17B7\u178F\u17D2\u178F: "/naay c\u0259t/",
  // heart yearns
  \u178F\u1794: "/t\u0251p/",
  // to reply
  \u1790\u17D2\u1780\u17B6\u1793: "/tkaan/",
  // magnificent (literary)
  \u1790\u17D2\u1780\u17BE\u1784: "/tk\u0259\u0259\u014B/",
  // to glorify
  \u1790\u17D2\u1780\u17BE\u1784\u1790\u17D2\u1780\u17B6\u1793: "/tk\u0259\u0259\u014B tkaan/",
  // glorious, magnificent
  \u1790\u17D2\u1793\u17BC\u179A: "/tnoo/",
  // dignity, nobility
  \u1790\u17D2\u1793\u17C2: "/tnae/",
  // aspect (partial segmentation of ថ្នែក)
  \u1790\u17D2\u1793\u17C2\u1780: "/tnaek/",
  // aspect, class
  \u1790\u17D2\u179C\u17B6\u178F\u17CB: "/twat/",
  // harshly
  \u1791\u178F: "/t\u0254t/",
  // to look, behold (royal)
  \u1791\u17B6\u179C: "/tiew/",
  // Teav (proper name)
  \u1791\u17BB\u1780\u17D2\u1781: "/tuk/",
  // suffering (dukkha)
  \u1791\u17BB\u1780\u17D2\u1781\u179C\u17C1\u1791\u1793\u17B6: "/tukweet\u0251\u0251nie/",
  // suffering
  \u1791\u17BC\u179B: "/tuul/",
  // to inform (royal register)
  \u1791\u17C1\u1796\u17D2\u178F\u17B6: "/teep\u0257aa/",
  // devas, celestial beings
  \u1793\u179A\u1794\u178F\u17B8: "/n\u0254r\u0254pa\u0257\u0259y/",
  // sovereign, king
  \u1793\u17B6\u179C: "/niew/",
  // Nav (proper name)
  \u1793\u17B7\u1798\u17CC\u179B: "/n\u0268mm\u0254l/",
  // pure, immaculate
  \u1794\u17C6\u1795\u17D2\u179B\u17B6\u1789: "/b\u0251mp\u02B0lie\u0272/",
  // to destroy
  \u1794\u17C6\u1795\u17D2\u179B\u17B7\u1785: "/b\u0251mp\u02B0l\u0259c/",
  // to demolish
  \u1794\u1796\u17B7\u178F\u17D2\u179A: "/b\u0251p\u0268t/",
  // lord, sir (polite address)
  \u1794\u17BB\u178F\u17D2\u179A: "/\u0253ot/",
  // son
  // --- Chbab Srey (Code for Women) ---
  \u1794\u17D2\u179A\u178A\u17C5: "/pr\u0251\u0257aw/",
  // to advise, counsel
  \u1794\u17D2\u179A\u178E\u17B8: "/pr\u0251n\u0259y/",
  // loving, affectionate
  \u1794\u17D2\u179A\u17B6\u1787\u17D2\u1789: "/praac/",
  // wisdom
  \u1794\u17D2\u179A\u17B6\u179F\u17B6\u1791: "/praasaat/",
  // temple, palace
  \u1796\u17B7\u179A\u17C4\u1792: "/piroot/",
  // anger
  \u1796\u17BB\u17C6: "/pum/",
  // not (literary)
  \u1796\u17BB\u17C6\u1787\u17BC: "/pum cuu/",
  // not proper
  \u1796\u17D2\u1799\u17B6\u1794\u17B6\u1791: "/pjiebaat/",
  // malice
  \u1796\u17D2\u179A\u17C7\u1798\u17A0\u17C1\u179F\u17B8: "/preah m\u0254hees\u0259y/",
  // queen consort (with ព្រះ)
  \u1796\u17D2\u179A\u17C7\u179A\u17B6\u1787\u1794\u17BB\u178F\u17D2\u179A: "/preah riec \u0253ot/",
  // prince (with ព្រះ)
  \u1796\u17D2\u179A\u17C7\u179A\u17B6\u1787\u17B6: "/preah riecie/",
  // the king
  \u1797\u17D0\u1780\u17D2\u179A\u17D2\u178F: "/p\u02B0eak/",
  // face (royal register)
  \u1797\u17B6\u178F\u179A: "/p\u02B0iet\u0251\u0251/",
  // brother (standalone; final រ silent)
  \u1797\u17B6\u178F\u179A\u1797\u17B6\u1796: "/p\u02B0iet\u0251\u0251r\u0251p\u02B0iep/",
  // brotherhood
  \u1797\u17D2\u1789\u17B6: "/p\u0272ie/",
  // to awaken
  \u1798\u1784\u17D2\u1782\u179B: "/m\u0254\u014Bk\u0254l/",
  // auspicious
  \u1798\u17A0\u17B6: "/m\u0254haa/",
  // great (prefix)
  \u1798\u17A0\u17B6\u1780\u17D2\u179F\u178F\u17D2\u179A: "/m\u0254haa ksaat/",
  // great king, maharaja
  \u1798\u17A0\u17B6\u1787\u17B6\u178F\u17B7: "/m\u0254haa ciet/",
  // great nation
  \u1798\u17A0\u17C1\u179F\u17B8: "/m\u0254hees\u0259y/",
  // queen consort
  \u179A\u17B6\u1787: "/riec/",
  // royal
  \u179A\u17B6\u1787\u1794\u17BB\u178F\u17D2\u179A: "/riec \u0253ot/",
  // prince
  \u179A\u17B6\u1787\u17B6: "/riecie/",
  // king
  \u179A\u17BB\u1784: "/ru\u014B/",
  // to shine
  \u179A\u17BB\u1784\u179A\u17BF\u1784: "/ru\u014B r\u0268\u0259\u014B/",
  // brilliant, prosperous
  \u179B\u17C6\u178A\u17B6\u1794\u17CB: "/l\u0254m\u0257aap/",
  // successively, in order
  \u179C\u1784\u17D2\u179F: "/w\u0254\u014B/",
  // dynasty, lineage
  \u179C\u17B7\u1785\u17B6\u179A\u178E\u1789\u17D2\u1789\u17B6\u178E: "/wicaar\u0251\u0272\u0272aan/",
  // discernment
  \u179C\u17C1\u1791\u1793\u17B6: "/weet\u0251\u0251nie/",
  // pain (vedana)
  \u179F\u178F\u17B7: "/sa\u0294te\u0294/",
  // consciousness, mindfulness
  \u179F\u178F\u17B7\u179F\u1798\u17D2\u1794\u1787\u1789\u17D2\u1789\u17C8: "/sa\u0294te\u0294 sampa\u0294c\u028A\u0259\u0272\u0272ea\u0294/",
  // conscience
  \u179F\u1798\u17D2\u1794\u1787\u1789\u17D2\u1789\u17C8: "/sampa\u0294c\u028A\u0259\u0272\u0272ea\u0294/",
  // awareness
  \u179F\u17B6\u1791\u179A: "/saat\u0254\u0254/",
  // to welcome
  \u179F\u17B7\u179A\u17B8: "/ser\u0259y/",
  // glory (from Pali)
  \u179F\u17BD: "/su\u0259/",
  // (first syllable of សួស្តី)
  \u179F\u17BD\u179F\u17D2\u178F\u17B8: "/su\u0259sd\u0259y/",
  // greeting, well-being
  \u179F\u17C1\u1785\u1780\u17D2\u178A\u17B8: "/sac kd\u0259y/",
  // matter, affair
  \u179F\u17C1\u1785\u1780\u17D2\u178A\u17B8\u1790\u17D2\u179B\u17C3\u1790\u17D2\u1793\u17BC\u179A: "/sac kd\u0259y tlay tnoo/",
  // dignity
  \u179F\u17D2\u178A\u17C1\u1785: "/sdac/",
  // king
  \u179F\u17D2\u178F\u17B6\u1794\u17CB: "/sdaap/",
  // to listen
  \u179F\u17D2\u178F\u17B8: "/sd\u0259y/",
  // (second syllable of សួស្តី)
  \u179F\u17D2\u1791\u17BB\u17C7: "/stuh/",
  // to rush
  \u179F\u17D2\u1793\u17C6: "/sn\u0251m/",
  // concubine
  \u17A0\u178F\u17D2\u1790\u17B6: "/hat\u0251\u0251t\u02B0aa/",
  // hand (literary)
  // --- Constitution Preamble ---
  \u17A2\u1784\u17D2\u1782\u179A: "/\u0294\u0251\u014Bk\u0254\u0254/",
  // Angkor
  \u17A2\u179A: "/\u0294\u0251\u0251/",
  // to rejoice
  \u17A2\u179F\u17D2\u1785\u17B6\u179A\u17D2\u1799: "/\u0294\u0251scaa/",
  // wonderful, marvelous
  \u17A2\u17B6\u179B: "/\u0294aal/",
  // to rush, boast
  \u17A2\u17BD\u178F: "/\u0294u\u0259t/",
  // to boast
  \u17B1\u17D2\u1799: "/\u0294aoy/"
  // to give, let
};

// src/overrides/ko.ts
var ko = {
  \uAC10\uB098\uBB34: "/kam.na.mu/",
  \uAC1C\uCC9C: "/k\u025B.t\u0255\u02B0\u028Cn/",
  \uACBD\uC131: "/kj\u028C\u014B.s\u028C\u014B/",
  // old name for Seoul
  // New sample overrides
  \uACE0\uB2EC\uD504\uB2E4: "/ko.dal.p\u02B0\u026F.da/",
  \uACE0\uC774: "/ko.i/",
  \uAD11\uC74C: "/kwa\u014B.\u026Fm/",
  \uAE08\uBE5B: "/k\u026Fm.pit\u031A/",
  \uAE30\uB098\uAE30\uB2E4: "/ki.na.\u0261i.da/",
  // very long
  \uAE38\uB3D9: "/kil.do\u014B/",
  // Hong Gildong (character name)
  \uAF79\uACFC\uB9AC: "/k\u0348w\u025B\u014B.gwa.\u027Ei/",
  // kkwaenggwari (percussion instrument)
  \uB04A\uC784: "/k\u0348\u026Fn.im/",
  \uB098\uD0C0\uC0E4: "/na.t\u02B0a.\u0255a/",
  // Natasha
  \uB17C\uAC00: "/non.\u0261a/",
  \uB2C8\uAE4C: "/ni.k\u0348a/",
  // because (suffix)
  \uB3C4\uC6B0\uB2E4: "/to.u.da/",
  // to help
  \uB3D9\uC9D3\uB2EC: "/to\u014B.d\u0291it\u031A.t\u0348al/",
  \uB9E4\uC5B4\uB2EC\uB9AC\uB2E4: "/m\u025B.\u028C.dal.li.da/",
  \uBB3C\uB4E4\uB2E4: "/mul.d\u026Fl.da/",
  \uBC31\uB450\uC0B0: "/p\u025Bk\u031A.t\u0348u.san/",
  // Mt. Baekdu
  \uBC94\uD558\uB2E4: "/p\u028Cm.ha.da/",
  \uC0B0\uBAA8\uD241\uC774: "/san.mo.t\u02B0u\u014B.i/",
  \uC18C\uC90F\uC9D1: "/so.d\u0291ut\u031A.t\u0348\u0255ip\u031A/",
  \uC544\uB77C\uB9AC\uC694: "/a.\u027Ea.\u027Ei.jo/",
  // arirang refrain
  \uC544\uC2DC\uB2E4: "/a.\u0255i.da/",
  // to know (honorific)
  \uC57D\uC0B0: "/jak\u031A.s\u0348an/",
  // Yaksan (place)
  \uC5B4\uB450: "/\u028C.du/",
  \uC5B4\uB860: "/\u028C.\u027Eon/",
  // archaic: elder
  \uC5ED\uACB9\uB2E4: "/j\u028Ck\u031A.kj\u028Cp\u031A.t\u0348a/",
  \uC601\uBCC0: "/j\u028C\u014B.bj\u028Cn/",
  // Yeongbyeon (place)
  \uC624\uB3D9\uB098\uBB34: "/o.do\u014B.na.mu/",
  \uC624\uB3D9\uC78E: "/o.do\u014B.ip\u031A/",
  \uC624\uC2DC\uB2E4: "/o.\u0255i.da/",
  // honorific: to come
  \uC654\uB2E4: "/wat\u031A.t\u0348a/",
  \uC6B0\uB7EC\uB974\uB2E4: "/u.\u027E\u028C.\u027E\u026F.da/",
  \uC6D0\uD1B5\uD558\uB2E4: "/w\u028Cn.t\u02B0o\u014B.ha.da/",
  \uC774\uB77C: "/i.\u027Ea/",
  // copula ending
  \uC774\uC5B4\uB4E0: "/i.\u028C.d\u026Fn/",
  // archaic conditional
  \uC774\uC5D0: "/i.e/",
  \uC78A\uD788\uB2E4: "/i.t\u02B0i.da/",
  \uC78E\uC0C8: "/ip\u031A.s\u0348\u025B/",
  \uC7A5\uB0A0: "/t\u0255a\u014B.nal/",
  \uC7A5\uD130: "/t\u0255a\u014B.t\u02B0\u028C/",
  \uC810\uC21C\uC774: "/t\u0255\u028Cm.su.ni/",
  // character name (diminutive)
  \uC9C0\uB9AC\uB2E4: "/t\u0255i.\u027Ei.da/",
  \uC9C0\uC904\uB300\uB2E4: "/t\u0255i.d\u0291ul.d\u025B.da/",
  \uCC44\uC2DD: "/t\u0255\u02B0\u025B.\u0255ik\u031A/",
  \uCCAD\uCC9C: "/t\u0255\u02B0\u028C\u014B.t\u0255\u02B0\u028Cn/",
  \uCD5C: "/t\u0255\u02B0we/",
  // surname Choi
  \uCD98\uD48D: "/t\u0255\u02B0un.p\u02B0u\u014B/",
  \uD0C0\uC791\uB9C8\uB2F9: "/t\u02B0a.d\u0291ak\u031A.ma.da\u014B/",
  // threshing ground
  \uD2F0\uB04C: "/t\u02B0i.k\u0348\u026Fl/",
  \uD53C\uC5B4\uC624\uB974\uB2E4: "/p\u02B0i.\u028C.o.\u027E\u026F.da/",
  \uD55C\uD14C: "/han.t\u02B0e/",
  \uD574\uC124\uD53C: "/h\u025B.s\u028Cl.p\u02B0i/",
  // poetic: at sunset
  \uD5E4\uB2E4: "/he.da/",
  // archaic: to count
  \uD718\uB2EC\uB9AC\uB2E4: "/hwi.dal.li.da/",
  \uD718\uB3CC\uB2E4: "/hwi.dol.da/"
};

// src/overrides/ma.ts
var ma = {
  Abdul: "abdul",
  bebas: "bebas",
  boleh: "boleh",
  gemawan: "\u0261\u0259mawan",
  Hamid: "hamid",
  ibni: "ibni",
  Jebat: "d\u0292\u0259bat",
  Johor: "d\u0292oho\u027E",
  kebebasan: "k\u0259bebasan",
  Lekir: "l\u0259ki\u027E",
  Lekiu: "l\u0259kiw",
  Malaysia: "malejsia",
  merdeka: "m\u0259\u027Edeka",
  mereka: "m\u0259reka",
  Oleh: "oleh",
  samarata: "samarata",
  Sarawak: "sa\u027Eawak",
  seekor: "s\u0259eko\u027E",
  seksaan: "seksa\u0294an",
  Selangor: "s\u0259la\u014Bo\u027E",
  selendang: "s\u0259l\u0259nda\u014B",
  Singapura: "si\u014B\u0261apu\u027Ea",
  Terengganu: "t\u0259\u027Ee\u014B\u0261anu",
  Tuhan: "tuhan"
};

// src/overrides/nb.ts
var nb = {
  // Old orthography (Riksmål/Danish) and common words
  al: "\u0251\u02D0l",
  \u00E5rsalderen: "o\u02D0\u027Es\u0251l\u02D0d\u0259\u027E\u0259n",
  arv: "\u0251\u027Ev",
  behold: "b\u0259h\u0254l",
  Bj\u00F8rgulfson: "bj\xF8\u02D0\u027E\u0261\u0289lfs\u0254n",
  Blik: "bl\u026Ak",
  Bliv: "bli\u02D0",
  b\u00F8r: "b\xF8\u02D0\u027E",
  B\u00F8rnene: "b\xF8\u02D0\u0273\u0259n\u0259",
  brorskapets: "b\u027Eu\u02D0\u027Esk\u0251\u02D0p\u0259ts",
  demokratiet: "d\u025Bmok\u027E\u0251ti\u02D0\u0259",
  dig: "d\u0251j",
  Dyb: "dy\u02D0p",
  Eftermiddag: "\u025Bft\u025B\u027Em\u026Ad\u0251\u02D0\u0261",
  ej: "\u0251j",
  fandens: "f\u0251n\u02D0\u0259ns",
  f\u00E5tt: "f\u0254t",
  fiender: "fi\u02D0\u025Bnd\u0259\u027E",
  Fjeldbygden: "fj\u025Blb\u028F\u0261d\u0259n",
  fordrukken: "f\u0254\u027Ed\u027E\u0289k\u02D0\u0259n",
  fornuft: "f\u0254\u027En\u0289ft",
  frem: "f\u027E\u025Bm",
  Fremmedkarl: "f\u027E\u025Bm\u02D0\u0259k\u0251\u027El",
  frostblaa: "f\u027E\u0254stblo\u02D0",
  frygt: "f\u027E\u028Fkt",
  gamlingen: "\u0261\u0251ml\u026A\u014B\u0259n",
  Gem: "j\u025Bm",
  gikk: "j\u026Ak",
  Gjesling: "j\u025Bsl\u026A\u014B",
  gjorde: "ju\u02D0\u027E\u0259",
  Glands: "\u0261l\u0251ns",
  gnistred: "\u0261n\u026Ast\u027E\u0259d",
  gr\u00E5spr\u00E6ngt: "\u0261\u027Eo\u02D0sp\u027E\u025B\u014Bt",
  Grunnlov: "\u0261\u027E\u0289n\u02D0lo\u02D0v",
  Guld: "\u0261\u0289l",
  Gyldenlak: "j\u028Fl\u02D0\u0259nl\u0251k",
  Gyldentop: "j\u028Fl\u02D0\u0259nt\u0254p",
  ham: "h\u0251m",
  Hej: "h\u0251j",
  h\u00F8r: "h\xF8\u02D0\u027E",
  Horisonten: "h\u0254\u027E\u026As\u0254nt\u0259n",
  humanistiske: "h\u0289m\u0251n\u026Ast\u026Ask\u0259",
  husbond: "h\u0289\u02D0sb\u0254n",
  Hvad: "v\u0251",
  hvem: "v\u025Bm",
  hverandre: "v\u025B\u027E\u0251nd\u027E\u0259",
  hverken: "v\u025B\u027Ek\u0259n",
  Hvor: "vu\u02D0\u027E",
  hvoraf: "vu\u02D0\u027E\u0251v",
  Hvorfor: "v\u0254\u027Ef\u0254\u027E",
  Hvormeget: "vu\u02D0\u027Em\u0251j\u02D0\u0259",
  iaften: "i\u0251ft\u0259n",
  Idet: "i\u02D0de\u02D0t",
  ihob: "iho\u02D0b",
  imellem: "im\u025Bl\u02D0\u0259m",
  imod: "i\u02D0mu\u02D0t",
  inn: "\u026An",
  intet: "\u026Ant\u0259",
  Intet: "\u026Ant\u0259",
  Ivar: "i\u02D0v\u0251\u027E",
  Jammer: "j\u0251m\u025B\u027E",
  Jensen: "j\u025Bns\u0259n",
  jordegods: "ju\u02D0\u027E\u0259\u0261\u0254ts",
  juletr\u00E6et: "j\u0289\u02D0l\u0259t\u027E\u025B\u02D0\u0259",
  just: "j\u0289st",
  kan: "k\u0251n",
  kanske: "k\u0251n\u0283\u0259",
  Kj\u00F8ttmeisene: "\xE7\xF8tm\xE6\u026As\u0259n\u0259",
  kold: "k\u0254l",
  kom: "k\u0254m",
  Kongeriket: "k\u0254\u014B\u0259\u027Ei\u02D0k\u0259",
  Kringsatt: "k\u027E\u026A\u014Bs\u0251t",
  Kristiania: "k\u027E\u026Ast\u026A\u0251\u02D0n\u026A\u0251",
  kristne: "k\u027E\u026Astn\u0259",
  Landets: "l\u0251n\u0259ts",
  Lavrans: "l\u0251\u02D0v\u027E\u0251ns",
  lektor: "l\u025Bkt\u0254\u027E",
  lig: "li\u02D0\u0261",
  m\u00E5: "mo\u02D0",
  maa: "mo\u02D0",
  Mandemagt: "m\u0251n\u02D0\u0259m\u0251kt",
  menneskerettighetene: "m\u025Bn\u02D0\u0259sk\u0259\u027E\u025Bt\u02D0\u026A\u0261he\u02D0t\u0259n\u0259",
  menneskerettigheter: "m\u025Bn\u02D0\u0259sk\u0259\u027E\u025Bt\u02D0\u026A\u0261he\u02D0t\u0259\u027E",
  menneskeverd: "m\u025Bn\u02D0\u0259sk\u0259ve\u02D0\u027Ed",
  mig: "m\u0251j",
  mod: "mo\u02D0d",
  monarkisk: "m\u0254n\u0251\u027Ek\u026Ask",
  Muld: "m\u0289l",
  Nej: "n\u0251j",
  noget: "no\u02D0\u0259",
  n\u00F8gne: "n\xF8jn\u0259",
  Norge: "n\u0254\u027E\u0261\u0259",
  nu: "n\u0289\u02D0",
  nysn\u00F8en: "ny\u02D0sn\xF8\u02D0\u0259n",
  \u00F8jne: "\u0254jn\u0259",
  op: "\u0254p",
  Peer: "pe\u02D0\u027E",
  Plads: "pl\u0251s",
  proppenerer: "p\u027E\u0254p\u0259ne\u02D0\u027E\u0259\u027E",
  Puslinger: "p\u0289\u02D0sl\u026A\u014B\u0259\u027E",
  raaber: "\u027Eo\u02D0b\u0259\u027E",
  R\u00E6kke: "\u027E\u025Bk\u0259",
  r\u00E6kker: "\u027E\u025Bk\u02D0\u0259\u027E",
  Ragnfrid: "\u027E\u0251\u014Bnf\u027Ei\u02D0d",
  regjeringsform: "\u027Eeje\u02D0\u027E\u026A\u014Bsf\u0254\u027Em",
  rettsstaten: "\u027E\u025Bt\u02D0s\u02D0t\u0251\u02D0t\u0259n",
  Rosentinter: "\u027Eu\u02D0s\u0259nt\u026Ant\u025B\u027E",
  rummeligt: "\u027E\u0289m\u02D0\u0259l\u026A\u0261t",
  Ryg: "\u027E\u028F\u0261",
  saganatt: "s\u0251\u02D0\u0261\u0251n\u0251t",
  samvittighet: "s\u0251mv\u026At\u02D0\u026A\u0261he\u02D0t",
  sidste: "s\u026Ast\u0259",
  siger: "si\u02D0\u0259\u027E",
  skabt: "sk\u0251pt",
  skal: "sk\u0251l",
  Skar: "sk\u0251\u02D0\u027E",
  skj\u00E6ms: "\u0283\u025Bms",
  Skodden: "sk\u0254d\u02D0\u0259n",
  Skydsskiftet: "\u0283\u028Ftssk\u026Aft\u0259",
  slig: "sli\u02D0\u0261",
  smaa: "smo\u02D0",
  Snees: "sne\u02D0s",
  Sneskavler: "sne\u02D0sk\u0251\u02D0vl\u025B\u027E",
  S\u00F8rby: "s\xF8\u02D0\u027Eby\u02D0",
  sp\u00F8r: "sp\xF8\u02D0\u027E",
  steget: "ste\u02D0\u0261\u0259",
  stundom: "st\u0289nd\u0254m",
  stygt: "st\u028Fkt",
  Sundbu: "s\u0289nb\u0289\u02D0",
  Syd: "sy\u02D0d",
  tabt: "t\u0251pt",
  Tant: "t\u0251nt",
  Terje: "t\u025B\u027Ej\u0259",
  T\u00F8v: "t\xF8\u02D0v",
  Tvi: "tvi\u02D0",
  tykkner: "t\u028Fk\u02D0n\u0259\u027E",
  uavhendelig: "\u0289\u02D0\u0251vh\u025Bnd\u0259l\u026A\u0261",
  udekket: "\u0289\u02D0d\u025Bk\u02D0\u0259",
  udelelig: "\u0289\u02D0de\u02D0l\u0259\u0261",
  uden: "\u0289\u02D0d\u0259n",
  underjordisk: "\u0289n\u02D0\u0259\u027Eju\u02D0\u027Ed\u026Ask",
  v\u00E6rbitt: "v\xE6\u02D0\u027Eb\u026At",
  vejr: "ve\u02D0\u027E",
  Verdigrunnlaget: "ve\u02D0\u027Ed\u026A\u0261\u027E\u0289n\u02D0l\u0251\u02D0\u0261\u0259",
  Vigen: "vi\u02D0\u0261\u0259n",
  vil: "v\u026Al",
  Vindvet: "v\u026Andv\u0259",
  visst: "v\u026Ast",
  Vorherres: "vo\u02D0\u027Eh\u025B\u027E\u0259s",
  yderste: "y\u02D0d\u0259\u027Est\u0259"
};

// src/overrides/nl.ts
var nl = {
  aandelen: "/\u02C8a\u02D0nd\u0259l\u0259n/",
  achterhaalt: "/\u02C8\u0251xt\u0259rha\u02D0lt/",
  ademde: "/\u02C8a\u02D0d\u0259md\u0259/",
  // past tense: breathed
  alchemie: "/\u0251lx\u0259\u02C8mi/",
  allen: "/\u02C8\u0251l\u0259n/",
  alsoo: "/\u0251l\u02C8so\u02D0/",
  // archaic: thus
  Anton: "/\u02C8\u0251nt\u0254n/",
  // proper name
  balkons: "/b\u0251l\u02C8k\u0254ns/",
  bange: "/\u02C8b\u0251\u014B\u0259/",
  benaeuwde: "/b\u0259\u02C8na\u02D0ud\u0259/",
  // archaic: oppressed
  // New sample overrides
  benoemd: "/b\u0259\u02C8nu\u02D0mt/",
  bevonden: "/b\u0259\u02C8v\u0254nd\u0259n/",
  bleeke: "/\u02C8ble\u02D0k\u0259/",
  // archaic: pale
  bloedroze: "/\u02C8blud\u02CCro\u02D0z\u0259/",
  boomen: "/\u02C8bo\u02D0m\u0259n/",
  // archaic: bomen (trees)
  braamstoelen: "/\u02C8bra\u02D0m\u02CCstu\u02D0l\u0259n/",
  // blackberry bushes
  bureautje: "/by\u02C8ro\u02D0tj\u0259/",
  burgery: "/b\u028Fr\u0263\u0259\u02C8r\u025Bi/",
  // archaic: burgerij
  dagelix: "/\u02C8da\u02D0\u0263\u0259l\u026Aks/",
  // archaic: dagelijks
  dagschemer: "/\u02C8d\u0251\u0263\u02CCsxe\u02D0m\u0259r/",
  dengenen: "/d\u025Bn\u02C8\u0263e\u02D0n\u0259n/",
  // archaic: those
  dese: "/\u02C8de\u02D0z\u0259/",
  // archaic: deze
  deselve: "/d\u0259\u02C8z\u025Blv\u0259/",
  // archaic: dezelfde
  dien: "/di\u02D0n/",
  // archaic dative: that
  draaide: "/\u02C8dra\u02D0id\u0259/",
  duitsen: "/\u02C8d\u0153yts\u0259n/",
  // archaic: German
  eenvouds: "/\u02C8e\u02D0nv\u0251uts/",
  // archaic: simplicity
  eert: "/e\u02D0rt/",
  egters: "/\u02C8\u025B\u0263t\u0259rs/",
  // proper name
  engelen: "/\u02C8\u025B\u014B\u0259l\u0259n/",
  erbarremt: "/\u025Br\u02C8b\u0251r\u0259mt/",
  // archaic: have mercy
  erembodegem: "/\u02C8e\u02D0r\u0259m\u02CCbo\u02D0d\u025B\u0263\u0259m/",
  erkers: "/\u02C8\u025Brk\u0259rs/",
  flauwe: "/\u02C8fl\u0251u\u0259/",
  frits: "/fr\u026Ats/",
  gekend: "/\u0263\u0259\u02C8k\u025Bnt/",
  // past participle: known
  gekomen: "/\u0263\u0259\u02C8ko\u02D0m\u0259n/",
  gekund: "/\u0263\u0259\u02C8k\u028Fnt/",
  gemeenschappelijks: "/\u0263\u0259\u02C8me\u02D0nsx\u0251p\u0259l\u0259ks/",
  // archaic genitive
  geschrey: "/\u0263\u0259\u02C8sxr\u025Bi/",
  // archaic: outcry
  gestelt: "/\u0263\u0259\u02C8st\u025Blt/",
  // archaic: placed
  geverfd: "/\u0263\u0259\u02C8v\u025Brft/",
  // painted, dyed
  gewone: "/\u0263\u0259\u02C8wo\u02D0n\u0259/",
  ghewelt: "/\u0263\u0259\u02C8w\u025Blt/",
  // archaic: violence
  godt: "/\u0263\u0254t/",
  // archaic: God
  goedkope: "/\u02C8\u0263utk\u02B0o\u02D0p\u0259/",
  graaft: "/\u0263ra\u02D0ft/",
  groeiden: "/\u02C8\u0263ru\u02D0id\u0259n/",
  grooten: "/\u02C8\u0263ro\u02D0t\u0259n/",
  // archaic: great
  Haarlem: "/\u02C8ha\u02D0rl\u0259m/",
  // city name
  had: "/h\u0251t/",
  hadden: "/\u02C8h\u0251d\u0259n/",
  hare: "/\u02C8ha\u02D0r\u0259/",
  // archaic: her
  hele: "/\u02C8he\u02D0l\u0259/",
  hemelsche: "/\u02C8he\u02D0m\u0259lsx\u0259/",
  // archaic: heavenly
  henri: "/\u0251\u0303\u02C8ri/",
  // French name
  herinner: "/h\u025B\u02C8r\u026An\u0259r/",
  hispanje: "/h\u026As\u02C8p\u0251\u0272\u0259/",
  // archaic: Spain
  hooft: "/ho\u02D0ft/",
  // archaic: hoofd
  hooren: "/\u02C8ho\u02D0r\u0259n/",
  // archaic: horen
  immense: "/\u026A\u02C8m\u025Bns\u0259/",
  inni: "/\u02C8\u026Ani/",
  // proper name (Mulisch)
  kapellekensbaan: "/ka\u02C8p\u025Bl\u0259k\u0259ns\u02CCba\u02D0n/",
  kennelick: "/\u02C8k\u025Bn\u0259l\u0259k/",
  // archaic: evident
  kraanwagentje: "/\u02C8kra\u02D0n\u02CCwa\u02D0\u0263\u0259ntj\u0259/",
  laatsten: "/\u02C8la\u02D0tst\u0259n/",
  lande: "/\u02C8l\u0251nd\u0259/",
  // archaic dative: land
  lauriergracht: "/l\u0251u\u02C8ri\u02D0r\u0263r\u0251xt/",
  lesen: "/\u02C8le\u02D0z\u0259n/",
  // archaic: lezen
  lesten: "/\u02C8l\u025Bst\u0259n/",
  // archaic: last
  louterende: "/\u02C8l\u0251ut\u0259r\u025Bnd\u0259/",
  mooiste: "/\u02C8mo\u02D0ist\u0259/",
  // superlative: prettiest
  my: "/m\u025Bi/",
  // archaic: mij
  nassouwe: "/n\u0251\u02C8s\u0251u\u0259/",
  // archaic: Nassau
  niemandsbos: "/\u02C8ni\u02D0m\u0251nts\u02CCb\u0254s/",
  // nobody's forest
  oeroeg: "/\u02C8uru\u0263/",
  // proper name (Haasse)
  ondersaten: "/\u02C8\u0254nd\u0259r\u02CCza\u02D0t\u0259n/",
  // archaic: subjects
  ontwaakte: "/\u0254nt\u02C8wa\u02D0kt\u0259/",
  onverveerd: "/\u02CC\u0254nv\u0259r\u02C8ve\u02D0rt/",
  opgerezen: "/\u02C8\u0254p\u0263\u0259\u02CCre\u02D0z\u0259n/",
  osewoudt: "/\u02C8o\u02D0z\u0259\u02CCw\u0251ut/",
  // proper name (Hermans)
  ouders: "/\u02C8\u0251ud\u0259rs/",
  philips: "/\u02C8fil\u026Aps/",
  plachten: "/\u02C8pl\u0251xt\u0259n/",
  // archaic: used to
  pleegde: "/\u02C8ple\u02D0\u0263d\u0259/",
  po\u00EBtische: "/po\u02C8e\u02D0t\u026Asx\u0259/",
  prince: "/\u02C8pr\u026Ans\u0259/",
  // archaic: prins
  prinse: "/\u02C8pr\u026Ans\u0259/",
  // archaic variant
  reed: "/re\u02D0t/",
  romans: "/ro\u02C8m\u0251ns/",
  saluyt: "/sa\u02D0\u02C8l\u0153yt/",
  // archaic: greeting
  sarphatistraat: "/s\u0251r\u02C8fa\u02D0tistra\u02D0t/",
  // street name (Nescio)
  scherpste: "/\u02C8sx\u025Brpst\u0259/",
  schilderskade: "/\u02C8sx\u026Ald\u0259rs\u02CCka\u02D0d\u0259/",
  // street name
  Seynaeve: "/\u02C8s\u025Bina\u02D0v\u0259/",
  // Belgian proper name
  sien: "/si\u02D0n/",
  // archaic: zien
  smalle: "/\u02C8sm\u0251l\u0259/",
  // inflected: narrow
  spoorwegzate: "/\u02C8spo\u02D0r\u02CCw\u025Bxza\u02D0t\u0259/",
  // archaic: railway junction
  Steenwijk: "/\u02C8ste\u02D0n\u028B\u025Bik/",
  // place name
  stonden: "/\u02C8st\u0254nd\u0259n/",
  stopte: "/\u02C8st\u0254pt\u0259/",
  tamarindeboomen: "/ta\u02D0ma\u02D0\u02C8r\u026And\u0259\u02CCbo\u02D0m\u0259n/",
  // archaic: tamarind trees
  tegenwoordighe: "/\u02C8te\u02D0\u0263\u0259n\u02CCwo\u02D0rd\u0259\u0263\u0259/",
  // archaic
  terechtgekomen: "/t\u0259\u02C8r\u025Bxt\u0263\u0259\u02CCko\u02D0m\u0259n/",
  termurenlaan: "/t\u025Br\u02C8my\u02D0r\u0259n\u02CCla\u02D0n/",
  terugdenk: "/t\u0259\u02C8r\u028Fxd\u025B\u014Bk/",
  // compound: think back
  torentjes: "/\u02C8to\u02D0r\u0259ntj\u0259s/",
  toverplaatjes: "/\u02C8to\u02D0v\u0259r\u02CCpla\u02D0tj\u0259s/",
  // magic pictures
  tracht: "/tr\u0251xt/",
  // verb: try
  tragische: "/\u02C8tra\u02D0\u0263\u026Asx\u0259/",
  tusschen: "/\u02C8t\u028Fsx\u0259n/",
  // archaic: tussen (between)
  velden: "/\u02C8v\u025Bld\u0259n/",
  vermoordde: "/v\u0259r\u02C8mo\u02D0rd\u0259/",
  verschijnt: "/v\u0259r\u02C8sx\u025Bint/",
  // appears
  vlamde: "/\u02C8vl\u0251md\u0259/",
  vloog: "/vlo\u02D0x/",
  volcx: "/v\u0254lks/",
  // archaic: volks
  volle: "/\u02C8v\u0254l\u0259/",
  voorschoten: "/\u02C8vo\u02D0rsxo\u02D0t\u0259n/",
  // place name
  voorzomermiddag: "/\u02C8vo\u02D0r\u02CCzo\u02D0m\u0259r\u02CCm\u026Ad\u0251x/",
  vroege: "/\u02C8vru\u0263\u0259/",
  warme: "/\u02C8v\u0251rm\u0259/",
  weerd: "/we\u02D0rt/",
  // archaic: waard
  weggegaan: "/\u02C8v\u025Bx\u0263\u0259\u02CC\u0263a\u02D0n/",
  werd: "/w\u025Brt/",
  werkmenschen: "/\u02C8w\u025Brk\u02CCm\u025Bnsx\u0259n/",
  // archaic: working people
  "west-java": "/\u02CCw\u025Bst\u02C8ja\u02D0va/",
  wintrop: "/\u02C8v\u026Antr\u0254p/",
  // proper name (Mulisch)
  wonderlijker: "/\u02C8\u028B\u0254nd\u0259rl\u025Bik\u0259r/",
  // comparative: more wonderful
  woonde: "/\u02C8wo\u02D0nd\u0259/",
  wordt: "/w\u0254rt/",
  yegelick: "/\u02C8je\u02D0\u0263\u0259l\u026Ak/",
  // archaic: iedereen
  zakkende: "/\u02C8z\u0251k\u0259nd\u0259/",
  // sinking
  zekere: "/\u02C8ze\u02D0k\u0259r\u0259/",
  // a certain
  zijne: "/\u02C8z\u025Bin\u0259/",
  // archaic: zijn
  zocht: "/z\u0254xt/",
  zuivere: "/\u02C8z\u0153yv\u0259r\u0259/",
  // pure (inflected)
  zulke: "/\u02C8z\u028Flk\u0259/"
};

// src/overrides/or.ts
var or_ = {
  // Bande Utkala Janani (Odia anthem)
  \u0B05\u0B19\u0B4D\u0B17\u0B47: "\u0254\u014B\u0261e",
  // Jagannath Das — Odia Bhagabata
  \u0B05\u0B28\u0B3E\u0B26\u0B3F: "\u0254na\u02D0d\u032Ai",
  \u0B05\u0B2A\u0B4D\u0B30\u0B2E\u0B3F\u0B24: "\u0254p\u027E\u0254mit\u032A\u0254",
  \u0B05\u0B30\u0B4D\u0B25\u0B47: "\u0254\u027Et\u032A\u02B0e",
  \u0B05\u0B36\u0B47\u0B37: "\u0254\u0283e\u0282\u0254",
  \u0B06\u0B17\u0B41\u0B01: "a\u02D0\u0261\u0169",
  \u0B06\u0B24\u0B4D\u0B2E\u0B40\u0B5F: "a\u02D0t\u032Ami\u02D0j\u0254",
  \u0B06\u0B30\u0B24: "a\u02D0\u027E\u0254t\u032A\u0254",
  // Gangadhar Meher — Tapaswinee
  \u0B06\u0B39\u0B4D\u0B32\u0B3E\u0B26: "a\u02D0hl\u032Aa\u02D0d\u032A\u0254",
  \u0B07\u0B28\u0B4D\u0B26\u0B4D\u0B30\u0B28\u0B40\u0B33: "ind\u032A\u027E\u0254ni\u02D0l\u032A\u0254",
  \u0B09\u0B24\u0B4D\u0B15\u0B33\u0B2D\u0B41\u0B2C\u0B28\u0B47: "ut\u032Ak\u0254l\u032A\u0254b\u02B1ub\u0254ne",
  \u0B09\u0B24\u0B4D\u0B15\u0B33\u0B30: "ut\u032Ak\u0254l\u032A\u0254\u027E\u0254",
  // Upendra Bhanja — Baidehisha Bilasa
  \u0B09\u0B26\u0B4D\u0B2D\u0B2C: "ud\u032Ab\u02B1\u0254b\u0254",
  \u0B0F\u0B23\u0B47: "e\u0273e",
  \u0B14\u0B30\u0B38\u0B41: "\u0254w\u027E\u0254su",
  // Rangabati (folk song)
  \u0B15\u0B28\u0B15\u0B32\u0B24\u0B3E: "k\u0254n\u0254k\u0254l\u032A\u0254t\u032Aa\u02D0",
  \u0B15\u0B33: "k\u0254l\u032A\u0254",
  \u0B15\u0B39\u0B32\u0B4B: "k\u0254h\u0254l\u032Ao\u02D0",
  \u0B15\u0B3E\u0B24\u0B30: "ka\u02D0t\u032A\u0254\u027E\u0254",
  // Fakir Mohan Senapati — Chha Mana Atha Guntha
  \u0B15\u0B3E\u0B30\u0B2C\u0B3E\u0B30: "ka\u02D0\u027E\u0254ba\u02D0\u027E\u0254",
  \u0B15\u0B3E\u0B30\u0B3E\u0B26\u0B23\u0B4D\u0B21\u0B47: "ka\u02D0\u027Ea\u02D0d\u032A\u0254\u0273\u0256e",
  \u0B15\u0B3F\u0B2E\u0B4D\u0B2A\u0B3E: "kimpa\u02D0",
  \u0B15\u0B43\u0B37\u0B4D\u0B23\u0B2A\u0B3E\u0B26: "k\u027Eu\u0282\u0273\u0254pa\u02D0d\u032A\u0254",
  \u0B15\u0B47\u0B2C\u0B3E: "keba\u02D0",
  \u0B15\u0B47\u0B36\u0B3E: "ke\u0283a\u02D0",
  \u0B15\u0B4B\u0B36: "ko\u02D0\u0283\u0254",
  // Madhusudan Das — Utkala Santana
  \u0B17\u0B19\u0B4D\u0B17\u0B3E: "\u0261\u0254\u014B\u0261a\u02D0",
  \u0B17\u0B33\u0B41\u0B1B\u0B3F: "\u0261\u0254l\u032Aut\u0361\u0283\u02B0i",
  \u0B17\u0B40\u0B24\u0B28\u0B3E\u0B26: "\u0261i\u02D0t\u032A\u0254na\u02D0d\u032A\u0254",
  \u0B17\u0B41\u0B23\u0B3E\u0B33\u0B5F: "\u0261u\u0273a\u02D0l\u032A\u0254j\u0254",
  \u0B17\u0B41\u0B01\u0B25\u0B3E: "\u0261\u0169t\u032A\u02B0a\u02D0",
  \u0B17\u0B4B: "\u0261o\u02D0",
  \u0B17\u0B4B\u0B26\u0B3E\u0B2C\u0B30\u0B40: "\u0261o\u02D0d\u032Aa\u02D0b\u0254\u027Ei\u02D0",
  \u0B18\u0B1F\u0B47: "\u0261\u02B1\u0254\u0288e",
  \u0B18\u0B41\u0B2E\u0B41\u0B38\u0B30\u0B05\u0B27\u0B3F\u0B2A: "\u0261\u02B1umus\u0254\u027E\u0254d\u032A\u02B1ip\u0254",
  \u0B1A\u0B33\u0B47: "t\u0361\u0283\u0254l\u032Ae",
  \u0B1A\u0B33\u0B47\u0B28\u0B3E\u0B39\u0B3F\u0B01: "t\u0361\u0283\u0254l\u032Aena\u02D0h\u0129",
  \u0B1A\u0B3E\u0B30\u0B41: "t\u0361\u0283a\u02D0\u027Eu",
  \u0B1A\u0B3E\u0B32\u0B3F\u0B2F\u0B3E\u0B06\u0B28\u0B4D\u0B24\u0B41: "t\u0361\u0283a\u02D0l\u032Aija\u02D0a\u02D0nt\u032Au",
  \u0B1C\u0B17\u0B28\u0B4D\u0B28\u0B3E\u0B25\u0B47: "d\u0361\u0292\u0254\u0261\u0254nna\u02D0t\u032A\u02B0e",
  \u0B1C\u0B28\u0B4D\u0B2E\u0B15\u0B3E\u0B33\u0B30\u0B41: "d\u0361\u0292\u0254nm\u0254ka\u02D0l\u032A\u0254\u027Eu",
  \u0B1C\u0B2E\u0B3F\u0B26\u0B3E\u0B30: "d\u0361\u0292\u0254mid\u032Aa\u02D0\u027E\u0254",
  \u0B1C\u0B3F\u0B24: "d\u0361\u0292it\u032A\u0254",
  \u0B1C\u0B4D\u0B5F\u0B4B\u0B24\u0B3F\u0B30\u0B4D\u0B2E\u0B5F\u0B40: "d\u0361\u0292jo\u02D0t\u032Ai\u027Em\u0254ji\u02D0",
  \u0B24\u0B26\u0B4D\u0B27\u0B24: "t\u032A\u0254d\u032Ad\u032A\u02B1\u0254t\u032A\u0254",
  \u0B24\u0B28\u0B41\u0B15\u0B3E\u0B28\u0B4D\u0B24\u0B3F: "t\u032A\u0254nuka\u02D0nt\u032Ai",
  \u0B24\u0B30\u0B19\u0B4D\u0B17\u0B47: "t\u032A\u0254\u027E\u0254\u014B\u0261e",
  \u0B24\u0B30\u0B3F\u0B32\u0B47: "t\u032A\u0254\u027Eil\u032Ae",
  \u0B24\u0B3E\u0B17\u0B47: "t\u032Aa\u02D0\u0261e",
  \u0B24\u0B41\u0B39\u0B3F: "t\u032Auhi",
  \u0B24\u0B4B\u0B39\u0B30: "t\u032Ao\u02D0h\u0254\u027E\u0254",
  // Fakir Mohan Senapati — Odia Kahiki Daridra
  \u0B26\u0B30\u0B3F\u0B26\u0B4D\u0B30\u0B24\u0B3E: "d\u032A\u0254\u027Eid\u032A\u027E\u0254t\u032Aa\u02D0",
  \u0B26\u0B40\u0B30\u0B4D\u0B18\u0B3F\u0B15\u0B3E: "d\u032Ai\u02D0\u027E\u0261\u02B1ika\u02D0",
  \u0B26\u0B40\u0B30\u0B4D\u0B18\u0B47: "d\u032Ai\u02D0\u027E\u0261\u02B1e",
  \u0B26\u0B41\u0B03\u0B16: "d\u032Auk\u02B1\u0254",
  \u0B26\u0B41\u0B30\u0B4D\u0B17\u0B24\u0B3F\u0B30: "d\u032Au\u027E\u0261\u0254t\u032Ai\u027E\u0254",
  \u0B26\u0B47\u0B36\u0B2C\u0B3E\u0B38\u0B40: "d\u032Ae\u0283\u0254ba\u02D0si\u02D0",
  \u0B26\u0B47\u0B36\u0B2E\u0B3E\u0B1F\u0B3F\u0B30\u0B47: "d\u032Ae\u0283\u0254ma\u02D0\u0288i\u027Ee",
  \u0B26\u0B4D\u0B5F\u0B41\u0B24\u0B3F: "d\u032Ajut\u032Ai",
  \u0B27\u0B28\u0B1E\u0B4D\u0B1C\u0B5F: "d\u032A\u02B1\u0254n\u0254\u0272d\u0361\u0292\u0254j\u0254",
  \u0B27\u0B3E\u0B28\u0B30: "d\u032A\u02B1a\u02D0n\u0254\u027E\u0254",
  \u0B28: "n\u0254",
  \u0B28\u0B17\u0B26: "n\u0254\u0261\u0254d\u032A\u0254",
  \u0B28\u0B28\u0B4D\u0B26\u0B28: "n\u0254nd\u032A\u0254n\u0254",
  \u0B28\u0B2E\u0B07\u0B01: "n\u0254m\u0254\u0129",
  \u0B28\u0B30\u0B4D\u0B15\u0B47: "n\u0254\u027Eke",
  \u0B28\u0B3E\u0B30\u0B40\u0B19\u0B4D\u0B15: "na\u02D0\u027Ei\u02D0\u014Bk\u0254",
  \u0B28\u0B3F\u0B30\u0B28\u0B4D\u0B24\u0B30\u0B47: "ni\u027E\u0254nt\u032A\u0254\u027Ee",
  \u0B28\u0B3F\u0B39\u0B3F\u0B24: "nihit\u032A\u0254",
  \u0B28\u0B40\u0B33\u0B3E\u0B2E\u0B4D\u0B2C\u0B41: "ni\u02D0l\u032Aa\u02D0mbu",
  \u0B28\u0B43\u0B2A: "n\u027Eup\u0254",
  \u0B2A\u0B21\u0B3C\u0B3F\u0B25\u0B3E\u0B09: "p\u0254\u0256\u032Ait\u032A\u02B0a\u02D0u",
  \u0B2A\u0B26\u0B41: "p\u0254d\u032Au",
  \u0B2A\u0B26\u0B47: "p\u0254d\u032Ae",
  \u0B2A\u0B5F\u0B30\u0B47: "p\u0254j\u0254\u027Ee",
  \u0B2A\u0B30\u0B3E: "p\u0254\u027Ea\u02D0",
  \u0B2A\u0B42\u0B30\u0B4D\u0B2C\u0B2A\u0B41\u0B30\u0B41\u0B37: "pu\u02D0\u027Eb\u0254pu\u027Eu\u0282\u0254",
  \u0B2A\u0B4B\u0B37\u0B23: "po\u02D0\u0282\u0254\u0273\u0254",
  \u0B2A\u0B4D\u0B30\u0B1C\u0B4D\u0B1E\u0B3E: "p\u027E\u0254d\u0361\u0292\u0272a\u02D0",
  \u0B2A\u0B4D\u0B30\u0B2C\u0B28\u0B4D\u0B27\u0B47: "p\u027E\u0254b\u0254nd\u032A\u02B1e",
  \u0B2A\u0B4D\u0B30\u0B3E\u0B23\u0B40\u0B2E\u0B3E\u0B28\u0B19\u0B4D\u0B15: "p\u027Ea\u02D0\u0273i\u02D0ma\u02D0n\u0254\u014Bk\u0254",
  \u0B2C\u0B28\u0B2D\u0B42\u0B2E\u0B3F: "b\u0254n\u0254b\u02B1u\u02D0mi",
  \u0B2C\u0B28\u0B4D\u0B26\u0B28: "b\u0254nd\u032A\u0254n\u0254",
  \u0B2C\u0B28\u0B4D\u0B26\u0B47: "b\u0254nd\u032Ae",
  \u0B2C\u0B30\u0B39\u0B3F\u0B2C\u0B02\u0B36\u0B47: "b\u0254\u027E\u0254hib\u0254\u014B\u0283e",
  \u0B2C\u0B30\u0B4D\u0B23\u0B4D\u0B23\u0B07: "b\u0254\u027E\u0273\u0273\u0254i",
  \u0B2C\u0B3F\u0B1A\u0B3E\u0B30\u0B47: "bit\u0361\u0283a\u02D0\u027Ee",
  \u0B2C\u0B3F\u0B28\u0B41: "binu",
  \u0B2C\u0B3F\u0B2C\u0B47\u0B15: "bibek",
  \u0B2C\u0B3F\u0B33\u0B3E\u0B38: "bil\u032Aa\u02D0s\u0254",
  \u0B2C\u0B3F\u0B36\u0B3F\u0B37\u0B4D\u0B1F\u0B47: "bi\u0283i\u0282\u0288e",
  \u0B2C\u0B3F\u0B39\u0B19\u0B4D\u0B17\u0B47: "bih\u0254\u014B\u0261e",
  \u0B2C\u0B47\u0B28\u0B3F: "beni",
  \u0B2C\u0B47\u0B36\u0B3E: "be\u0283a\u02D0",
  \u0B2C\u0B4B\u0B32\u0B3E\u0B09: "bo\u02D0l\u032Aa\u02D0u",
  \u0B2C\u0B4D\u0B5F\u0B25\u0B3E: "bj\u0254t\u032A\u02B0a\u02D0",
  \u0B2C\u0B4D\u0B30\u0B1C: "b\u027E\u0254d\u0361\u0292\u0254",
  \u0B2D\u0B23\u0B4D\u0B21\u0B3E\u0B30: "b\u02B1\u0254\u0273\u0256a\u02D0\u027E\u0254",
  \u0B2D\u0B3E\u0B24\u0B43\u0B2D\u0B3E\u0B2C: "b\u02B1a\u02D0t\u032A\u027Eub\u02B1a\u02D0b\u0254",
  \u0B2D\u0B3E\u0B37\u0B2E\u0B5F\u0B40: "b\u02B1a\u02D0\u0282\u0254m\u0254ji\u02D0",
  \u0B2D\u0B40\u0B30\u0B41: "b\u02B1i\u02D0\u027Eu",
  \u0B2D\u0B42\u0B27\u0B30\u0B2E\u0B3E\u0B33\u0B3E: "b\u02B1u\u02D0d\u032A\u02B1\u0254\u027E\u0254ma\u02D0l\u032Aa\u02D0",
  \u0B2D\u0B47\u0B26\u0B3F: "b\u02B1ed\u032Ai",
  \u0B2E\u0B15\u0B30\u0B28\u0B4D\u0B26: "m\u0254k\u0254\u027E\u0254nd\u032A\u0254",
  \u0B2E\u0B19\u0B4D\u0B17\u0B30\u0B3E\u0B1C: "m\u0254\u014B\u0261\u0254\u027Ea\u02D0d\u0361\u0292\u0254",
  \u0B2E\u0B28\u0B30\u0B47: "m\u0254n\u0254\u027Ee",
  \u0B2E\u0B28\u0B4B\u0B39\u0B30: "m\u0254no\u02D0h\u0254\u027E\u0254",
  \u0B2E\u0B2B\u0B38\u0B32\u0B30: "m\u0254p\u02B0\u0254s\u0254l\u032A\u0254\u027E\u0254",
  \u0B2E\u0B30\u0B3E\u0B33\u0B2E\u0B3E\u0B33\u0B3F\u0B28\u0B40: "m\u0254\u027Ea\u02D0l\u032A\u0254ma\u02D0l\u032Aini\u02D0",
  \u0B2E\u0B30\u0B4D\u0B2F\u0B4D\u0B5F\u0B3E\u0B26\u0B3E: "m\u0254\u027Ed\u0292ja\u02D0d\u032Aa\u02D0",
  \u0B2E\u0B39\u0B3E\u0B1C\u0B28: "m\u0254ha\u02D0d\u0361\u0292\u0254n\u0254",
  \u0B2E\u0B39\u0B3E\u0B1C\u0B28\u0B40: "m\u0254ha\u02D0d\u0361\u0292\u0254ni\u02D0",
  \u0B2E\u0B3E\u0B32\u0B47: "ma\u02D0l\u032Ae",
  \u0B2E\u0B3F\u0B36\u0B41: "mi\u0283u",
  \u0B2E\u0B41\u0B16\u0B30\u0B3F\u0B24: "muk\u02B1\u0254\u027Eit\u032A\u0254",
  \u0B30\u0B19\u0B4D\u0B17\u0B2C\u0B24\u0B40: "\u027E\u0254\u014B\u0261\u0254b\u0254t\u032Ai\u02D0",
  \u0B30\u0B1C\u0B3E: "\u027E\u0254d\u0361\u0292a\u02D0",
  \u0B30\u0B3E\u0B1C\u0B3F\u0B24: "\u027Ea\u02D0d\u0361\u0292it\u032A\u0254",
  \u0B30\u0B3E\u0B28\u0B3F: "\u027Ea\u02D0ni",
  \u0B30\u0B4B\u0B26\u0B28: "\u027Eo\u02D0d\u032A\u0254n\u0254",
  \u0B36\u0B41\u0B23\u0B3E\u0B2F\u0B3E\u0B0F: "\u0283u\u0273a\u02D0ja\u02D0e",
  \u0B36\u0B41\u0B26\u0B4D\u0B27: "\u0283ud\u032Ad\u032A\u02B1\u0254",
  \u0B36\u0B41\u0B2D: "\u0283ub\u02B1\u0254",
  \u0B36\u0B41\u0B2D\u0B4D\u0B30: "\u0283ub\u02B1\u027E\u0254",
  \u0B36\u0B4B\u0B2D\u0B3E\u0B30: "\u0283ob\u02B1a\u02D0\u027E\u0254",
  \u0B38\u0B2A\u0B28: "s\u0254p\u0254n\u0254",
  \u0B38\u0B39\u0B41: "s\u0254hu",
  \u0B38\u0B3E\u0B1C\u0B47: "sa\u02D0d\u0361\u0292e",
  \u0B38\u0B41\u0B24\u0B3E: "sut\u032Aa\u02D0",
  \u0B38\u0B41\u0B2A\u0B4D\u0B30\u0B15\u0B3E\u0B36: "sup\u027E\u0254ka\u02D0\u0283\u0254",
  \u0B38\u0B47\u0B2E\u0B3E\u0B28\u0B19\u0B4D\u0B15\u0B20\u0B3E\u0B30\u0B47: "sema\u02D0n\u0254\u014Bk\u0254\u0288\u02B0a\u02D0\u027Ee",
  \u0B38\u0B4B\u0B26\u0B30: "so\u02D0d\u032A\u0254\u027E\u0254",
  \u0B38\u0B4D\u0B2B\u0B41\u0B30\u0B4D\u0B24\u0B4D\u0B24\u0B3F: "sp\u02B0u\u027Et\u032At\u032Ai",
  "\u0B38\u0B4D\u0B71\u0B3E\u0B27\u0B40\u0B28": "swa\u02D0d\u032A\u02B1i\u02D0n\u0254",
  \u0B39\u0B3E\u0B38\u0B2E\u0B5F\u0B40: "ha\u02D0s\u0254m\u0254ji\u02D0",
  \u0B39\u0B3F\u0B24\u0B47: "hit\u032Ae",
  \u0B39\u0B43\u0B26\u0B5F\u0B30\u0B47: "h\u027Eud\u032A\u0254j\u0254\u027Ee",
  \u0B39\u0B43\u0B26\u0B47: "h\u027Eud\u032Ae",
  \u0B39\u0B47\u0B32: "hel\u032A\u0254",
  \u0B39\u0B47\u0B32\u0B3E\u0B24: "hel\u032Aa\u02D0t\u032A\u0254"
};

// src/overrides/pt.ts
var pt = {
  a: "/a/",
  agora: "/a\u02C8\u0261o\u027Ea/",
  ainda: "/a\u02C8\u0129da/",
  alto: "/\u02C8awtu/",
  alvejei: "/awve\u02C8\u0292ej/",
  antes: "/\u02C8\xE3t\u0283is/",
  // New sample overrides
  Antigamente: "/\xE3t\u0283i\u0261a\u02C8m\u1EBDt\u0283i/",
  aparecesse: "/apa\u027Ee\u02C8sesi/",
  assembleia: "/as\u1EBD\u02C8bleja/",
  autom\u00F3veis: "/awto\u02C8m\u0254vejs/",
  autor: "/aw\u02C8to\u027E/",
  aventura: "/av\u1EBD\u02C8tu\u027Ea/",
  bater: "/ba\u02C8tex/",
  beija: "/\u02C8bej\u0292a/",
  bem: "/b\u1EBDj/",
  calmo: "/\u02C8kawmu/",
  cama: "/\u02C8k\xE3ma/",
  certa: "/\u02C8s\u025B\u027Eta/",
  chamam: "/\u02C8\u0283\xE3m\xE3w/",
  como: "/\u02C8komu/",
  contraparente: "/k\xF5t\u027Eapa\u02C8\u027E\u1EBDt\u0283i/",
  costume: "/kos\u02C8t\u0169mi/",
  cotovelo: "/koto\u02C8velu/",
  cruzarmos: "/k\u027Euz\u02C8a\u027Emus/",
  d: "/de/",
  // abbreviation for Dom
  de: "/d\u0292i/",
  dei: "/dej/",
  dele: "/\u02C8deli/",
  dentro: "/\u02C8d\u1EBDt\u027Eu/",
  deu: "/dew/",
  dia: "/\u02C8d\u0292ia/",
  direita: "/d\u0292i\u02C8\u027Eejta/",
  direito: "/d\u0292i\u02C8\u027Eejtu/",
  disco: "/\u02C8d\u0292isku/",
  disposto: "/d\u0292is\u02C8postu/",
  disse: "/\u02C8d\u0292isi/",
  dois: "/dojs/",
  \u00E9: "/\u025B/",
  emprenhou: "/\u1EBDp\u027Ee\u02C8\u0272ow/",
  encante: "/\u1EBD\u02C8k\xE3t\u0283i/",
  entre: "/\u02C8\u1EBDt\u027Ei/",
  escolherei: "/isko\u028Ee\u02C8\u027Eej/",
  esf\u00EDngico: "/es\u02C8f\u0129\u0292iku/",
  eterna: "/e\u02C8t\u025B\u027Ena/",
  eu: "/ew/",
  Europa: "/ew\u02C8\u027E\u0254pa/",
  face: "/\u02C8fasi/",
  faixas: "/\u02C8faj\u0283as/",
  falsa: "/\u02C8fawsa/",
  faz: "/fas/",
  fim: "/f\u0129/",
  Fita: "/\u02C8fita/",
  fita: "/\u02C8fita/",
  fitando: "/fi\u02C8t\xE3du/",
  flameja: "/fla\u02C8me\u0292a/",
  focinhando: "/fosi\u02C8\u0272\xE3du/",
  foi: "/foj/",
  fossem: "/\u02C8fos\u1EBDj/",
  fosses: "/\u02C8fosis/",
  fragorosas: "/f\u027Ea\u0261o\u02C8\u027Eozas/",
  francisco: "/f\u027E\xE3\u02C8sisku/",
  frente: "/\u02C8f\u027E\u1EBDt\u0283i/",
  f\u00FAlgidos: "/\u02C8fuw\u0292idus/",
  gente: "/\u02C8\u0292\u1EBDt\u0283i/",
  girassol: "/\u0292i\u027Ea\u02C8s\u0254w/",
  gosto: "/\u02C8\u0261ostu/",
  grandes: "/\u02C8\u0261\u027E\xE3d\u0292is/",
  havia: "/a\u02C8via/",
  hei: "/ej/",
  hesitei: "/ezi\u02C8tej/",
  homens: "/\u02C8\xF5m\u1EBDjs/",
  houve: "/\u02C8ovi/",
  individuais: "/\u0129d\u0292ividu\u02C8ajs/",
  instante: "/\u0129s\u02C8t\xE3t\u0283i/",
  instituir: "/\u0129st\u0283itu\u02C8i\u027E/",
  isso: "/\u02C8isu/",
  jos\u00E9: "/\u0292o\u02C8z\u025B/",
  liberdade: "/libe\u027E\u02C8dad\u0292i/",
  lisboa: "/liz\u02C8boa/",
  louca: "/\u02C8loka/",
  louvor: "/lo\u02C8vo\u027E/",
  lua: "/\u02C8lua/",
  maior: "/ma\u02C8jo\u027E/",
  mais: "/majs/",
  mal: "/maw/",
  margens: "/\u02C8ma\u027E\u0292\u1EBDjs/",
  maus: "/maws/",
  mesti\u00E7ara: "/mesti\u02C8sa\u027Ea/",
  m\u00E9todo: "/\u02C8m\u025Btudu/",
  mil: "/miw/",
  mim: "/m\u0129/",
  morder: "/mo\u027E\u02C8dex/",
  morte: "/\u02C8m\u0254\u027Et\u0283i/",
  morto: "/\u02C8mo\u027Etu/",
  mostra: "/\u02C8m\u0254st\u027Ea/",
  mulher: "/mu\u02C8\u028E\u025Bx/",
  nacional: "/nasio\u02C8naw/",
  nascimento: "/nasi\u02C8m\u1EBDtu/",
  nele: "/\u02C8neli/",
  nem: "/n\u1EBDj/",
  nesse: "/\u02C8nesi/",
  noite: "/\u02C8nojt\u0283i/",
  nonada: "/no\u02C8nada/",
  // nothing (Guimarães Rosa)
  nosso: "/\u02C8nosu/",
  novo: "/\u02C8novu/",
  num: "/n\u0169/",
  o: "/u/",
  \u00F3: "/\u0254/",
  ondas: "/\u02C8\xF5das/",
  onde: "/\u02C8\xF5d\u0292i/",
  ou: "/ow/",
  ousadia: "/oza\u02C8d\u0292ia/",
  outono: "/ow\u02C8tonu/",
  outra: "/\u02C8owt\u027Ea/",
  outro: "/\u02C8owt\u027Eu/",
  para: "/\u02C8pa\u027Ea/",
  parte: "/\u02C8pa\u027Et\u0283i/",
  passadeira: "/pasa\u02C8dej\u027Ea/",
  pode: "/\u02C8p\u0254d\u0292i/",
  portugal: "/po\u027Etu\u02C8\u0261aw/",
  posso: "/\u02C8posu/",
  posta: "/\u02C8p\u0254sta/",
  povo: "/\u02C8povu/",
  praia: "/\u02C8p\u027Eaja/",
  prometia: "/p\u027Eome\u02C8t\u0283ia/",
  quanto: "/\u02C8kw\xE3tu/",
  quarto: "/\u02C8kwa\u027Etu/",
  quer: "/k\u025B\u027E/",
  querer: "/ke\u02C8\u027Eex/",
  ramalhete: "/xama\u02C8\u028Eet\u0283i/",
  // mansion name (Eça de Queirós)
  real: "/xe\u02C8aw/",
  rebentavam: "/xeb\u1EBD\u02C8tav\xE3w/",
  rei: "/xej/",
  reino: "/\u02C8xejnu/",
  remota: "/xe\u02C8m\u0254ta/",
  retumbante: "/xet\u0169\u02C8b\xE3t\u0283i/",
  rir: "/xi\u027E/",
  riso: "/\u02C8xizu/",
  rosto: "/\u02C8xostu/",
  rua: "/\u02C8xua/",
  s: "/s\xE3w/",
  // abbreviation for São
  saber: "/sa\u02C8bex/",
  sal: "/saw/",
  sei: "/sej/",
  sem: "/s\u1EBDj/",
  sempre: "/\u02C8s\u1EBDp\u027Ei/",
  senhor: "/se\u02C8\u0272o\u027E/",
  sequer: "/se\u02C8k\u025B\u027E/",
  ser: "/\u02C8sex/",
  sim: "/s\u0129/",
  simples: "/\u02C8s\u0129plis/",
  Sob: "/s\u0254b/",
  sociais: "/sosi\u02C8ajs/",
  sol: "/s\u0254w/",
  sua: "/\u02C8sua/",
  sublimaram: "/subli\u02C8ma\u027E\xE3w/",
  tanto: "/\u02C8t\xE3tu/",
  taprobana: "/tap\u027Eo\u02C8b\xE3na/",
  // ancient Sri Lanka
  te: "/t\u0283i/",
  tempo: "/\u02C8t\u1EBDpu/",
  ter: "/tex/",
  tive: "/\u02C8t\u0283ivi/",
  "toldam-lhe": "/\u02C8towd\xE3w\u028Ei/",
  trapiche: "/t\u027Ea\u02C8pi\u0283i/",
  triste: "/\u02C8t\u027Eist\u0283i/",
  um: "/\u0169/",
  uma: "/\u02C8\u0169ma/",
  universo: "/uni\u02C8v\u025B\u027Esu/",
  uso: "/\u02C8uzu/",
  vale: "/\u02C8vali/",
  verde: "/\u02C8ve\u027Ed\u0292i/",
  verdes: "/\u02C8ve\u027Ed\u0292is/",
  vez: "/ves/",
  viagem: "/vi\u02C8a\u0292\u1EBDj/",
  "viv\xEA-lo": "/vi\u02C8velu/",
  "vou-me": "/\u02C8vomi/",
  zelo: "/\u02C8zelu/",
  zomba: "/\u02C8z\xF5ba/"
};

// src/overrides/ro.ts
var ro = {
  // New sample overrides
  admirabil\u0103: "/admira\u02C8bil\u0259/",
  alearg\u0103: "/a\u02C8le\u032Farg\u0259/",
  alergi: "/a\u02C8lerd\u0292\u02B2/",
  Armadia: "/ar\u02C8madja/",
  b\u0103ie\u0219a\u0219: "/b\u0259je\u02C8\u0283a\u0283/",
  "ca-ntr-un": "/kan\u02C8trun/",
  C\u00E2rlibaba: "/k\u0268rli\u02C8baba/",
  ciob\u0103nei: "/t\u0283ob\u0259\u02C8nej/",
  ciople\u0219te: "/t\u0283o\u02C8ple\u0283te/",
  cirezilor: "/t\u0283i\u02C8rezilor/",
  cl\u0103di: "/kl\u0259\u02C8d\u02B2i/",
  codrul: "/\u02C8kodrul/",
  crengi: "/krend\u0292\u02B2/",
  "croie\u0219te-\u021Bi": "/kro\u02C8je\u0283tet\u0283i/",
  curat\u0103: "/ku\u02C8rat\u0259/",
  deshame: "/des\u02C8hame/",
  desprind: "/des\u02C8prind/",
  "De\u0219teapt\u0103-te": "/de\u0283\u02C8te\u032Fapt\u0259te/",
  fecioara: "/fet\u0283o\u02C8ara/",
  ferestrelor: "/fe\u02C8restrelor/",
  furtunoas\u0103: "/furtu\u02C8no\u032Fas\u0259/",
  gem\u00E2nd: "/d\u0292e\u02C8m\u0268nd/",
  grupurile: "/\u02C8\u0261rupurile/",
  homer: "/\u02C8homer/",
  humule\u0219tii: "/humu\u02C8le\u0283ti.i/",
  \u00EEncepusem: "/\u0268nt\u0283e\u02C8pusem/",
  \u00EEntinsese: "/\u0268ntin\u02C8sese/",
  \u00EEntov\u0103r\u0103\u0219ind: "/\u0268ntov\u0259r\u0259\u02C8\u0283ind/",
  intrat\u0103: "/in\u02C8trat\u0259/",
  Jidovi\u021Ba: "/\u0292ido\u02C8vits\u0259/",
  "l-a": "/la/",
  "le-nt\xE2lnesc": "/lent\u0268l\u02C8nesk/",
  luasem: "/lu\u02C8asem/",
  lustruie\u0219te: "/lustru\u02C8je\u0283te/",
  "m\xE2n\u0103-n": "/\u02C8m\u0268n\u0259n/",
  "m\xE2ndr\u0103-n": "/\u02C8m\u0268ndr\u0259n/",
  morminte: "/mor\u02C8minte/",
  muceg\u0103it\u0103: "/mut\u0283e\u0261\u0259\u02C8it\u0259/",
  na\u021Bional\u0103: "/natsio\u02C8nal\u0259/",
  Obloanele: "/oblo\u032Fa\u02C8nele/",
  "ochiu-adormit": "/\u02C8okju ador\u02C8mit/",
  paraschi: "/para\u02C8ski/",
  p\u0103rin\u021Bilor: "/p\u0259\u02C8rintsilor/",
  "pe-ntinsele": "/pentin\u02C8sele/",
  Prahovei: "/pra\u02C8hovej/",
  r\u0103sar: "/r\u0259\u02C8sar/",
  r\u0103zuie\u0219te: "/r\u0259zu\u02C8je\u0283te/",
  r\u0103zvr\u0103tit\u0103: "/r\u0259zvr\u0259\u02C8tit\u0259/",
  sc\u00E2r\u021B\u00E2iau: "/sk\u0268rts\u0268\u02C8jaw/",
  sc\u00E2r\u021B\u00E2ie: "/sk\u0268r\u02C8ts\u0268je/",
  scoal\u0103: "/\u02C8sko\u032Fal\u0259/",
  scrumit: "/skru\u02C8mit/",
  scuipat: "/skui\u02C8pat/",
  "se-nchine": "/sen\u02C8kine/",
  sfin\u021Bi: "/sfints\u02B2/",
  sicriele: "/si\u02C8krijele/",
  socoate: "/so\u02C8ko\u032Fate/",
  Some\u0219ul: "/\u02C8some\u0283ul/",
  \u0219oseaua: "/\u0283o\u02C8se\u032Fawa/",
  stam: "/stam/",
  strivesc: "/stri\u02C8vesk/",
  sublocotenent: "/sublokote\u02C8nent/",
  t\u0103ind: "/t\u0259\u02C8ind/",
  tainele: "/\u02C8tajnele/",
  "te-ad\xE2ncir\u0103": "/tead\u0268n\u02C8t\u0283ir\u0259/",
  "toate-s": "/\u02C8to\u032Fates/",
  tr\u0103snesc: "/tr\u0259s\u02C8nesk/",
  ucid: "/u\u02C8t\u0283id/",
  uneltele: "/u\u02C8neltele/",
  v\u0103ii: "/\u02C8v\u0259ji/",
  v\u00E2nturat\u0103: "/v\u0268ntu\u02C8rat\u0259/",
  ve\u0219tm\u00E2nt: "/ve\u0283t\u02C8m\u0268nt/",
  "vijelia-ngrozitoare": "/vi\u02C8\u0292eli.a \u014B\u0261rozi\u02C8to\u032Fare/",
  "Vino-n": "/vi\u02C8non/"
};

// src/overrides/sv.ts
var sv = {
  \u00E4dlaste: "/\u02C8\u025B\u02D0dlast\u025B/",
  \u00E4ngder: "/\u02C8\u025B\u014B\u02D0d\u025Br/",
  // New entries for updated samples
  anl\u00E4nt: "/\u02C8an\u02D0l\u025Bnt/",
  arbetsrum: "/\u02C8ar\u02D0bets\u02CCr\u0275m/",
  arvedel: "/\u02C8ar\u02D0ve\u02CCde\u02D0l/",
  atenare: "/at\u025B\u02C8n\u0251\u02D0r\u025B/",
  \u00E5ttiotv\u00E5: "/\u02C8\u0254t\u02D0\u026A\u0254\u02CCtvo\u02D0/",
  \u00E5ttitalet: "/\u02C8\u0254t\u02D0\u026A\u02CCt\u0251\u02D0l\u025Bt/",
  barnkullarnas: "/\u02C8b\u0251\u02D0\u0273\u02CCk\u0275l\u02D0arnas/",
  // New sample overrides
  Birck: "/b\u026Ark/",
  b\u00F6nderna: "/\u02C8b\u0153nd\u025B\u0273a/",
  brynja: "/\u02C8bryn\u02D0ja/",
  bryt: "/bry\u02D0t/",
  d\u00E4rn\u00E4st: "/\u02C8d\xE6\u02D0\u0273\u025Bst/",
  dimmors: "/\u02C8d\u026Am\u02D0\u0254\u0282/",
  eget: "/\u02C8e\u02D0\u0261\u025Bt/",
  f\u00E5gels\u00E5ng: "/\u02C8fo\u02D0\u0261\u025Bl\u02CCs\u0254\u014B\u02D0/",
  fj\u00E4llh\u00F6ga: "/\u02C8fj\u025Bl\u02D0\u02CCh\xF8\u02D0\u0261a/",
  fjol\u00E5rets: "/\u02C8fju\u02D0l\u02CCo\u02D0r\u025Bts/",
  fornstora: "/\u02C8f\u0254\u02D0\u0273\u02CCstu\u02D0ra/",
  f\u00F6rt\u00E4tas: "/f\u0153r\u02C8t\u025B\u02D0tas/",
  fredagskv\u00E4ll: "/\u02C8fre\u02D0da\u0261s\u02CCkv\u025Bl\u02D0/",
  gamla: "/\u02C8\u0261am\u02D0la/",
  gl\u00E4djerika: "/\u02C8\u0261l\u025Bd\u02D0j\u025B\u02CCri\u02D0ka/",
  haga: "/\u02C8h\u0251\u02D0\u0261a/",
  // place name
  halvm\u00F6rkret: "/\u02C8halv\u02CCm\u0153rkr\u025Bt/",
  "hedeby-\xF6n": "/\u02C8he\u02D0d\u025Bby\u02D0\u02CC\xF8\u02D0n/",
  // place name
  henrik: "/\u02C8h\u025Bnr\u026Ak/",
  // proper name
  h\u00F6llo: "/\u02C8h\xF8l\u02D0\u0254/",
  improvisationer: "/\u026Ampr\u0254v\u026Asa\u02C8\u0267u\u02D0n\u025Br/",
  Ingmarss\u00F6nerna: "/\u02C8\u026A\u014B\u02D0mar\u02CCs\xF8\u02D0n\u025B\u0273a/",
  j\u00E4ttestaden: "/\u02C8j\u025Bt\u02D0\u025B\u02CCst\u0251\u02D0d\u025Bn/",
  kriminalkommissarie: "/kr\u026Am\u026A\u02C8n\u0251\u02D0lk\u0254m\u026A\u02CCs\u0251\u02D0r\u026A\u025B/",
  kristina: "/kr\u026A\u02C8sti\u02D0na/",
  // proper name
  kv\u00E4llsm\u00F6rkret: "/\u02C8kv\u025Bl\u02D0s\u02CCm\u0153rkr\u025Bt/",
  l\u00E5ngstrump: "/\u02C8l\u0254\u014B\u02D0\u02CCstr\u0275mp/",
  linh\u00E5rig: "/\u02C8li\u02D0n\u02CCho\u02D0r\u026A\u0261/",
  ljusl\u00E5gor: "/\u02C8j\u0289\u02D0s\u02CClo\u02D0\u0261\u0254r/",
  l\u00F6vsamlingar: "/\u02C8l\xF8\u02D0v\u02CCsam\u02D0l\u026A\u014B\u02D0ar/",
  lyfte: "/\u02C8l\u028Fft\u025B/",
  lyftes: "/\u02C8l\u028Fft\u025Bs/",
  m\u00E4nniskor: "/\u02C8m\u025Bn\u02D0\u026A\u02CC\u0267\u0254r/",
  m\u00E4nniskors: "/\u02C8m\u025Bn\u02D0\u026A\u02CC\u0267\u0254\u0282/",
  Martin: "/\u02C8mar\u02D0t\u026An/",
  m\u00E4ssfall: "/\u02C8m\u025Bs\u02D0\u02CCfal\u02D0/",
  mosebacke: "/\u02C8mu\u02D0s\u025B\u02CCbak\u02D0\u025B/",
  new: "/nju\u02D0/",
  nilsson: "/\u02C8n\u026Al\u02D0s\u0254n/",
  // proper name
  nordamerika: "/\u02C8nu\u02D0\u0256a\u02CCme\u02D0r\u026Aka/",
  numret: "/\u02C8n\u0275m\u02D0r\u025Bt/",
  nytt: "/n\u028Ft\u02D0/",
  odygd: "/\u02C8u\u02D0\u02CCdy\u0261d/",
  ofruktsamt: "/\u02C8u\u02D0\u02CCfr\u0275ktsamt/",
  \u00F6mt\u00E5ligare: "/\u02C8\xF8m\u02D0\u02CCto\u02D0l\u026A\u0261ar\u025B/",
  \u00F6ppnad: "/\u02C8\u0153p\u02D0nad/",
  orimligheter: "/u\u02D0\u02C8ri\u02D0ml\u026A\u0261\u02CChe\u02D0t\u025Br/",
  oskar: "/\u02C8\u0254skar/",
  // proper name
  page: "/\u02C8p\u0251\u02D0\u0267\u0259/",
  p\u00E4rontr\u00E4d: "/\u02C8p\xE6\u02D0r\u0254n\u02CCtr\u025B\u02D0d/",
  paulun: "/pa\u02C8l\u0289\u02D0n/",
  presentpappret: "/pr\u025B\u02C8s\u025Bnt\u02CCpap\u02D0r\u025Bt/",
  proportionerad: "/pr\u0254p\u0254\u0282\u0254\u02C8ne\u02D0rad/",
  roligast: "/\u02C8ru\u02D0l\u026A\u0261ast/",
  r\u00F6tm\u00E5nadshetta: "/\u02C8r\xF8\u02D0t\u02CCmo\u02D0nads\u02CCh\u025Bt\u02D0a/",
  runor: "/\u02C8r\u0289\u02D0n\u0254r/",
  saffransblommorna: "/\u02C8saf\u02D0rans\u02CCbl\u0254m\u02D0\u0254\u0273a/",
  scharlakanssol: "/\u02C8\u0267\u0251\u02D0rlakans\u02CCsu\u02D0l/",
  si: "/si\u02D0/",
  siljan: "/\u02C8s\u026Al\u02D0jan/",
  silverne: "/\u02C8s\u026Alv\u025B\u0273\u025B/",
  sk\u00E4ms: "/\u0267\u025Bm\u02D0s/",
  sn\u00F6dropparna: "/\u02C8sn\xF8\u02D0\u02CCdr\u0254p\u02D0a\u0273a/",
  sommarmorgon: "/\u02C8s\u0254mar\u02CCm\u0254r\u0261\u0254n/",
  spiralgalax: "/sp\u026A\u02C8r\u0251\u02D0l\u0261a\u02CClaks/",
  t\u00E4r: "/t\u025B\u02D0r/",
  telefonsamtal: "/t\u025Bl\u025B\u02C8fo\u02D0n\u02CCsam\u02CCt\u0251\u02D0l/",
  tillaga: "/t\u026Al\u02C8l\u0251\u02D0\u0261a/",
  tjugosex: "/\u02C8\u0255\u0289\u02D0\u0261\u0254\u02CCs\u025Bks/",
  upp\u00E5: "/\u02C8\u0275p\u02D0o\u02D0/",
  uppgr\u00E4vda: "/\u02C8\u0275p\u02D0\u02CC\u0261r\u025B\u02D0vda/",
  utflyttade: "/\u02C8\u0289\u02D0t\u02CCfl\u028Ft\u02D0ad\u025B/",
  utsiktspunkt: "/\u02C8\u0289\u02D0t\u02CCs\u026Akts\u02CCp\u0275\u014Bkt/",
  v\u00E4lvuxen: "/\u02C8v\u025B\u02D0l\u02CCv\u0275ks\u025Bn/",
  vanger: "/\u02C8va\u014B\u02D0\u025Br/",
  // surname
  verklighetsgrund: "/\u02C8v\u025B\u02D0kl\u026A\u0261\u02CChe\u02D0ts\u02CC\u0261r\u0275nd/",
  vilka: "/\u02C8v\u026Al\u02D0ka/",
  vingad: "/\u02C8v\u026A\u014B\u02D0ad/",
  voro: "/\u02C8vu\u02D0r\u0254/",
  york: "/j\u0254rk/",
  zephyrens: "/s\u025B\u02C8fy\u02D0r\u025Bns/"
};

// src/overrides/sw.ts
var sw = {
  // G2P handles most Swahili words. These overrides cover edge cases
  // that G2P can't handle (Arabic loanwords, foreign names).
  // Arabic loanword with "qi" — G2P doesn't have a rule for 'q'
  sadiqi: "/sa\u02C8di\u02D0ki/",
  ulithiqi: "/uli\u02C8\u03B8iki/",
  // Arabic "kh" = /x/ (voiceless velar fricative), not /kh/
  usiikhini: "/usii\u02C8xini/",
  // Foreign name — 'c' alone is not a Swahili phoneme
  Victoria: "/vikt\u0254\u02C8\u027Eia/"
};

// src/overrides/vi.ts
var vi = {
  chinh: "/t\u0255\u026A\u014B\u02E7/",
  li\u00EAu: "/li\u0259w\u02E7/",
  ng\u00E1c: "/\u014Bak\u02E7\u02E5/",
  tru\u00E2n: "/\u0288w\u0259n\u02E7/"
};

// src/foreign.ts
var IPA_SLASH_RE = /^\/|\/$/g;
var WHITESPACE_SPLIT_RE = /(\s+)/;
var WHITESPACE_RE = /^\s+$/;
var LEADING_NON_LETTER_RE = /^[^\p{L}\p{M}]/u;
var TRAILING_NON_LETTER_RE = /[^\p{L}\p{M}]$/u;
var CONTRACTION_SPLIT_RE = /(?<=['-])|(?=['-])/;
var khmerSegmenter = typeof Intl !== "undefined" && typeof Intl.Segmenter === "function" ? new Intl.Segmenter("km", { granularity: "word" }) : void 0;
function segmentKhmerText(text) {
  if (khmerSegmenter === void 0) {
    return text;
  }
  const normalized = text.replaceAll("\u200B", " ");
  const segments = [...khmerSegmenter.segment(normalized)];
  let result = "";
  for (let i = 0; i < segments.length; i++) {
    const seg = segments[i];
    result += seg.segment;
    const next = segments[i + 1];
    if (seg.isWordLike === true && next?.isWordLike === true) {
      result += " ";
    }
  }
  return result;
}
var LANGUAGES = [
  { code: "ar", label: "Arabic" },
  { code: "yue", label: "Cantonese" },
  { code: "nl", label: "Dutch" },
  { code: "eo", label: "Esperanto" },
  { code: "fi", label: "Finnish" },
  { code: "fr", label: "French" },
  { code: "de", label: "German" },
  { code: "is", label: "Icelandic" },
  { code: "ja", label: "Japanese" },
  { code: "km", label: "Khmer" },
  { code: "ko", label: "Korean" },
  { code: "ma", label: "Malay" },
  { code: "zh", label: "Mandarin" },
  { code: "nb", label: "Norwegian" },
  { code: "or", label: "Odia" },
  { code: "fa", label: "Persian" },
  { code: "pt", label: "Portuguese" },
  { code: "ro", label: "Romanian" },
  { code: "es", label: "Spanish" },
  { code: "sw", label: "Swahili" },
  { code: "sv", label: "Swedish" },
  { code: "vi", label: "Vietnamese" }
];
var IPA_WORD_OVERRIDES = {
  ar,
  de,
  eo,
  es,
  fa,
  fi,
  fr,
  is,
  ja,
  km,
  ko,
  ma,
  nb,
  nl,
  or: or_,
  pt,
  ro,
  sv,
  sw,
  vi
};
function ipaToIngglish(ipa) {
  const clean = ipa.replaceAll(IPA_SLASH_RE, "").replaceAll(".", "");
  const arpabet = ipaToArpabet(clean);
  return arpabetToIngglish(arpabet);
}
function lookupIpa(dict, word) {
  const { entries, lang } = dict;
  const override = getIpaOverride(lang, word) ?? getIpaOverride(lang, word.toLowerCase());
  if (override) {
    return override;
  }
  const lower = word.toLowerCase();
  const title = lower.charAt(0).toUpperCase() + lower.slice(1);
  const stripped = stripDiacritics(lower);
  if (entries[word] ?? entries[lower] ?? entries[title] ?? entries[stripped]) {
    return entries[word] ?? entries[lower] ?? entries[title] ?? entries[stripped];
  }
  if (lower.includes("\xDF")) {
    const ssLower = lower.replaceAll("\xDF", "ss");
    const ssTitle = ssLower.charAt(0).toUpperCase() + ssLower.slice(1);
    return entries[ssLower] ?? entries[ssTitle];
  }
  if (word.includes("'")) {
    const curly = word.replaceAll("'", "\u2019");
    const curlyLower = curly.toLowerCase();
    const curlyResult = entries[curly] ?? entries[curlyLower];
    if (curlyResult) {
      return curlyResult;
    }
  }
  if (Object.hasOwn(LEMMATIZERS, lang)) {
    const lemmaResult = LEMMATIZERS[lang](entries, lower);
    if (lemmaResult) {
      return lemmaResult;
    }
  }
  if (lang === "km") {
    const compound = lookupKhmerCompound(entries, word);
    if (compound !== void 0) {
      return compound;
    }
  }
  if (Object.hasOwn(G2P_CONVERTERS, lang)) {
    return G2P_CONVERTERS[lang](lower);
  }
  return void 0;
}
var khmerMergedDict;
var khmerDictKeys;
function applyDefaultStress(arpabet) {
  const hasStress = arpabet.some((p) => isVowel2(p) && getStress2(p) !== null);
  if (hasStress) {
    return arpabet;
  }
  const result = [...arpabet];
  for (let i = result.length - 1; i >= 0; i--) {
    if (isVowel2(result[i])) {
      result[i] = result[i] + "1";
      break;
    }
  }
  return result;
}
function decomposeKhmer(dict, keys, remaining, acc) {
  if (remaining.length === 0) {
    return acc;
  }
  for (const key of keys) {
    const ipa = dict[key];
    if (remaining.startsWith(key) && ipa !== void 0) {
      const result = decomposeKhmer(dict, keys, remaining.slice(key.length), [...acc, ipa]);
      if (result !== null) {
        return result;
      }
    }
  }
  return null;
}
function getIpaOverride(lang, word) {
  return IPA_WORD_OVERRIDES[lang]?.[word];
}
function ipaToFormat(ipa, format, lang) {
  const clean = ipa.replaceAll(IPA_SLASH_RE, "").replaceAll(".", "");
  const overrides = lang ? IPA_LANGUAGE_OVERRIDES[lang] : void 0;
  const arpabet = applyDefaultStress(ipaToArpabet(clean, overrides));
  return arpabetToFormat(arpabet, format, { disableRColoring: true });
}
function lookupKhmerCompound(entries, word) {
  if (khmerMergedDict === void 0) {
    khmerMergedDict = { ...entries };
    const overrides = IPA_WORD_OVERRIDES.km;
    if (overrides) {
      for (const [k, v] of Object.entries(overrides)) {
        khmerMergedDict[k] = v;
      }
    }
  }
  khmerDictKeys ?? (khmerDictKeys = Object.keys(khmerMergedDict).toSorted((a, b) => b.length - a.length));
  const parts = decomposeKhmer(khmerMergedDict, khmerDictKeys, word, []);
  if (parts === null || parts.length < 2) {
    return void 0;
  }
  return parts.map((p) => p.replaceAll(IPA_SLASH_RE, "")).join(" ");
}
var NOT_FOUND_MARKER = "\uFFFD";
var SENTENCE_END_RE = /[.!?。！？]$/;
function translateForeign(text, dict, format = "ingglish") {
  const tokens = translateForeignWithMapping(text, dict, format);
  return tokens.map((t) => !t.matched && t.isWord ? NOT_FOUND_MARKER + t.original : t.translated).join("");
}
function translateForeignWithMapping(text, dict, format = "ingglish") {
  const { lang } = dict;
  let atSentenceStart = true;
  const processed = lang === "km" ? segmentKhmerText(text) : text;
  const tokens = [];
  for (const segment of normalizeApostrophes(processed).split(WHITESPACE_SPLIT_RE)) {
    if (WHITESPACE_RE.test(segment)) {
      tokens.push({ isWord: false, matched: true, original: segment, translated: segment });
      continue;
    }
    if (!segment) {
      continue;
    }
    const leading = [];
    const trailing = [];
    let core = segment;
    while (core.length > 0 && LEADING_NON_LETTER_RE.test(core)) {
      leading.push(core[0]);
      core = core.slice(1);
    }
    while (core.length > 0 && TRAILING_NON_LETTER_RE.test(core)) {
      trailing.unshift(core.at(-1));
      core = core.slice(0, -1);
    }
    if (!core) {
      tokens.push({ isWord: false, matched: true, original: segment, translated: segment });
      continue;
    }
    let casePattern = detectCasePattern(core);
    const preservesCase = getFormatPreservesCase(format);
    if (atSentenceStart && preservesCase && casePattern === "lower" && isCaselessWord(core)) {
      casePattern = "capitalized";
    }
    atSentenceStart = SENTENCE_END_RE.test(trailing.join(""));
    const leadStr = leading.join("");
    const trailStr = trailing.join("");
    const ipa = lookupIpa(dict, core);
    if (ipa) {
      const translated = ipaToFormat(ipa, format, lang);
      const cased = preservesCase ? applyCasePattern(translated, casePattern) : translated;
      tokens.push({
        isWord: true,
        matched: true,
        original: segment,
        translated: leadStr + cased + trailStr
      });
      continue;
    }
    const parts = core.split(CONTRACTION_SPLIT_RE);
    if (parts.length > 1) {
      const partIpas = parts.map((part, i) => {
        if (part === "'" || part === "-") {
          return;
        }
        let ipa2;
        if (parts[i + 1] === "'") {
          ipa2 = lookupIpa(dict, part + "'");
        }
        ipa2 ?? (ipa2 = lookupIpa(dict, part));
        return ipa2;
      });
      const allFound = parts.every(
        (part, i) => part === "'" || part === "-" || partIpas[i] !== void 0
      );
      if (allFound) {
        const groups = [[]];
        for (const [i, part_] of parts.entries()) {
          const part = part_;
          if (part === "'") {
            continue;
          }
          if (part === "-") {
            groups.push([]);
            continue;
          }
          const ipa2 = partIpas[i];
          groups.at(-1).push(ipa2.replaceAll(IPA_SLASH_RE, "").replaceAll(".", ""));
        }
        const translated2 = groups.map((ipas) => ipaToFormat(ipas.join(""), format, lang)).join("-");
        const cased = preservesCase ? applyCasePattern(translated2, casePattern) : translated2;
        tokens.push({
          isWord: true,
          matched: true,
          original: segment,
          translated: leadStr + cased + trailStr
        });
        continue;
      }
      let isFirstPart = true;
      const translated = parts.map((part, i) => {
        if (part === "'" || part === "-") {
          return part;
        }
        const partCase = isFirstPart ? casePattern : detectCasePattern(part);
        isFirstPart = false;
        const partIpa = partIpas[i];
        if (partIpa) {
          const partTranslated = ipaToFormat(partIpa, format, lang);
          return preservesCase ? applyCasePattern(partTranslated, partCase) : partTranslated;
        }
        return NOT_FOUND_MARKER + part;
      });
      if (translated.some(
        (t, i) => parts[i] !== "'" && parts[i] !== "-" && !t.startsWith(NOT_FOUND_MARKER)
      )) {
        const translatedText = translated.map((t) => t.replaceAll(NOT_FOUND_MARKER, "")).join("");
        tokens.push({
          isWord: true,
          matched: false,
          original: segment,
          translated: leadStr + translatedText + trailStr
        });
        continue;
      }
    }
    tokens.push({ isWord: true, matched: false, original: segment, translated: segment });
  }
  return tokens;
}
function isCaselessWord(word) {
  const ch = word[0];
  return ch !== void 0 && ch.toUpperCase() === ch.toLowerCase();
}

// src/index.ts
function registerIPA() {
  registerFormat("ipa", {
    forward: arpabetToIPARaw,
    isLatinScript: true,
    joinSeparator: " ",
    label: "IPA",
    preservesCase: false
  });
}
export {
  LANGUAGES,
  NOT_FOUND_MARKER,
  arpabetPhonemeToIPA,
  arpabetToIPARaw,
  ipaToArpabetClean,
  ipaToIngglish,
  lookupIpa,
  registerIPA,
  segmentKhmerText,
  translateForeign,
  translateForeignWithMapping
};
